---
title: "Residence Time Experiment"
author: "Emmi Mueller"
date: "July 23, 2019"
output:
  pdf_document: default
  html_document: default
editor_options: 
  chunk_output_type: inline
---

##Setup
```{r setup, include = FALSE}
rm(list=ls())
getwd()

#package.list <- c("ggplot2", "vegan", "mgcv", "scales", "tidymv", "cowplot", "dplyr", "table1")

package.list <- c("Matrix", "jpeg", "cli", "png", "vegan", "ggplot2", "ggpubr", "cowplot", "ggpmisc", "scales", "fitdistrplus", "tidyr", "ade4", "viridis", "gplots", "BiodiversityR", "indicspecies", "knitr", "here", "dplyr", "BiocManager", "car", "here", "actuar", "tibble", "scales", "dplyr", "stringr", "iNEXT", "lmtest", "betapart", "reshape2", "mgcv", "modelr", "purrr", "lme4", "tidymv", "merTools", "MuMIn", "htmltools", "ggrepel", "broom", "tidyverse", "pander", "spaa", "lsr", "fpc", "table1", "ape", "devtools", "phytools", "phylobase", "diptest")

for (package in package.list) {
  if (!require(package, character.only=T, quietly=T)) { 
    install.packages(package, dependencies = TRUE)
    library(package, character.only=T)
  } }

# BiocManager::install()
BiocManager::install(c("flowCore", "ggcyto", "ggtree"))

#devtools::install_github("BlakeRMills/MoMAColors")
library(MoMAColors)

source("./code/residence-time-test_functions.R")
source("./code/mothur_tools.R")
```

##Set up figure theme
```{r figure_setup}
my.cols <- RColorBrewer::brewer.pal(n = 4, name = "Greys")[3:4]

# Set theme for figures in the paper
theme_set(theme_classic() + 
  theme(axis.title = element_text(size = 16),
        axis.title.x = element_text(margin = margin(t = 15, b = 15)),
        axis.title.y = element_text(margin = margin(l = 15, r = 15)),
        axis.text = element_text(size = 14),
        axis.text.x = element_text(margin = margin(t = 5)),
        axis.text.y = element_text(margin = margin(r = 5)),
        #axis.line.x = element_line(linewidth = 1),
        #axis.line.y = element_line(linewidth = 1),
        axis.line.x = element_blank(),
        axis.line.y = element_blank(),
        axis.ticks.x = element_line(linewidth = 1),
        axis.ticks.y = element_line(linewidth = 1),
        axis.ticks.length = unit(.1, "in"),
        panel.border = element_rect(color = "black", fill = NA, linewidth = 1.5),
        legend.text = element_text(size = 10),
        strip.text = element_text(size = 14),
        strip.background = element_blank()
        ))
```


```{r}
#Define Inputs

#Tau = general design file for experiment, paired Day 0 and 20
Tau <- read.csv("data/RTLC/2021_RTLC_Tau_Combined.csv", header = TRUE)
Tau_ctrl <- Tau[50:51,]
Tau <- Tau[1:49,]
Tau$Tau <- as.numeric(Tau$Tau)
Tau$Set <- as.character(Tau$Set)
Tau$Day_0_Seq[Tau$Day_0_Seq == ""] <- NA
Tau$Day_20_Seq[Tau$Day_20_Seq == ""] <- NA

#Tau_Seq = Full list of sequenced samples, non-paired Day 0 and 20
Tau_Seq <- read.csv("data/RTLC/2021_RTLC_Tau_Seq.csv", header = TRUE)

#Import Shared Files
OTUs <- read.otu(shared = "mothur/RTLC_A+B/RTLC.final.shared", cutoff = "0.03")
rownames(OTUs)[rownames(OTUs) == "GSF3365_RTLC"] <- "GSF3365_RTLC_002"

ASVs <- read.csv("data/RTLC_ASV.csv", header = TRUE, sep = ",")
ASVs$X <- sub("-", "_", ASVs$X)
rownames(ASVs) <- ASVs$X
ASVs <- ASVs[,-1]

#Import Taxonomy
OTU.tax <- read.tax(taxonomy = "mothur/RTLC_A+B/RTLC.final.taxonomy", format = "rdp")

#Abundance data
#Tau <- N_fxn(read.csv("data/RTLC/RTLC_S1/20210602_RTLC_S1_N.csv", header = TRUE), Tau)
Tau <- N_fxn(read.csv("data/RTLC/RTLC_S2/20210628_RTLC_S2_N_M1gate.csv", header = TRUE), Tau)
Tau <- N_fxn(read.csv("data/RTLC/RTLC_S3/20210723_RTLC_S3_N_M1gate.csv", header = TRUE), Tau)
Tau <- N_fxn(read.csv("data/RTLC/RTLC_S4/20210904_RTLC_S4_N_M1gate.csv", header = TRUE), Tau)


#Biofilm data
Tau <- OT_fxn(read.csv("data/RTLC/RTLC_S1/20210602_RTLC_S1_Otoole.csv", header = TRUE), Tau)
Tau <- OT_fxn(read.csv("data/RTLC/RTLC_S2/20210628_RTLC_S2_Otoole.csv", header = TRUE), Tau)
Tau <- OT_fxn(read.csv("data/RTLC/RTLC_S3/20210723_RTLC_S3_Otoole.csv", header = TRUE), Tau)
Tau <- OT_fxn(read.csv("data/RTLC/RTLC_S4/20210904_RTLC_S4_Otoole.csv", header = TRUE), Tau)

#BP data
Tau <- as.data.frame(BP_fxn(read.csv("data/RTLC/RTLC_S1/20210602_RTLC_S1_BP.csv", header = FALSE), Tau, 1))
Tau <- as.data.frame(BP_fxn(read.csv("data/RTLC/RTLC_S2/20210628_RTLC_S2_BP.csv", header = FALSE), Tau, 2))
Tau <- as.data.frame(BP_fxn(read.csv("data/RTLC/RTLC_S3/20210723_RTLC_S3_BP.csv", header = FALSE), Tau, 3))
Tau <- as.data.frame(BP_fxn(read.csv("data/RTLC/RTLC_S4/20210904_RTLC_S4_BP.csv", header = FALSE), Tau, 4))
Tau$ind_P <- Tau$uMChr/Tau$N
Tau$ind_P_g <- (Tau$gCLhr/1000)/Tau$N
Tau$turnover_t <- (26e-15)/Tau$ind_P_g


#Biolog data
Tau <- EP_fxn(read.csv("data/RTLC/eco.data_rt.txt", header = TRUE, sep = "\t"), Tau)

Tau$Set <- as.factor(Tau$Set)
Tau_Seq$Tau <- as.numeric(Tau_Seq$Tau)


Tau$Tau <- log(((10^Tau$Tau)/60), 10)
Tau_Seq$Tau <- log(((10^Tau_Seq$Tau)/60), 10)

# Tau$N_turn <- (20 * 24)/(10^Tau$Tau)
# 
# turnover <- log(20*24, 10)
# 
# Tau$Turn <- Tau$Tau <= log(24*20, 10)
# 
# Tau$Pump <- Tau$Tau < 1.6
```



##Read in all EcoPlate 48 hour data and create a resource by sample matrix (wbs - well response by site) and environmental matrix (env - Tau only)

```{r}
EcoPlate <- read.csv("data/RTLC/eco.data_rt_S1_48.txt", header = TRUE, sep = "\t")
EcoPlate <- rbind(EcoPlate, read.csv("data/RTLC/eco.data_rt_S2_48.txt", header = TRUE, sep = "\t"))
EcoPlate <- rbind(EcoPlate, read.csv("data/RTLC/eco.data_rt_S3_48.txt", header = TRUE, sep = "\t"))
EcoPlate <- rbind(EcoPlate, read.csv("data/RTLC/eco.data_rt_S4_48.txt", header = TRUE, sep = "\t"))

x <- 1
while(x < nrow(EcoPlate)){
  EcoPlate[x, "NumRes"] <- sum(EcoPlate[x, 4:34] > 0.125)
  x <- x + 1
}

Tau$NumRes <- EcoPlate$NumRes

EcoPlate_env <- subset(EcoPlate, select = c(Tau, Set))
EcoPlate$Tau <- log(((10^EcoPlate$Tau)/60), 10)
rownames(EcoPlate) <- EcoPlate$Tau
EcoPlate <- subset(EcoPlate, select =-c(Tau, NumRes, Avg, Set, Hours))
```

##Load Resource Use data for first 72 hours
```{r}
EcoPlate_72 <- read.csv("data/RTLC/eco.data_rt_S1_0.txt", header = TRUE, sep = "\t")
file_list <- c("data/RTLC/eco.data_rt_S1_24.txt", "data/RTLC/eco.data_rt_S1_48.txt", "data/RTLC/eco.data_rt_S1_72.txt", "data/RTLC/eco.data_rt_S2_0.txt", "data/RTLC/eco.data_rt_S2_24.txt", "data/RTLC/eco.data_rt_S2_48.txt", "data/RTLC/eco.data_rt_S2_72.txt", "data/RTLC/eco.data_rt_S3_0.txt", "data/RTLC/eco.data_rt_S3_24.txt", "data/RTLC/eco.data_rt_S3_48.txt", "data/RTLC/eco.data_rt_S3_72.txt", "data/RTLC/eco.data_rt_S4_0.txt", "data/RTLC/eco.data_rt_S4_24.txt", "data/RTLC/eco.data_rt_S4_48.txt", "data/RTLC/eco.data_rt_S4_72.txt")

for(file in file_list){
  EcoPlate_72 <- rbind(EcoPlate_72, read.csv(file, header = TRUE, sep = "\t"))
}

EcoPlate_72$Tau <- log(((10^EcoPlate_72$Tau)/60), 10)

long_EP <- EcoPlate_72 %>% gather(C_Source, OD, X2.Hydroxy.Benzoic.Acid:Avg)
```

##Load slopes of resource use data (48-24 hours)
```{r}
EcoPlate_S1 <- read.csv("data/RTLC/eco.data_rt_S1_48.txt", header = TRUE, sep = "\t")
EcoPlate_S1 <- cbind(EcoPlate_S1[,1:2], EcoPlate_S1[,4:34])

EcoPlate_S2 <- read.csv("data/RTLC/eco.data_rt_S2_48.txt", header = TRUE, sep = "\t")
EcoPlate_S2 <- cbind(EcoPlate_S2[,1:2], EcoPlate_S2[,4:34])

EcoPlate_S3 <- read.csv("data/RTLC/eco.data_rt_S3_48.txt", header = TRUE, sep = "\t")
EcoPlate_S3 <- cbind(EcoPlate_S3[,1:2], EcoPlate_S3[,4:34])

EcoPlate_S4 <- read.csv("data/RTLC/eco.data_rt_S4_48.txt", header = TRUE, sep = "\t")
EcoPlate_S4 <- cbind(EcoPlate_S4[,1:2], EcoPlate_S4[,4:34])


EcoPlate_48 <- rbind(EcoPlate_S1, EcoPlate_S2, EcoPlate_S3, EcoPlate_S4)

ResType <- read.csv("code/resource_type.txt", header = TRUE, sep = "\t")

colnames(EcoPlate_48) <- c("Tau", "Set", "2-Hydroxy.Benzoic.Acid", "4-Hydroxy.Benzoic.Acid", "alpha-Cyclodextrin", "alpha-D-Lactose", "alpha-Ketobutyric.Acid", "beta-Methyl-D-Glucoside", "D-Cellobiose", "D-Galactonic.Acid.gamma-Lactone", "D-Galacturonic.Acid","D-Glucosaminic.Acid", "D-Malic.Acid", "D-Mannitol", "D-Xylose", "D,L-alpha-Glycerol.Phosphate","gamma-Hydroxybutyric.Acid", "Glucose-1-Phosphate", "Glycogen", "Glycyl-L-Glutamic.Acid", "i-Erythritol","Itaconic.Acid","L-Arginine", "L-Asparagine","L-Phenylalanine", "L-Serine", "L-Threonine","N-Acetyl-D-Glucosamine", "Phenylethylamine", "Putrescine", "Pyruvic.Acid.Methyl.Ester", "Tween.40", "Tween.80")

EcoPlate_48$Tau <- log(((10^EcoPlate_48$Tau)/60), 10)


long_EP_48 <- EcoPlate_48 %>% gather(C_Source, OD_48, "2-Hydroxy.Benzoic.Acid":"Tween.80")

ResType <- distinct(ResType)

long_EP_48$Type <- NA

for(row in rownames(long_EP_48)){
  long_EP_48[row,"Type"] <- ResType$Type[ResType$Resource == long_EP_48[row,"C_Source"]]
}


```

##Clean and rarefy the OTU table
```{r}
#Remove samples with fewer than 10000 reads
coverage <- rowSums(OTUs)
cutoff <- 10000
lows <- which(coverage < cutoff)
if (length(lows) > 0){
  OTUs <- OTUs[-which(coverage < cutoff),]
}

#Remove OTUs with less thatn 2 reads across all samples
OTUs <- OTUs[,which(colSums(OTUs) > 2)]

#Generate rarefacation plot
otu.min <- min(rowSums(OTUs))
asv.min <- min(rowSums(ASVs))

# png("./output/OTU_rarefaction.png", width = 800, height = 480)
# par(mar = c(1, 1, 1, 1) + 0.1)
#   rarecurve(x = OTUs, step = 100, sample = otu.min, col = "blue", cex = 0.6, las = 1)
# #  abline(0,1, col = "red")
# #  text(1500, 1500, "1:1", pos = 2, col = "red")
# dev.off()

#rarefy OTUs to the sample with the lowest number of reads then remove OTUs with 0 reads after rarefaction
set.seed(47405)
OTU_r <- rrarefy(OTUs, otu.min)
OTU_r <- OTU_r[,-which(colSums(OTU_r) == 0)]
rm(OTUs)

ASV_r <- rrarefy(ASVs, asv.min)
ASV_r <- ASV_r[,-which(colSums(ASV_r) == 0)]
rm(ASVs)

ASVsr_20 <- ASV_r[rownames(ASV_r) %in% unique(Tau$Day_20_Seq),]
ASVsr_20 <- ASVsr_20[,which(colSums(ASVsr_20) >0)]
rownames(ASVsr_20) <- subset(Tau_Seq, Day == 20)$Tau

Tau_20 <- subset(Tau[is.na(Tau$Day_20_Seq) == 0,])
Tau_Seq$Day <- as.factor(Tau_Seq$Day)
Tau_Seq$Tau <- as.numeric(Tau_Seq$Tau)

#Generate OTU tables with p/a, relative abundance
OTUsREL <-decostand(OTU_r, method = "total")
OTUsREL_20 <- OTUsREL[rownames(OTUsREL) %in% unique(Tau$Day_20_Seq),]
OTUsREL_20 <- OTUsREL_20[,which(colSums(OTUsREL_20) > 0)]
rownames(OTUsREL_20) <- subset(Tau_Seq, Day == 20)$Tau
# 
OTUs.PA <- decostand(OTU_r, method = "pa")
OTUsPA_20<- OTUs.PA[rownames(OTUs.PA) %in% unique(Tau$Day_20_Seq),]
OTUsPA_20 <- OTUsPA_20[,which(colSums(OTUsPA_20) > 0)]
rownames(OTUsPA_20) <- subset(Tau_Seq, Day == 20)$Tau


OTUsr_20 <- OTU_r[rownames(OTU_r) %in% unique(Tau$Day_20_Seq),]
OTUsr_20 <- OTUsr_20[,which(colSums(OTUsr_20) >0)]
rownames(OTUsr_20) <- subset(Tau_Seq, Day == 20)$Tau

OTU.tax.20 <- OTU.tax[OTU.tax$OTU %in% unique(colnames(OTUsr_20)),]

Tau_Seq$ACE <- S.ace(OTU_r)

S <- as.data.frame(cbind(row.names(OTU_r), unname(S.cal(OTU_r))))
S_ASV <- as.data.frame(cbind(row.names(ASV_r), unname(S.cal(ASV_r))))

SimpE <- as.data.frame(cbind(row.names(OTU_r), unname(SimpE.cal(OTU_r))))
SimpE_ASV <- as.data.frame(cbind(row.names(ASV_r), unname(SimpE.cal(ASV_r))))
```
#Figure S2. Diversity metrics with ASVs
##Observed richness - ASVs
```{r}
Tau <- S_fxn(S_ASV, Tau)
Tau$Day_20.S <- as.numeric(Tau$Day_20.S)
Tau$Day_0.S <- as.numeric(Tau$Day_0.S)

Tau <- SimpE_fxn(SimpE_ASV, Tau)
Tau$Day_0.SimpE <- as.numeric(Tau$Day_0.SimpE)
Tau$Day_20.SimpE <- as.numeric(Tau$Day_20.SimpE)

write.csv(Tau, "./data/Tau_ASV.csv")

S_gam <- gam(Day_20.S ~ s(Tau, k = 14), family = gaussian(link = "identity"), data = Tau, method = "REML")
S_gam_re <- gam(Day_20.S ~ s(Tau)  + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")

AIC(S_gam, S_gam_re)
anova(S_gam, S_gam_re, test = "Chisq")

summary(S_gam)
k.check(S_gam)
gam.check(S_gam)
mean(summary(S_gam)$s.table[,4])

S.ob_Tau_ASV <- predict_gam(S_gam) %>%
  ggplot(aes(Tau, fit))+
#  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_smooth_ci(color = "#008d98", cex = 2, ci_alpha= 0.2)+
  geom_point(data = Tau, aes(x = Tau, y = Day_20.S))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab("Observed richness")+
  theme(axis.title.x = element_blank())+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(S_gam)$dev, 2))))+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.5), "cm"))

S.ob_Tau_ASV
```

##Species evenness - ASVs
```{r}
SimpE_gam <- gam(Day_20.SimpE ~ s(Tau, k = 15), family = gaussian(link = "identity"), data = Tau, method = "REML")
SimpE_gam_re <- gam(Day_20.SimpE ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")

AIC(SimpE_gam, SimpE_gam_re)
anova(SimpE_gam, SimpE_gam_re, test = "Chisq")

summary(SimpE_gam_re)
k.check(SimpE_gam_re)
gam.check(SimpE_gam_re)
mean(summary(SimpE_gam_re)$s.table[,4])

SimpE_Tau_ASV <- predict_gam(SimpE_gam_re, exclude_terms = "s(Set)") %>%
  filter(Set == "1") %>%
  ggplot(aes(Tau, fit))+
#  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_smooth_ci(color = "#008d98", cex = 2, ci_alpha= 0.2)+
  geom_point(data = Tau, aes(x = Tau, y = Day_20.SimpE))+
#  scale_y_continuous(breaks = c(0, 0.02, 0.04, 0.06, 0.08, 0.1), limits = c(0, 0.11))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab(expression(paste("Evenness", " (E"[D^"-1"/S], ")")))+
  xlab(expression(paste("Residence time (", tau, ", h)")))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(SimpE_gam_re)$dev, 2))))

SimpE_Tau_ASV
```

##Draw figure
```{r}
ggdraw()+
  draw_plot(S.ob_Tau_ASV, x = 0, y = 0.55, width = 1, height = 0.45)+
  draw_plot(SimpE_Tau_ASV, x = 0.04, y = 0, width = 0.96, height = 0.55)+
  draw_plot_label(label = c("(a)", "(b)"), size = 20, x=c(-0.02, -0.02), y = c(1, 0.55))+
  theme(plot.background = element_rect(fill= "white", color = NA))

ggsave("./output/CommStr_Cstat_ASV.pdf")
ggsave("./output/CommStr_Cstat_ASV.png", width = 5, height = 6)

```



#Figure 1. Community Function
##Microbial abundance
```{r}
# test <- as.data.frame(seq(0, 0.1, 0.0001))
# colnames(test) <- "n"
# for(i in seq(0, 0.1, 0.0001)){
#   N_gam_re <- gam(log_N ~ s(Tau, sp = i) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")
#   test[test$n == i, "AIC"] <- N_gam_re$aic
#   test[test$n == i, "REML"] <- N_gam_re$gcv.ubre[[1]]
# }
# 
# plot(test$AIC ~ test$n)
# plot(test$REML ~ test$n, ylim = c(0,30))

N_gam <- gam(log_N ~ s(Tau), family = gaussian(link = "identity"), data = Tau, method = "REML")
N_gam_re <- gam(log_N ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")

AIC(N_gam, N_gam_re)
anova(N_gam, N_gam_re, test = "Chisq")

x <- predict_gam(N_gam_re)

summary(N_gam)
k.check(N_gam)
gam.check(N_gam)

N_Tau <- predict_gam(N_gam) %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "#008d98", cex = 2, ci_alpha= 0.2)+
  geom_point(data = Tau, aes(x = Tau, y = log_N))

N_Tau

summary(N_gam_re)
k.check(N_gam_re)
gam.check(N_gam_re)
mean(summary(N_gam_re)$s.table[,4])

gam.vcomp(N_gam_re)

N_Tau <- predict_gam(N_gam_re, exclude_terms = "s(Set)") %>%
  filter(Set == "2") %>% 
  ggplot(aes(Tau, fit))+
#  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_smooth_ci(color = "#008d98", cex = 2, ci_alpha= 0.2)+
  geom_point(data = Tau, aes(x = Tau, y = log_N))+
  theme(axis.title.x = element_blank())+
  ylab("Abundance, N \n(cells/mL)")+
  scale_x_continuous(labels = label_math(10^.x))+
  scale_y_continuous(breaks = c(6, 7, 8), labels = label_math(10^.x))+
  labs(title = bquote("Deviance explained =" ~ .(round_pad(summary(N_gam_re)$dev, 2))))+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.8), "cm"))+
  theme(axis.title.y = element_text(vjust = 6))

N_Tau

summary(N_gam_re)$dev

ggsave("./output/RTLC_N.pdf")
ggsave("./output/RTLC_N.png")
```

##Biomass Production (uM C/hr)
```{r}
# BP_lm <- lm(uMChr ~ Tau, data = Tau)
# BP_poly <- lm(uMChr ~ poly(Tau, 2, raw = TRUE), data = Tau)
# BP_gam <- gam(uMChr ~ s(Tau), family = gaussian(link = "identity"), data = Tau, method = "REML")
# BP_gam_gm <- gam(uMChr ~ s(Tau), family = Gamma(link = "log"), data = Tau, method = "REML")
# 
# AIC(BP_lm, BP_poly, BP_gam, BP_gam_gm)

BP_gam_re <- gam(log(uMChr, 10) ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")
BP_gam <- gam(log(uMChr, 10) ~ s(Tau), family = gaussian(link = "identity"), data = Tau, method = "REML")
anova(BP_gam, BP_gam_re, test = "Chisq")

summary(BP_gam_re)
mean(summary(BP_gam_re)$s.table[,4])
k.check(BP_gam_re)
gam.check(BP_gam_re)

BP_Tau <- predict_gam(BP_gam_re, exclude_terms = "s(Set)") %>%
  filter(Set == "1") %>% 
  ggplot(aes(Tau, fit))+
#  geom_vline(xintercept = turnover, linetype = "dashed", color = "blue", size = 1)+
  geom_smooth_ci(color = "#008d98", cex = 2, ci_alpha= 0.2)+
  geom_point(data = Tau, aes(x = Tau, y = log(uMChr, 10)))+
  xlab(expression(paste("Residence time (", tau, ", h)")))+
  ylab(expression(atop("Bacterial productivity", paste("(", mu, "mol C/hr)"))))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(limits = c(-3, -1), breaks = c(-1, -2, -3), labels = label_math(expr = 10^.x, format = force))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(BP_gam_re)$dev, 2))))

BP_Tau

ggsave("./output/RTLC_BP.pdf")
ggsave("./output/RTLC_BP.png", width = 6.5, height = 5)
```

##Average well response at 48 hours
```{r}

#cor.test(Tau_20$Tau, tau.plot$NumRes, method="spearman", exact=FALSE)

Avg_lm <- lm(AvgRes ~ Tau, data = Tau)
Avg_gam <- gam(AvgRes ~ s(Tau), data = Tau, family = gaussian(link = "identity"), method = "REML")
Avg_gam_re <- gam(AvgRes ~ s(Tau) + s(Set, bs = "re"), data = Tau, family = gaussian(link = "identity"), method = "REML")

anova(Avg_gam, Avg_gam_re, test = "Chisq")
AIC(Avg_lm, Avg_gam, Avg_gam_re)

summary(Avg_gam_re)
mean(summary(Avg_gam_re)$s.table[,4])

AvgRes <- predict_gam(Avg_gam_re, exclude_terms = "s(Set)") %>%
  filter(Set == "1") %>% 
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "#008d98", cex = 2, ci_alpha= 0.2)+
  geom_point(data = Tau, aes(x = Tau, y = AvgRes))+
  xlab(expression(paste("Residence time (", tau, ", h)")))+
  ylab("Average resource use\n(OD590)")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(Avg_gam_re)$dev, 2))))+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.8), "cm"))+
  theme(axis.title.y = element_text(vjust = 6))

AvgRes

ggsave("./output/RTLC_AvgRes.pdf")
ggsave("./output/RTLC_AvgRes.png", width = 6.5, height = 5)

```

##Draw Figure
```{r}
ggdraw()+
  draw_plot(N_Tau, x = 0.5, y = 0.55, width = 0.5, height = 0.45)+ 
  draw_plot(BP_Tau, x = 0, y = 0, width = 0.5, height = 0.55)+ 
  draw_plot(AvgRes, x = 0.5, y = 0, width = 0.5, height = 0.55)+
  draw_plot_label(label = c("(a)", "(b)", "(c)", "(d)"), size = 20, x = c(0,0.5,0,0.5), y = c(1,1,0.55,0.55))+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("./output/N_productivity_Cstat.pdf")
ggsave("./output/N_productivity_Cstat.png", width = 12, height = 8)
```



#Figure 2. Microbial diversity
```{r}
Tau <- S_fxn(S, Tau)
Tau$Day_20.S <- as.numeric(Tau$Day_20.S)
Tau$Day_0.S <- as.numeric(Tau$Day_0.S)

Tau <- SimpE_fxn(SimpE, Tau)
Tau$Day_0.SimpE <- as.numeric(Tau$Day_0.SimpE)
Tau$Day_20.SimpE <- as.numeric(Tau$Day_20.SimpE)
```



```{r}
ggplot(data = Tau, aes(x = Day_20.S, y = uMChr, color = Tau))+
  geom_point()+
  xlab(expression(paste("Observed richness")))+
  ylab(expression(atop("Bacterial productivity", paste("(", mu, "mol C/hr)"))))+
  scale_color_gradientn(colors = moma.colors("ustwo"), labels = label_math(expr = 10^.x, format = force))

```

##Observed Richness - OTUs
```{r}
# test <- as.data.frame(seq(0, 0.1, 0.0001))
# colnames(test) <- "n"
# for(i in seq(0, 0.1, 0.0001)){
#   S_gam_re <- gam(log(Day_20.S, 10) ~ s(Tau, sp = i) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")
#   test[test$n == i, "AIC"] <- S_gam_re$aic
#   test[test$n == i, "REML"] <- S_gam_re$gcv.ubre[[1]]
# }
# 
# plot(test$AIC ~ test$n)
# plot(test$REML ~ test$n, ylim = c(-50,-20))

S_gam <- gam(Day_20.S ~ s(Tau, k = 14), family = gaussian(link = "identity"), data = Tau, method = "REML")
S_gam_re <- gam(Day_20.S ~ s(Tau)  + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")

k.check(S_gam)
gam.check(S_gam)

AIC(S_gam, S_gam_re)
anova(S_gam, S_gam_re, test = "Chisq")

summary(S_gam_re)
k.check(S_gam_re)
gam.check(S_gam_re)
mean(summary(S_gam_re)$s.table[,4])

S.ob_Tau <- predict_gam(S_gam_re, exclude_terms = "s(Set)") %>%
  filter(Set == "1") %>%
  ggplot(aes(Tau, fit))+
#  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_smooth_ci(color = "#008d98", cex = 2, ci_alpha= 0.2)+
  geom_point(data = Tau, aes(x = Tau, y = Day_20.S))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab("Observed richness")+
  theme(axis.title.x = element_blank())+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(S_gam)$dev, 2))))
#  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.5), "cm"))
  

S.ob_Tau

ggsave("./output/RTLC_S.pdf")
ggsave("./output/RTLC_S.png")
```

##Simpson's Evenness (E[1/D]) - OTUs
```{r}
# test <- as.data.frame(seq(0, 0.1, 0.0001))
# colnames(test) <- "n"
# for(i in seq(0, 0.1, 0.0001)){
#   SimpE_gam <- gam(Day_20.SimpE ~ s(Tau, sp = i), family = gaussian(link = "identity"), data = Tau, method = "REML")
#   test[test$n == i, "AIC"] <- SimpE_gam$aic
#   test[test$n == i, "REML"] <- SimpE_gam$gcv.ubre[[1]]
# }
# 
# plot(test$AIC ~ test$n)
# plot(test$REML ~ test$n)

SimpE_gam <- gam(Day_20.SimpE ~ s(Tau, k = 15), family = gaussian(link = "identity"), data = Tau, method = "REML")
SimpE_gam_re <- gam(Day_20.SimpE ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")

AIC(SimpE_gam, SimpE_gam_re)
anova(SimpE_gam, SimpE_gam_re, test = "Chisq")

summary(SimpE_gam)
k.check(SimpE_gam)
gam.check(SimpE_gam)
mean(summary(SimpE_gam)$s.table[,4])

SimpE_Tau <- predict_gam(SimpE_gam) %>%
  ggplot(aes(Tau, fit))+
#  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_smooth_ci(color = "#008d98", cex = 2, ci_alpha= 0.2)+
  geom_point(data = Tau, aes(x = Tau, y = Day_20.SimpE))+
  scale_y_continuous(breaks = c(0, 0.02, 0.04, 0.06, 0.08, 0.1), limits = c(0, 0.11))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
#  ylab(expression("E"[D^"-1"/S]))+
  ylab(expression(paste("Evenness", " (E"[D^"-1"/S], ")")))+
  xlab(expression(paste("Residence time (", tau, ", h)")))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(SimpE_gam)$dev, 2))))

SimpE_Tau

ggsave("./output/RTLC_SimpE.pdf")
ggsave("./output/RTLC_SimpE.png")

```

##Draw figure
```{r}
ggdraw()+
  draw_plot(S.ob_Tau, x = 0, y = 0.55, width = 1, height = 0.45)+
  draw_plot(SimpE_Tau, x = 0, y = 0, width = 1, height = 0.55)+
  draw_plot_label(label = c("(a)", "(b)"), size = 20, x=c(0, 0), y = c(1.01, 0.57))+
  theme(plot.background = element_rect(fill= "white", color = NA))

ggsave("./output/Diversity_Cstat.pdf")
ggsave("./output/Diversity_Cstat.png", width = 5, height = 6)
```


```{r}
tau.db <- vegdist(OTU_r, method = "bray", upper = TRUE, diag = TRUE)

tau.pcoa <- cmdscale(tau.db, eig = TRUE, k = 3)
explainvar1 <- round(tau.pcoa$eig[1] / sum(tau.pcoa$eig), 3) *100
explainvar2 <- round(tau.pcoa$eig[2] / sum(tau.pcoa$eig), 3) *100
explainvar3 <- round(tau.pcoa$eig[3] / sum(tau.pcoa$eig), 3) *100
sum.eig <- sum(explainvar3, explainvar2, explainvar1)

tau.plot <- as.data.frame(tau.pcoa$points)
tau.plot <- cbind(tau.plot, Tau_Seq)


PCoA_D0 <- ggplot(tau.plot, aes(x = V1, y = V2))+
  geom_point(cex = 3, aes(color = as.numeric(Tau), shape = Day))+
  xlab(paste("PCoA 1 (", explainvar1, "%)", sep = ""))+
  ylab(paste("PCoA 2 (", explainvar2, "%)", sep = ""))+
  labs(color = expression(paste(tau,  " (h)")))+
  scale_color_gradientn(colors = moma.colors("ustwo"), labels = label_math(expr = 10^.x, format = force))+
  xlim(c(-0.35, 0.3))+
#  scale_color_viridis(direction = -1, labels = label_math(expr = 10^.x, format = force))+
  theme(legend.title = element_text(size = 15), legend.position = c(0.1, 0.5), legend.text = element_text(size = 15))

PCoA_D0

ggsave("./output/RTLC_BC_PCoA_D0.pdf")
ggsave("./output/RTLC_BC_PCoA_D0.png", width = 10, height = 10)

```


#Figure 3. Community composition & PCoA axes
```{r}
tau.db <- vegdist(OTUsr_20, method = "bray", upper = TRUE, diag = TRUE)

tau.pcoa <- cmdscale(tau.db, eig = TRUE, k = 3)
explainvar1 <- round(tau.pcoa$eig[1] / sum(tau.pcoa$eig), 3) *100
explainvar2 <- round(tau.pcoa$eig[2] / sum(tau.pcoa$eig), 3) *100
explainvar3 <- round(tau.pcoa$eig[3] / sum(tau.pcoa$eig), 3) *100
sum.eig <- sum(explainvar3, explainvar2, explainvar1)

tau.plot <- as.data.frame(tau.pcoa$points)
tau.plot <- cbind(tau.plot, Tau_20)
tau.plot$Pump <- factor(tau.plot$Pump, levels=c("TRUE", "FALSE"))
tau.plot$Turn <- factor(tau.plot$Turn, levels=c("TRUE", "FALSE"))


tau.plot.order <- tau.plot[order(as.numeric(tau.plot$Tau)),]
OTUsr_20.order <- OTUsr_20[order(as.numeric(row.names(OTUsr_20))),]

cor.test(x =tau.plot.order$Tau, y = tau.plot.order$V1)
cor.test(x = tau.plot.order$Tau, y = tau.plot.order$V2)

PCoA <- ggplot(tau.plot.order, aes(x = V1, y = V2))+
  geom_path(linewidth = 1.25, color = alpha("black", 0.2))+
  geom_point(cex = 3, aes(color = as.numeric(Tau)))+
  xlab(paste("PCoA 1 (", explainvar1, "%)", sep = ""))+
  ylab(paste("PCoA 2 (", explainvar2, "%)", sep = ""))+
  labs(color = expression(paste(tau,  " (h)")))+
  scale_color_gradientn(colors = moma.colors("ustwo"), labels = label_math(expr = 10^.x, format = force))+
#  scale_color_viridis(direction = -1, labels = label_math(expr = 10^.x, format = force))+
  theme(legend.title = element_text(size = 15), legend.position = c(0.8, 0.2), legend.text = element_text(size = 15))

PCoA

ggsave("./output/RTLC_BC_PCoA.pdf")
ggsave("./output/RTLC_BC_PCoA.png", width = 10, height = 10)

PCoA1_gam <- gam(V1 ~ s(Tau), family = gaussian(link = "identity"), data = tau.plot, method = "REML")
PCoA1_gam_re <- gam(V1 ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = tau.plot, method = "REML")

anova(PCoA1_gam, PCoA1_gam_re, test = "Chisq")

AIC(PCoA1_gam, PCoA1_gam_re)
summary(PCoA1_gam_re)
mean(summary(PCoA1_gam_re)$s.table[,4])

PCoA1 <- predict_gam(PCoA1_gam_re, exclude_terms = "s(Set)") %>%
  filter(Set == "1") %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "#008d98", cex = 2, ci_alpha= 0.2)+
  geom_point(data = tau.plot, aes(x = Tau, y = V1))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab(paste("PCoA 1 (", explainvar1, "%)", sep = ""))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(PCoA1_gam_re)$dev, 2))))+
  theme(axis.title.x = element_blank())

PCoA1

ggsave("./output/RTLC_BC_PCoA1.pdf")
ggsave("./output/RTLC_BC_PCoA1.png", width = 6.5, height = 5)

PCoA2_gam <- gam(V2 ~ s(Tau), family = gaussian(link = "identity"), data = tau.plot, method = "REML")
PCoA2_gam_re <- gam(V2 ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = tau.plot, method = "REML")
summary(PCoA2_gam)
mean(summary(PCoA2_gam)$s.table[,4])
AIC(PCoA2_gam_re, PCoA2_gam)
anova(PCoA2_gam, PCoA2_gam_re, test = "Chisq")

PCoA2 <- predict_gam(PCoA2_gam) %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "#008d98", cex = 2, ci_alpha= 0.2)+
  geom_point(data = tau.plot, aes(x = Tau, y = V2))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  xlab(expression(paste("Residence time (", tau, ", h)")))+
  ylab(paste("PCoA 2 (", explainvar2, "%)", sep = ""))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(PCoA2_gam)$dev, 2))))+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.5), "cm"))

PCoA2

ggsave("./output/RTLC_BC_PCoA2.pdf")
ggsave("./output/RTLC_BC_PCoA2.png", width = 6.5, height = 5)
```

##Draw Figure
```{r}
ggdraw()+
  draw_plot(PCoA, x = 0, y = 0, width = 0.5, height = 1)+
  draw_plot(PCoA1, x = 0.5, y = 0.55, width = 0.5, height = 0.45)+
  draw_plot(PCoA2, x = 0.5, y = 0, width = 0.5, height = 0.55)+
  draw_plot_label(label = c("(a)", "(b)", "(c)"), size = 20, x = c(0,0.5,0.5), y = c(1.005,1.005,0.555))+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("./output/CommStr_PCoA.pdf")
ggsave("./output/CommStr_PCoA.png", width = 12, height = 6)
```



#Figure 4. Niche overlap

```{r}
#Generate OTU distance matrix using Spearman's correlation coefficient
OTUs.70 <- as.data.frame(cbind(Tau_20$Tau, OTUsREL_20[,colnames(OTUsREL_20) %in% colnames(as.data.frame(OTUsPA_20) %>% select_if(colSums(OTUsPA_20) >= 25))]))[, -1]

#Needed this for grep on fasta file for rep seq for Mothur
#lapply(colnames(OTUs.70), write, "niche_OTUs.txt", append = TRUE)

OTUs.70 <- OTUs.70[order(as.numeric(row.names(OTUs.70))),]

OTUs.70_heat <- as.data.frame(colnames(OTUs.70))
colnames(OTUs.70_heat) <- c("OTU")

for (otu1 in OTUs.70_heat$OTU){
  corr <- c()
  for(otu2 in OTUs.70_heat$OTU){
    corr <- c(corr, cor(as.numeric(OTUs.70[,paste0(otu1)]), as.numeric(OTUs.70[,paste0(otu2)]), method = "pearson"))
  }
  OTUs.70_heat[,paste0(otu1)] <- corr
  print(otu1)
}

OTUs.70_dist <- ((1-OTUs.70_heat[,2:2205])/2)
rownames(OTUs.70_dist) <- colnames(OTUs.70_heat[,2:2205])
OTUs.70_dist <- as.matrix(OTUs.70_dist)
OTUs.70_dist <- as.dist(OTUs.70_dist, upper = TRUE, diag = TRUE)


#Perform clustering with 6 clusters
set.seed(47401)
kmeans.clust <- kmeans(OTUs.70_dist,centers=5)


OTU.niches <- data.frame()
OTU.niche.sum <- as.data.frame(as.numeric(rownames(OTUs.70)))
n.cluster <- 1
while(n.cluster <= 5){
  cond <- sapply(kmeans.clust$cluster, function(x) x ==n.cluster)
  cluster <- names(kmeans.clust$cluster[cond])
  OTUs.cluster <- cbind(as.numeric(rownames(OTUs.70)), rep(n.cluster, 35))
  colnames(OTUs.cluster) <- c("Tau", "Cluster")
  OTUs.cluster <- cbind(OTUs.cluster, OTUs.70[ ,colnames(OTUs.70) %in% cluster])
  OTU.niche.sum <- cbind(OTU.niche.sum, rowSums(OTUs.70[ ,colnames(OTUs.70) %in% cluster]))
  OTUs.cluster <- as.data.frame(OTUs.cluster) %>% gather(OTU, REL, cluster[1]:cluster[length(cluster)])
  colnames(OTUs.cluster) <- c("Tau", "Cluster", "OTU", "REL")
  OTU.niches <- rbind(OTU.niches, OTUs.cluster)
  n.cluster <- n.cluster + 1
}

colnames(OTU.niche.sum) <- c("Tau", 1:(n.cluster-1))
OTU.niche.sum <- as.data.frame(OTU.niche.sum) %>% gather(Cluster, REL, "1":paste(n.cluster-1))

OTU.niche.sum$Cluster <- as.numeric(OTU.niche.sum$Cluster)
OTU.niche.sum$Cluster <- as.factor(OTU.niche.sum$Cluster)
```


```{r}
OTUs.70 <- as.data.frame(cbind(Tau_20$Tau, OTUsREL_20[,colnames(OTUsREL_20) %in% colnames(as.data.frame(OTUsPA_20) %>% select_if(colSums(OTUsPA_20) >= 25))]))[, -1]

#initial.overlap <- mean(niche.overlap(as.data.frame(OTUsREL_20), method = "pianka"))
#niche overlap of all OTUsREL_20 = 0.120262184194187

initial.overlap <- mean(niche.overlap(as.data.frame(OTUs.70), method = "pianka"))

uniform.niche.overlap <- read.csv("./data/uniform_nicheoverlap.csv", header = TRUE)[,-1]

uniform.fixedzero.niche.overlap <- read.csv("./data/uniform_fixedzeros_nicheoverlap.csv", header = TRUE)[,-1]

uniform.niche.overlap <- gather(uniform.niche.overlap, key = "Subset", value = "Pianka", L1.niche.overlap:A.niche.overlap)
uniform.fixedzero.niche.overlap <- gather(uniform.fixedzero.niche.overlap, key = "Subset", value = "Pianka", L1.niche.overlap:A.niche.overlap)

cohensD(x = initial.overlap, y = subset(uniform.fixedzero.niche.overlap, uniform.fixedzero.niche.overlap$Subset == "A.niche.overlap")$Pianka)

cohensD(x = initial.overlap, y = subset(uniform.niche.overlap, uniform.niche.overlap$Subset == "A.niche.overlap")$Pianka)

niche.overlap.uniform.long <- as.data.frame(subset(uniform.niche.overlap, uniform.niche.overlap$Subset == "A.niche.overlap")$Pianka)
niche.overlap.uniform.long <- cbind(niche.overlap.uniform.long, subset(uniform.fixedzero.niche.overlap, uniform.fixedzero.niche.overlap$Subset == "A.niche.overlap")$Pianka)
colnames(niche.overlap.uniform.long) <- c("uniform", "uniform.fixedzero")
niche.overlap.uniform.long <- gather(niche.overlap.uniform.long, key = "Null_model", value = "Pianka")


niche_uniform_nullmodels <- ggplot()+
  geom_hline(yintercept = initial.overlap, linetype = "dotted",color = "#008d98", cex = 2)+
  annotate("rect", xmin = 1.25, xmax = 1.75, ymin = initial.overlap -0.01, ymax = initial.overlap + 0.01, fill = "white")+
  geom_violin(data = niche.overlap.uniform.long, aes(x = Null_model, y = Pianka), fill = "#95caa6", color = "white", cex = 1.5)+
  geom_jitter(data = niche.overlap.uniform.long, width = 0.1, color = alpha("black", 0.1), aes(x = Null_model, y = Pianka), cex = 0.3)+
  ylab("Niche overlap")+
  scale_x_discrete(labels = c("Null model 1", "Null model 2"))+
#  scale_x_discrete(labels = c("Uniform", "Uniform +\n Fixed zeros"))+
  theme(legend.position = "none", axis.title.x = element_blank())+
  annotate("text", x = 1.5, y = initial.overlap, label = "Observed", color = "#008d98")

niche_uniform_nullmodels

ggsave("./output/Niche_uniform_nullmodels.pdf")
ggsave("./output/Niche_uniform_nullmodels.png", width = 5, height = 5)
```


```{r}
niche.gams <- data.frame()
for(n in 1:5){
  niche.gams <- rbind(niche.gams, cbind(rep(n, 50), predict_gam(gam(REL ~ s(Tau), family = gaussian(link = "identity"), data = subset(OTU.niche.sum, OTU.niche.sum$Cluster == n), method = "REML"))))
  print(summary(gam(REL ~ s(Tau), family = gaussian(link = "identity"), data = subset(OTU.niche.sum, OTU.niche.sum$Cluster == n), method = "REML")))
  print(mean(summary(gam(REL ~ s(Tau), family = gaussian(link = "identity"), data = subset(OTU.niche.sum, OTU.niche.sum$Cluster == n), method = "REML"))$s.table[,4]))
}
colnames(niche.gams) <- c("Cluster", "Tau", "fit", "se.fit")

niche.gams$Cluster <- as.numeric(niche.gams$Cluster)
niche.gams$Cluster <- as.factor(niche.gams$Cluster)

niche.gams$umin <- (log(2)/(1/((10^niche.gams$Tau)*60)))
niche.gams$umin <- na_if(niche.gams$umin, niche.gams$Tau < 3)
scale.factor <- max(subset(niche.gams, niche.gams$Cluster %in% c(1, 4))$fit)/ max(niche.gams$umin)

tau.1 <- log10(1/(log(2)))
tau.100 <- log10(100/(log(2)))
tau.12000 <- log10(12000/(log(2)))
tau.876000 <- log10(876000/(log(2)))


OTU.niches_totalREL <- ggplot(data = subset(niche.gams, niche.gams$Cluster %in% c(1, 4)), aes(x = Tau, y = fit, group = Cluster))+
  geom_smooth_ci(cex = 2, ci_alpha= 0.2, color = "grey")+
  geom_line(aes(color = Cluster),linewidth = 2)+
  scale_y_continuous(name = "Relative abundance")+
  geom_point(data = subset(OTU.niche.sum, OTU.niche.sum$Cluster %in% c(1,4)), aes(x = Tau, y = REL, color = Cluster))+
  geom_vline(xintercept = c(tau.1, tau.100, tau.12000, tau.876000), color = alpha("black", 0.2), linetype = "13", linewidth = 1.5)+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(-0.3, 6.95), sec.axis = sec_axis(~log10(log(2)/(1/(10^.))), name = "Maximum generation time", breaks = c(0,2,log10(12000), log10(876000)), labels = c("1 h", "100 h", "500 d", "100 y")))+
  scale_color_manual(values = as.vector(c(moma.colors("ustwo", 5, return_hex = TRUE)[1],moma.colors("ustwo", 5, return_hex = TRUE)[3])))+
  xlab(expression(paste("Residence time (", tau, ", h)")))+
  theme(legend.position = "none")+
  geom_text(x = 6.72, y = 0.48, label = expression(paste("Long ", tau)))+
  geom_text(x = 6.72, y = 0.45, label = "niche")+
  geom_text(x = 6.72, y = 0.12, label = expression(paste("Short ", tau)))+
  geom_text(x = 6.72, y = 0.09, label = "niche")

OTU.niches_totalREL

ggsave("./output/RTLC_niches_totalREL.pdf")
ggsave("./output/RTLC_niches_totalREL.png", width = 6, height = 4)
```


```{r}
niche.taxa <- data.frame()
for(n in 1:5){
  num <- nrow(subset(OTU.tax, OTU.tax$OTU %in% names(kmeans.clust$cluster[sapply(kmeans.clust$cluster, function(x) x ==n)])))
  niche.taxa <- rbind(niche.taxa, cbind(rep(n, num), rep(1/num,num), subset(OTU.tax, OTU.tax$OTU %in% names(kmeans.clust$cluster[sapply(kmeans.clust$cluster, function(x) x ==n)]))))
}
colnames(niche.taxa) <- c("Niche", "Prop_num", "OTU", "Domain", "Phylum", "Class", "Order", "Family", "Genus")
niche.taxa$Niche <- as.factor(niche.taxa$Niche)
niche.taxa$Prop_num <- as.numeric(niche.taxa$Prop_num)

for(x in 1:5){
  print(paste("Niche: ", x, "   N:", nrow(subset(niche.taxa, niche.taxa$Niche == x)), "   %:", 
round(nrow(subset(niche.taxa, niche.taxa$Niche == x))/nrow(niche.taxa), digits = 4)))
}


OTU.niches$Cluster <- as.numeric(OTU.niches$Cluster)
OTU.niches$Cluster <- as.factor(OTU.niches$Cluster)

OTUs_plot <- ggplot(subset(OTU.niches, OTU.niches$Cluster %in% c(1, 4)), aes(x = Tau, y = REL, group = Cluster, color = Cluster))+
#  geom_point()+
  geom_smooth(method = "loess")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab("Relative abundance")+
  xlab(expression(paste(tau, " (h)")))+
  scale_color_manual(values = as.vector(moma.colors("ustwo", 3, return_hex = TRUE))[1:2])+
  theme(legend.position = "none")

OTUs_plot

ggsave("./output/RTLC_niches_avgREL.pdf")
ggsave("./output/RTLC_niches_avgREL.png", width = 6, height = 4)
```

#Draw figure
```{r}
ggdraw()+
  draw_plot(niche_uniform_nullmodels, x = 0, y = 0.1, width = 0.42, height = 0.9) + 
  draw_plot(OTU.niches_totalREL, x = 0.42, y = 0, width = 0.58, height = 1) +
  draw_plot_label(label = c("(a)", "(b)"), size = 20, x = c(0,0.42), y = c(1,1))+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("./output/CommStr_niches.pdf")
ggsave("./output/CommStr_niches.png", width = 10, height = 5)
```


```{r}
niche1.otus <- subset(niche.taxa, niche.taxa$Niche == 1)$OTU
niche4.otus <- subset(niche.taxa, niche.taxa$Niche == 4)$OTU
niche3.otus <- subset(niche.taxa, niche.taxa$Niche == 3)$OTU
niche2.otus <- subset(niche.taxa, niche.taxa$Niche == 2)$OTU
niche5.otus <- subset(niche.taxa, niche.taxa$Niche == 5)$OTU

niche_dist <- read.csv("./data/niche_oturep.phylip.dist.csv", header = FALSE)[,-1]
tip_otus <- read.csv("./seq_to_otu.csv", header = FALSE)
colnames(tip_otus) <- c("Tips", "Otus")
rownames(niche_dist) <- tip_otus$Otus
colnames(niche_dist) <- tip_otus$Otus
niche_dist[is.na(niche_dist)] <- 0
niche_dist <- as.matrix(niche_dist)
niche_dist[upper.tri(niche_dist)] <- NA
df <- reshape2::melt(niche_dist, na.rm = TRUE)
df <- df %>% dplyr::filter(!(value == 0)) %>%
  dplyr::rename(otu1 = Var1, otu2 = Var2, distance = value)

df.sub <- cbind(df[which(df$otu1 %in% niche1.otus & df$otu2 %in% niche1.otus),], niche = rep("high tau niche", nrow(df[which(df$otu1 %in% niche1.otus & df$otu2 %in% niche1.otus),])))
df.sub <- rbind(df.sub, cbind(df[which(df$otu1 %in% niche4.otus & df$otu2 %in% niche4.otus),], niche = rep("low tau niche", nrow(df[which(df$otu1 %in% niche4.otus & df$otu2 %in% niche4.otus),]))))
df.sub <- rbind(df.sub, cbind(df[which(df$otu1 %in% niche1.otus & df$otu2 %in% niche4.otus),], niche = rep("high-low", nrow(df[which(df$otu1 %in% niche1.otus & df$otu2 %in% niche4.otus),]))))
df.sub <- rbind(df.sub, cbind(df[which(df$otu1 %in% niche4.otus & df$otu2 %in% niche1.otus),], niche = rep("high-low", nrow(df[which(df$otu1 %in% niche4.otus & df$otu2 %in% niche1.otus),]))))



# n.1.4.dist <- mean(c(df[which(df$otu1 %in% niche1.otus & df$otu2 %in% niche4.otus),]$distance, df[which(df$otu1 %in% niche4.otus & df$otu2 %in% niche1.otus),]$distance))
# 
# n.1.dist <- mean(df[which(df$otu1 %in% niche1.otus & df$otu2 %in% niche1.otus),]$distance)
# 
# n.4.dist <- mean(df[which(df$otu1 %in% niche4.otus & df$otu2 %in% niche4.otus),]$distance)
# 
# columns <- c("N", "N1.N4.dist", "N1.dist","N2.dist","N3.dist","N4.dist", "N5.dist")
# niche.rand.dist <- data.frame(matrix(nrow = 0, ncol = length(columns)))
# colnames(niche.rand.dist) <- columns
# niche.rand <- niche.taxa[,c("Niche","OTU")]
# x <- 0
# while(x <= 1000){
#   niche.rand$Niche <- sample(niche.rand$Niche)
# 
#   niche1.otus <- subset(niche.rand, niche.rand$Niche == 1)$OTU
#   niche2.otus <- subset(niche.rand, niche.rand$Niche == 2)$OTU
#   niche3.otus <- subset(niche.rand, niche.rand$Niche == 3)$OTU
#   niche4.otus <- subset(niche.rand, niche.rand$Niche == 4)$OTU
#   niche5.otus <- subset(niche.rand, niche.rand$Niche == 5)$OTU
#   
#   n.1.4.dist <- mean(c(df[which(df$otu1 %in% niche1.otus & df$otu2 %in% niche4.otus),]$distance, df[which(df$otu1 %in% niche4.otus & df$otu2 %in% niche1.otus),]$distance))
#   n.1.dist <- mean(df[which(df$otu1 %in% niche1.otus & df$otu2 %in% niche1.otus),]$distance)
#   n.2.dist <- mean(df[which(df$otu1 %in% niche2.otus & df$otu2 %in% niche2.otus),]$distance)
#   n.3.dist <- mean(df[which(df$otu1 %in% niche3.otus & df$otu2 %in% niche3.otus),]$distance)
#   n.4.dist <- mean(df[which(df$otu1 %in% niche4.otus & df$otu2 %in% niche4.otus),]$distance)
#   n.5.dist <- mean(df[which(df$otu1 %in% niche5.otus & df$otu2 %in% niche5.otus),]$distance)
#   
#   print(c(x))
#   
#   niche.rand.dist <- rbind(niche.rand.dist, c(x, n.1.4.dist, n.1.dist, n.2.dist, n.3.dist, n.4.dist, n.5.dist))
#   
#   x <- x + 1
# }
# 
# colnames(niche.rand.dist) <- columns
# 

# n.1.4.dist <- mean(c(df[which(df$otu1 %in% niche1.otus & df$otu2 %in% niche4.otus),]$distance, df[which(df$otu1 %in% niche4.otus & df$otu2 %in% niche1.otus),]$distance))
# n.1.dist <- mean(df[which(df$otu1 %in% niche1.otus & df$otu2 %in% niche1.otus),]$distance)
# n.4.dist <- mean(df[which(df$otu1 %in% niche4.otus & df$otu2 %in% niche4.otus),]$distance)
# 
# N.1.4 <- ggplot(niche.rand.dist, aes(x = N1.N4.dist))+
#   geom_histogram()+
#   geom_vline(xintercept = n.1.4.dist, color = "red")+
#   theme(axis.title.x = element_blank())+
#   ylab("Between low and \nhigh tau niche")
# 
# N.1.4
# 
# N.1 <- ggplot(niche.rand.dist, aes(x = N1.dist))+
#   geom_histogram()+
#   geom_vline(xintercept = n.1.dist, color = "red")+
#   theme(axis.title.x = element_blank())+
#   ylab("High tau niche")
# 
# N.1
# 
# N.4 <- ggplot(niche.rand.dist, aes(x = N4.dist))+
#   geom_histogram()+
#   geom_vline(xintercept = n.4.dist, color = "red")+
#   xlab("Mean phylogenetic distance between OTUs in niche")+
#   ylab("Low tau niche")
# 
# N.4
# 
# ggdraw()+
#   draw_plot(N.1.4, x = 0, y = 0.7, width = 1, height = 0.3)+
#   draw_plot(N.1, x = 0, y = 0.4, width = 1, height = 0.3)+
#   draw_plot(N.4, x = 0, y = 0, width = 1, height = 0.4)+
#   theme(plot.background = element_rect(fill="white", color = NA))
# 
# ggsave("./output/Niche_phylodist.pdf")
# ggsave("./output/Niche_phylodist.png", width = 6.5, height = 10)


niche_tree <- read.tree("./data/niche_oturep.phylip.tre") 
tip_otus <- read.csv("./seq_to_otu.csv", header = FALSE)
colnames(tip_otus) <- c("Tips", "Otus")
tip_otus$Tips <- factor(tip_otus$Tips, levels = unique(as.character(niche_tree$tip.label)))
tip_otus <- tip_otus[order(tip_otus$Tips),]
niche_tree$tip.label <- tip_otus$Otus

niche1 <- sapply(niche1.otus,grep,niche_tree$tip.label)
niche4 <- sapply(niche4.otus,grep,niche_tree$tip.label)
niche3 <- sapply(niche3.otus,grep,niche_tree$tip.label)
niche2 <- sapply(niche2.otus,grep,niche_tree$tip.label)
niche5 <- sapply(niche5.otus,grep,niche_tree$tip.label)

niche_tree$edge.length[1] <- niche_tree$edge.length[296]




niche.list <- as.data.frame(niche.taxa[,"Niche"])
rownames(niche.list) <- niche.taxa$OTU
colnames(niche.list) <- c("Niche")
niche.list$Niche <- as.numeric(niche.list$Niche)
niche.list <- as.data.frame((niche.list == 4)*1)
niche.list$name <- rownames(niche.list)

#install.packages('caper')
library('caper')

#library("adephylo")

niche.list <- as.data.frame(niche.taxa[,"Niche"])
rownames(niche.list) <- niche.taxa$OTU
colnames(niche.list) <- c("Niche")
niche.list$Niche <- as.numeric(niche.list$Niche)
niche.list <- as.data.frame((niche.list == 4)*1)
niche.list$name <- rownames(niche.list)

rtlc.niche <- comparative.data(niche_tree, niche.list, "name")
D.otu.4 <- phylo.d(rtlc.niche, binvar = Niche)

niche.list <- as.data.frame(niche.taxa[,"Niche"])
rownames(niche.list) <- niche.taxa$OTU
colnames(niche.list) <- c("Niche")
niche.list$Niche <- as.numeric(niche.list$Niche)
niche.list <- as.data.frame((niche.list == 1)*1)
niche.list$name <- rownames(niche.list)

rtlc.niche <- comparative.data(niche_tree, niche.list, "name")
D.otu.1 <- phylo.d(rtlc.niche, binvar = Niche)

print(D.otu.4)
print(D.otu.1)
plot(D.otu.4)
plot(D.otu.1)


niche.1.D <- D.otu.1$Permutations[["random"]]
cohensD(x = as.numeric(D.otu.1$Parameters[["Observed"]]), y = test)


niche.4.D <- D.otu.4$Permutations[["random"]]
cohensD(x = as.numeric(D.otu.4$Parameters[["Observed"]]), y = test)

test <- subset(niche.taxa, niche.taxa$Niche == 4)
test <- subset(test, !(test$Order %in% intersect(unique(subset(niche.taxa, niche.taxa$Niche == 1)$Order),  unique(subset(niche.taxa, niche.taxa$Niche == 4)$Order))))

test <- subset(niche.taxa, niche.taxa$Niche == 1)
test <- subset(test, !(test$Order %in% intersect(unique(subset(niche.taxa, niche.taxa$Niche == 1)$Order),  unique(subset(niche.taxa, niche.taxa$Niche == 4)$Order))))

test <- subset(niche.taxa, niche.taxa$Niche %in% c(1,4))

test <- subset(niche.taxa, niche.taxa$Niche %in% c(1,4) & niche.taxa$Phylum == "Planctomycetota" & niche.taxa$Order %in% intersect(unique(subset(niche.taxa, niche.taxa$Niche == 1)$Order),  unique(subset(niche.taxa, niche.taxa$Niche == 4)$Order)))

unique(subset(test, test$Class == "Betaproteobacteria")$Order)



test <- subset(niche.taxa, niche.taxa$Niche == 1)
test <- subset(test, !(test$Family %in% intersect(unique(subset(niche.taxa, niche.taxa$Niche == 1)$Family),  unique(subset(niche.taxa, niche.taxa$Niche == 4)$Family))))

test <- subset(niche.taxa, niche.taxa$Niche == 4)
test <- subset(test, !(test$Class %in% intersect(unique(subset(niche.taxa, niche.taxa$Niche == 1)$Class),  unique(subset(niche.taxa, niche.taxa$Niche == 4)$Class))))

niche.4d <- phylo4d(niche_tree, niche.list)
table.phylo4d(x = niche.4d, treetype = "phylo", symbol = "colors", col = c("red", "blue"), show.node = TRUE, cex.label = 0.5, edge.color = "black",
              edge.width = 2, box = FALSE, pch = 15, cex.symbol = 1.25, cex.legend = 1.5, show.tip.label = FALSE)


pdf("./output/Niche_phylotree.pdf")
plotTree(niche_tree, type = "fan", ftype="off", node.numbers = F, color = alpha("black", 0.25))
#add.arrow(niche_tree, tip=niche1,arrl=0.05, col = alpha(moma.colors("ustwo", 4)[2], 0.25))
add.arrow(niche_tree, tip = niche4, arrl = 0.025, col = alpha(moma.colors("ustwo", 4)[3], 00.75))
dev.off()
```


#Figure S3. Niche cluster # and full REL clusters
```{r}
#Determine number of clusters where within group sum of squares asymptotes
wss <- (nrow(OTUs.70_dist)-1)*sum(apply(OTUs.70_dist,2,var))
  for (i in 2:15) wss[i] <- sum(kmeans(OTUs.70_dist,
                                       centers=i)$withinss)
wss.df <- as.data.frame(1:15)
wss.df$wss <- wss
colnames(wss.df) <- c("clusters", "wss")

nclust <- ggplot(data = wss.df, aes(x = clusters, y = wss))+
  geom_point()+
  xlab("Number of Clusters")+
  ylab("Within groups sum of squares")+
  geom_smooth(method = "loess", color = "#008d98")

nclust

ggsave("./output/RTLC_niches_nclust.pdf")
ggsave("./output/RTLC_niches_nclust.png", width = 6, height = 4)

```

```{r}
OTU.niches_totalREL_all <- ggplot(data =niche.gams, aes(x = Tau, y = fit, group = Cluster))+
  geom_smooth_ci(cex = 2, ci_alpha= 0.2, color = "grey")+
  geom_line(linewidth = 2, aes(color = Cluster))+
  geom_point(data = OTU.niche.sum, aes(x = Tau, y = REL, color = Cluster))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab("Relative abundance")+
  scale_color_manual(values = as.vector(c(moma.colors("ustwo", 5)[1], moma.colors("ustwo", 5)[2], moma.colors("ustwo", 5)[4], moma.colors("ustwo", 5)[3], moma.colors("ustwo", 5)[5])))+
  xlab(expression(paste("Residence time (", tau, ", h)")))+
  labs(color = "Niche")+
  theme(legend.position = c(0.88, 0.5), axis.title.x = element_blank(), legend.text = element_text(size = 15), legend.title = element_text(size = 15))

OTU.niches_totalREL_all

ggsave("./output/RTLC_allniches_totalREL.pdf")
ggsave("./output/RTLC_allniches_totalREL.png", width = 6, height = 4)
```

```{r}
#ggplot(data = subset(OTU.niches, OTU.niches$Cluster %in% c(1,4)), aes(x = Tau, y = REL, group = OTU))+
  

OTU.niches$Cluster <- as.factor(OTU.niches$Cluster)
OTU.niches$Cluster <- ordered(OTU.niches$Cluster, levels = c("4", "1", "2", "3", "5"))
  
OTU.niche.ind <- ggplot(data = subset(OTU.niches, OTU.niches$Cluster %in% c(1,4)), aes(x = Tau, y = REL, group = OTU, color = Cluster))+
  geom_smooth(se = FALSE)+
  scale_color_discrete(label= c(expression(paste("Short ", tau, " niche")), expression(paste("Long ", tau, " niche"))), type = c(moma.colors("ustwo", 5)[3], moma.colors("ustwo", 5)[1]))+
  xlab(expression(paste("Residence time (", tau, ", h)")))+
  ylab("Relative abundance")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  theme(legend.title = element_blank(), legend.position = c(0.78, 0.8), legend.text = element_text(size = 15))
  
OTU.niche.ind


```

```{r}
ggdraw()+
  draw_plot(nclust, x = 0, y = 0.675, width = 1, height = 0.325) +
  draw_plot(OTU.niches_totalREL_all, x = 0, y = 0.35, width = 1, height = 0.325)+
  draw_plot(OTU.niche.ind, x = 0, y = 0, width = 1, height = 0.35)+
  draw_plot_label(label = c("(a)", "(b)", "(c)"), size = 20, x = c(0,0,0), y = c(1,0.675,0.35))+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("./output/CommStr_niches_supp.pdf")
ggsave("./output/CommStr_niches_supp.png", width = 10, height = 15)
```

##Niche area - NOT USING
```{r}
OTU.niches_area <- ggplot(data = OTU.niche.sum, aes(x = Tau, y = REL, group = Cluster, fill = Cluster))+
  geom_area()+
  ylab("Niche cluster total \nrelative abundance")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_fill_manual(values = as.list(moma.colors("ustwo", 7, return_hex = TRUE))[1:6])+
  xlab(expression(paste(tau, " (h)")))+
  theme(legend.position = "bottom")+
  guides(fill = guide_legend(nrow = 1))

OTU.niches_area

ggsave("./output/RTLC_niches_area.pdf")
ggsave("./output/RTLC_niches_area.png", width = 6, height = 4)
```

##Pianka sliding window - NOT USING
```{r}
OTUs.70 <- as.data.frame(cbind(Tau_20$Tau, OTUsREL_20[,colnames(OTUsREL_20) %in% colnames(as.data.frame(OTUsPA_20) %>% select_if(colSums(OTUsPA_20) >= 25))]))[, -1]

#initial.overlap <- mean(niche.overlap(as.data.frame(OTUsREL_20), method = "pianka"))
#niche overlap of all OTUsREL_20 = 0.120262184194187

initial.overlap <- mean(niche.overlap(as.data.frame(OTUs.70), method = "pianka"))



cohensD(x = initial.overlap, y = subset(uniform.fixedzero.niche.overlap, uniform.fixedzero.niche.overlap$Subset == "A.niche.overlap")$Pianka)

cohensD(x = initial.overlap, y = subset(uniform.niche.overlap, uniform.niche.overlap$Subset == "A.niche.overlap")$Pianka)




# OTUs.70 <- as.data.frame(cbind(Tau_20$Tau, OTUsREL_20[,colnames(OTUsREL_20) %in% colnames(as.data.frame(OTUsPA_20) %>% select_if(colSums(OTUsPA_20) >= 25))]))[, -1]
# 
# Pianka_turnover_three <- c()
# Taus.ordered <- sort(as.numeric(rownames(OTUs.70)))
# 
# x <- 2
# while(x <= length(Taus.ordered)-1){
#   subset.OTUs.70 <- subset(OTUs.70, as.numeric(rownames(OTUs.70)) >= Taus.ordered[x-1] & as.numeric(rownames(OTUs.70)) <= Taus.ordered[x+1])
#   niches <- niche.overlap(subset.OTUs.70, method = "pianka")
#   niches[is.na(niches)] <- 0
#   taus.three <- mean(niches)
#   Pianka_turnover_three <- c(Pianka_turnover_three, taus.three)
#   print(paste(Taus.ordered[x], " : ", taus.three))
#   x <- x + 1
# }
# 
# window.pianka <- as.data.frame(Taus.ordered[2:34])
# window.pianka$Pianka.three <- c(Pianka_turnover_three)
# 
# window.pianka.long <- gather(window.pianka, key = "Side", value = "Pianka", Pianka.three)
# 
# write.csv(window.pianka.long, file = "./data/window.pianka.long.three.csv")



window.pianka.long <- read.csv("./data/window.pianka.long.three.csv", header = TRUE)[,-1]

colnames(window.pianka.long) <- c("Tau", "Side", "Pianka")

window_niche_gam <- gam(Pianka ~ s(Tau), family = gaussian(link = "identity"), data = window.pianka.long, method = "REML")
summary(window_niche_gam)
mean(summary(window_niche_gam)$s.table[,4])
AIC(window_niche_gam)

window.niche.gam <- predict_gam(window_niche_gam) %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "#008d98", cex = 2, ci_alpha= 0.2)+
  geom_point(data = window.pianka.long, aes(x = Tau, y = Pianka))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  xlab(expression(paste(tau, " (h)")))+
  ylab("Niche overlap")+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(window_niche_gam)$dev, 2))))+
  theme(axis.title.x = element_blank())

window.niche.gam

ggsave("./output/RTLC_window.niche.gam.pdf")
ggsave("./output/RTLC_window.niche.gam.png", width = 6.5, height = 5)

```


#Manual vs. pump for community comp - NOT USING
```{r}
#Manual vs. pump doesn't matter
OTU_pump <- ggplot(tau.plot.order, aes(x = V1, y = V2))+
  geom_path(linewidth = 1.25, color = alpha("black", 0.2))+
  xlab(paste("PCoA 1 (", explainvar1, "%)", sep = ""))+
  ylab(paste("PCoA 2 (", explainvar2, "%)", sep = ""))+
  geom_point(cex = 3, aes(color = Pump))+
  labs(color = "Treatment\nmethod")+
  scale_color_manual(values = c("#008d98", "black"), labels = c("Pump", "Pipette"))+
  theme(legend.title = element_text(size = 14))

OTU_pump

ggsave("./output/RTLC_BC_OTUpump.pdf")
ggsave("./output/RTLC_BC_OTUpump.png", width = 8, height = 5)
```


#REL of phyla over tau - NOT USING
```{r}
# taxa <- as.data.frame(imp.otus$Tau)
# for(x in unique(OTU.tax.20$Phylum)){
#   list <- data.frame(rep(0, 35))
#   colnames(list) <- c(as.character(x))
#   for(y in OTU.tax.20$OTU){
#     if(subset(OTU.tax.20, OTU.tax.20$OTU == y)$Phylum == x){
#       list[,x] <- list[,x] + as.data.frame(OTUsr_20[,y])
#     }
#   }
#   taxa[,as.character(x)] <- list[,x]
# }
# rownames(taxa) <- taxa$`imp.otus$Tau`
# taxa.20 <- taxa[,-1]
# write.csv(taxa.20, "./data/taxa.20.csv")

taxa.20 <- read.csv("./data/taxa.20.csv", header = TRUE)
rownames(taxa.20) <- taxa.20$X
taxa.20 <- taxa.20[,-1]
taxa.20.rel <- decostand(taxa.20, method = "total")
taxa.20.rel <- cbind(imp.otus$Tau, taxa.20.rel)
colnames(taxa.20.rel) <- c("Tau", unique(OTU.tax.20$Phylum))
taxa.20.long <- gather(taxa.20.rel, phylum, REL, Proteobacteria:Modulibacteria)

phylum_plot <- ggplot(taxa.20.long, aes(x = Tau, y = REL, group = phylum))+
  geom_smooth(method = "loess", se = F, aes(color = phylum, group = phylum))+
  ylab("Relative abundance")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  labs(color = "Phylum")+
  xlab(expression(paste(tau, " (h)")))+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 15), axis.title = element_text(size = 20))+
  guides(color = guide_legend(ncol = 3))
phylum_plot

ggsave("./output/RTLC_rel_phylum.pdf")
ggsave("./output/RTLC_rel_phylum.png", width = 12, height = 8)
```


#Figure S6 - Individual carbon sources
```{r}
csource_sig <- data.frame()
for(c in unique(long_EP_48$C_Source)){
  type <- unique(subset(long_EP_48, long_EP_48$C_Source == c)$Type)
  csource.lm <- lm(data = subset(long_EP_48, long_EP_48$C_Source == c), OD_48 ~ Tau)
  csource.gam <- gam(data = subset(long_EP_48, long_EP_48$C_Source == c), OD_48 ~ s(Tau), family = gaussian(link="identity"), method = "REML")
  csource_sig <- rbind(csource_sig, c(c, type, summary(csource.lm)$coefficients[2,4], mean(summary(csource.gam)$s.table[,4])))
  assign(paste("lm.", gsub("-", "_", c), sep = ""), csource.lm)
  assign(paste("gam.", gsub("-", "_", c), sep = ""), csource.gam)
}
colnames(csource_sig) <- c("Resource", "Type", "P_lm", "P_gam")
csource_sig$BH_lm <- p.adjust(csource_sig$P_lm, "BH")
csource_sig$BH_gam <- p.adjust(csource_sig$P_gam, "BH")

csource_sig.gam <- subset(csource_sig, csource_sig$BH_gam < 0.05)

csource.gams <- data.frame()
for(c in as.list(csource_sig.gam$Resource)){
  csource.gams <- rbind(csource.gams, cbind(rep(c, 50), rep(subset(csource_sig.gam, csource_sig.gam$Resource == c)$Type, 50), predict_gam(gam(data = subset(long_EP_48, long_EP_48$C_Source == c), OD_48 ~ s(Tau), family = gaussian(link="identity"), method = "REML"))))
#  print(c)
#  print(summary(gam(data = subset(long_EP_48, long_EP_48$C_Source == c), OD_48 ~ s(Tau), family = gaussian(link="identity"), method = "REML")))
#  print(mean(summary(gam(REL ~ s(Tau), family = gaussian(link = "identity"), data = subset(OTU.niche.sum, OTU.niche.sum$Cluster == n), method = "REML"))$s.table[,4]))
}
colnames(csource.gams) <- c("C_Source", "Type", "Tau", "fit", "se.fit")

csource.gams$C_Source <- as.factor(csource.gams$C_Source)

#install.packages("lemon")
library(lemon)



Sig.csource.gams <- ggplot(data = csource.gams, aes(x = Tau, y = fit, group = C_Source))+
  geom_smooth_ci(cex = 2, ci_alpha= 0.2, color = "grey")+
  geom_line(cex = 2, aes(color = C_Source))+
  facet_wrap(~Type, scales = "free")+
  geom_point(data = subset(long_EP_48, long_EP_48$C_Source %in% csource_sig.gam$Resource), aes(x = Tau, y = OD_48, color = C_Source, group = C_Source))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab("Resource use at 48 hours (OD 590)")+
  scale_color_discrete(type = moma.colors("ustwo", 16, type = "continuous"))+
  xlab(expression(paste("Residence time (", tau, ", h)")))+
  guides(color = guide_legend(nrow = 6))+
  theme(legend.position = "bottom", legend.title = element_blank(), legend.box.background = element_rect(color = "black", size = 2))

Sig.csource.gams

gtable_show_names(Sig.csource.gams)

ggsave("./output/RTLC_Res_Csource_sig.pdf", plot = reposition_legend(Sig.csource.gams, "left", panel = "panel-3-2"))
ggsave("./output/RTLC_Res_Csource_sig.png", plot = reposition_legend(Sig.csource.gams, "left", panel = "panel-3-2"), width = 12, height = 10)

```


#Figure 5. Community Function - Resource Use
#Bray-Curtis distance Resource use
```{r}
EcoPlate_20 <- EcoPlate[rownames(OTUsr_20),]
EP.db.20 <- vegdist(EcoPlate_20, method = "bray")
EP.db <-vegdist(EcoPlate, method = "bray")
EP.pcoa <- cmdscale(EP.db, eig = TRUE)

EPREL <- EcoPlate
for(i in 1:nrow(EcoPlate)){
  EPREL[i,] = EcoPlate[i,]/sum(EcoPlate[i,])
}

EP.pcoa <- add.spec.scores(EP.pcoa, EPREL, method = "pcoa.scores")

explainvar1.ep <- round(EP.pcoa$eig[1] / sum(EP.pcoa$eig), 3) * 100
explainvar2.ep <- round(EP.pcoa$eig[2] / sum(EP.pcoa$eig), 3) * 100
explainvar3.ep <- round(EP.pcoa$eig[3] / sum(EP.pcoa$eig), 3) * 100

types <- read.csv("data/RTLC/EcoPlate/Csourcetypes.csv", header = TRUE, sep = ",")

EP.plot <- as.data.frame(EP.pcoa$points)
EP.plot <- cbind(EP.plot, Tau$Tau, Tau$Turn, Tau$Pump, Tau$Set)
colnames(EP.plot) <- c("V1", "V2", "Tau", "Turn", "Pump", "Set")
EP.plot$Turn <- factor(EP.plot$Turn, levels = c("TRUE", "FALSE"))
EP.plot$Pump <- factor(EP.plot$Pump, levels = c("TRUE", "FALSE"))

EP.cproj <- as.data.frame(EP.pcoa$cproj)
EP.cproj <- cbind(EP.cproj, as.data.frame(row.names(EP.pcoa$cproj)))
spe.corr.ep <- add.spec.scores(EP.pcoa, EPREL, method = "cor.scores")$cproj
spe.corr.ep <- as.data.frame(cbind(spe.corr.ep,types$Type))
corrcut.ep <- 0.7
imp.spp.ep <- as.data.frame(spe.corr.ep[abs(as.numeric(spe.corr.ep[,1])) >= corrcut.ep | abs(as.numeric(spe.corr.ep[,2])) >= corrcut.ep,])

Dim1.ep <- c()
Dim2.ep <- c()
for(x in unique(spe.corr.ep$V3)){
  Dim1.ep <- c(Dim1.ep, mean(as.numeric(subset(spe.corr.ep, spe.corr.ep$V3 == x)$Dim1)))
  Dim2.ep <- c(Dim2.ep, mean(as.numeric(subset(spe.corr.ep, spe.corr.ep$V3 == x)$Dim2)))
}

spe.corr.ep.means <- as.data.frame(unique(spe.corr.ep$V3))
spe.corr.ep.means <- cbind(spe.corr.ep.means, as.vector(Dim1.ep), as.vector(Dim2.ep))
colnames(spe.corr.ep.means) <- c("C_types", "Dim1", "Dim2")

fit <- envfit(EP.pcoa, EPREL, perm = 999)
fit

EP.plot$Tau <- as.numeric(EP.plot$Tau)
typeof(EP.plot$Tau)

spe.corr.ep<- spe.corr.ep[spe.corr.ep$C_source %in% c(gsub("-", ".", csource_sig.gam$Resource), "X4.Hydroxy.Benzoic.Acid"),]
spe.corr.ep$Dim1 <- as.numeric(spe.corr.ep$Dim1)
spe.corr.ep$Dim2 <- as.numeric(spe.corr.ep$Dim2)
spe.corr.ep$C_source <- rownames(spe.corr.ep)

spe.corr.ep.means[6, "Dim1"]


EP_BC <- ggplot(EP.plot, aes(x = V1, y = V2))+
  geom_point(shape = 21, cex = 2, aes(fill = as.numeric(Tau)), color = "black")+
  xlab(paste("PCoA 1 (", explainvar1.ep, "%)", sep = ""))+
  ylab(paste("PCoA 2 (", explainvar2.ep, "%)", sep = ""))+
#  xlim(c(-0.5, 0.3))+
  # labs(color = "Treatment\nmethod")+
  # scale_color_manual(values = c("#008d98", "black"), labels = c("Pump", "Manual"))+
  labs(fill = expression(paste(tau,  " (h)")))+
  scale_fill_gradientn(colors = alpha(moma.colors("ustwo"),0.9), labels = label_math(expr = 10^.x, format = force))+
  theme(legend.title = element_text(size = 14))+
  geom_text(size = 5, data = subset(spe.corr.ep.means, spe.corr.ep.means$C_types != "amine"), aes(x = Dim1, y = Dim2, label = C_types), color = "black")+
  geom_text(size =5, data = subset(spe.corr.ep.means, spe.corr.ep.means$C_types == "amine"), aes(x = Dim1, y = Dim2+0.20, label = C_types), color = "black")+
    geom_segment(aes(x = as.numeric(spe.corr.ep.means[6, "Dim1"]), y = as.numeric(spe.corr.ep.means[6, "Dim2"]), xend = as.numeric(spe.corr.ep.means[6, "Dim1"]), yend = as.numeric(spe.corr.ep.means[6, "Dim2"])+0.15))+
  theme(legend.title = element_text(size = 15), legend.position = c(0.2, 0.3), legend.text = element_text(size = 15))

EP_BC

ggsave("./output/RTLC_BC_Res.pdf")
ggsave("./output/RTLC_BC_Res.png", width = 8, height = 5)

dip.x <- c(rnorm(100, 10, 2), rnorm(100, 20, 2))

hist(dip.x)
hist(EP.plot$V2)

dip.test(EP.plot$V2)

EcoPlate_env$Set <- c(rep(1, 13), rep(2,12), rep(3,10), rep(4,14))

EP.dist <- vegdist(EcoPlate, method = "bray")
Env.dist <- vegdist(scale(EcoPlate_env[,-2]), method = "euclid")

mantel(EP.dist, Env.dist)

Res_PCoA1_lm <- lm(V1 ~ Tau, data = EP.plot)
Res_PCoA1_gam <- gam(data = EP.plot, V1 ~ s(Tau), family = gaussian(link = "identity"), method = "REML")
Res_PCoA1_gam_re <- gam(data = EP.plot, V1 ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), method = "REML")

anova(Res_PCoA1_gam, Res_PCoA1_gam_re, test = "Chisq")
summary(Res_PCoA1_gam)
mean(summary(Res_PCoA1_gam)$s.table[,4])

AIC(Res_PCoA1_lm, Res_PCoA1_gam, Res_PCoA1_gam_re)


Res_PCoA1 <- predict_gam(Res_PCoA1_gam) %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "#008d98", cex = 2, ci_alpha= 0.2)+
#  geom_vline(xintercept = turnover, linetype = "dashed", color = "blue", size = 1)+
  geom_point(data = EP.plot, aes(x = Tau, y = V1))+
  ylab(paste("PCoA 1 (", explainvar1.ep, "%)\n", sep = ""))+
#  labs(color = "Number of turnovers")+
#  scale_color_manual(values = c(alpha("blue", 0.3), alpha("black", 0.3)), labels = c("> 1", "< 1"))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  theme(axis.title.y = element_text(margin = margin(t = 0, r = -6, b = 0, l = 12)))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(Res_PCoA1_gam)$dev, 2))))+
  theme(axis.title.x = element_blank())

Res_PCoA1

ggsave("./output/RTLC_Res_BC_PCoA1.pdf")
ggsave("./output/RTLC_Res_BC_PCoA1.png", width = 6.5, height = 5)

Res_PCoA2_gam <- gam(data = EP.plot, V2 ~ s(Tau), family = gaussian(link = "identity"), method = "REML")
Res_PCoA2_gam_re <- gam(data = EP.plot, V2 ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), method = "REML")
Res_PCoA2_lm <- lm(data = EP.plot, V2 ~ Tau)
anova(Res_PCoA2_gam, Res_PCoA2_gam_re, test = "Chisq")
summary(Res_PCoA2_gam_re)

mean(summary(Res_PCoA2_gam_re)$s.table[,4])

AIC(Res_PCoA2_gam, Res_PCoA2_gam_re, Res_PCoA2_lm)

Res_PCoA2 <- predict_gam(Res_PCoA2_gam_re, exclude_terms = "s(Set)") %>%
  filter(Set == "1") %>% 
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "#008d98", cex = 2, ci_alpha= 0.2)+
#  geom_vline(xintercept = turnover, linetype = "dashed", color = "blue", size = 1)+
  geom_point(data = EP.plot, aes(x = Tau, y = V2))+
  ylab(paste("PCoA 2 (", explainvar2.ep, "%)\n", sep = ""))+
  xlab(expression(paste("Residence time (",tau, ", h)")))+
#  labs(color = "Number of turnovers")+
#  scale_color_manual(values = c(alpha("blue", 0.3), alpha("black", 0.3)), labels = c("> 1", "< 1"))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  theme(axis.title.y = element_text(margin = margin(t = 0, r = -6, b = 0, l = 12)))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(Res_PCoA2_gam_re)$dev, 2))))

Res_PCoA2

ggsave("./output/RTLC_Res_BC_PCoA2.pdf")
ggsave("./output/RTLC_Res_BC_PCoA2.png", width = 6.5, height = 5)
```
```{r}
ggdraw()+
  draw_plot(EP_BC, x = 0, y = 0, width = 0.5, height = 1)+
  draw_plot(Res_PCoA1, x = 0.5, y = 0.55, width = 0.5, height = 0.45)+
  draw_plot(Res_PCoA2, x = 0.5, y = 0, width = 0.5, height = 0.55)+
  draw_plot_label(label = c("(a)", "(b)", "(c)"), size = 20, x = c(0,0.5,0.5), y = c(1.005,1.005,0.555))+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("./output/CommFun_PCoA.pdf")
ggsave("./output/CommFun_PCoA.png", width = 12, height = 6)
```

```{r}
Dim1_sites <- EP.plot$V1
Dim2_sites <- EP.plot$V2

EP_corr <- as.data.frame(matrix(NA, dim(EPREL)[2], 4))
row.names(EP_corr) <- colnames(EPREL)
colnames(EP_corr)  <- c("V1_rho", "V1_p.value", "V2_rho", "V2_p.value")

for (i in 1:dim(EPREL)[2]){
  out.i <- cor.test(EPREL[,i], EP.plot$V1, method="spearman",
                    exact=FALSE)
  EP_corr[i,1] <- out.i$estimate
  EP_corr[i,2] <- out.i$p.value
  
  out.i <- cor.test(EPREL[,i], EP.plot$V2, method="spearman",
                    exact=FALSE)
  EP_corr[i,3] <- out.i$estimate
  EP_corr[i,4] <- out.i$p.value
}

EP_corr     <- na.omit(EP_corr)
EP_corr$V1_BH <- p.adjust(EP_corr$V1_p.value, method = "BH")
EP_corr$V2_BH <- p.adjust(EP_corr$V2_p.value, method = "BH")
EP_corr_V2 <- EP_corr[abs(EP_corr$V2_rho) > 0.5, ]
EP_corr_V1 <- EP_corr[abs(EP_corr$V1_rho) > 0.5, ]
```


##Draw Figure
```{r}
ggdraw()+
  draw_plot(EP_BC, x = 0, y = 0, width = 1, height = 1) + 
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("./output/CommRes_PCoA.pdf")
ggsave("./output/CommRes_PCoA.png", width = 5, height = 5)

ggdraw()+
  draw_plot(Res_PCoA1, x = 0, y = 0.55, width = 1, height = 0.45) + 
  draw_plot(Res_PCoA2, x = 0, y = 0, width = 1, height = 0.55) + 
  draw_plot_label(label = c("A", "B"), size = 30, x = c(0.02,0.02), y = c(1,0.55))+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("./output/CommRes_PCoA_Axes.pdf")
ggsave("./output/CommRes_PCoA_Axes.png", width = 8, height = 10)
```

##FIGURE S1. Number of resources
```{r}
NumRes_gam <- gam(data = Tau, NumRes ~ s(Tau), family = gaussian(link = "identity"), method = "REML")
NR_lm <- lm(NumRes ~ Tau, data = Tau)
summary(NR_lm)
summary(NumRes_gam)


mean(summary(NumRes_gam)$s.table[,4])

NumRes <- ggplot(Tau, aes(x = Tau, y = NumRes))+
  geom_point(size = 2, alpha = 0.6)+
  xlab(expression(paste(tau, " (h)")))+
  ylab("Number of Resources Used \n at 48 hours")+
  geom_smooth(method = "lm", formula = y~x, color = "blue", fill = alpha("black", 0.4))+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),
               formula = y~x, parse = TRUE, size = 4)+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))

ggsave("./output/RTLC_NumRes.pdf")
ggsave("./output/RTLC_NumRes.png", width = 8, height = 5)
```







#Unused figures - IBMs
#Figure 7. IBMs

```{r}
IBM_IC <- read.csv("../residence-time/Model/results/data/SimData.csv", header = TRUE, sep = ",")
IBM_IC_long <- as.data.frame(unique(IBM_IC$sim))
colnames(IBM_IC_long) <- c("sim")
IBM_IC_long$ct <- rep(1000, 50)

for(x in unique(IBM_IC_long$sim)){
  IBM_IC_long[IBM_IC_long$sim == x, "V"] <- mean(subset(IBM_IC, IBM_IC$sim == x)$V)
  IBM_IC_long[IBM_IC_long$sim == x, "Q"] <- mean(subset(IBM_IC, IBM_IC$sim == x)$Q)
  IBM_IC_long[IBM_IC_long$sim == x, "N"] <- mean(subset(IBM_IC, IBM_IC$sim == x)$total.abundance)
  IBM_IC_long[IBM_IC_long$sim == x, "R"] <- mean(subset(IBM_IC, IBM_IC$sim == x)$resource.particles)
  IBM_IC_long[IBM_IC_long$sim == x, "P"] <- mean(subset(IBM_IC, IBM_IC$sim == x)$ind.production)
  IBM_IC_long[IBM_IC_long$sim == x, "Dorm"] <- mean(subset(IBM_IC, IBM_IC$sim == x)$dormant.total.abundance)
  IBM_IC_long[IBM_IC_long$sim == x, "S"] <- mean(subset(IBM_IC, IBM_IC$sim == x)$species.richness)
}

IBM_IC_long$N_turn <- IBM_IC_long$ct/(IBM_IC_long$V/IBM_IC_long$Q)
IBM_IC_long$turn <- IBM_IC_long$N_turn > 1
IBM_IC_long$ct_f <- as.factor(IBM_IC_long$ct)
IBM_IC_long$Tau <- log(IBM_IC_long$V/IBM_IC_long$Q, 10)
IBM_IC_long$DormPer <- (IBM_IC_long$Dorm/IBM_IC_long$N)*100

plot(IBM_IC_long$P ~ IBM_IC_long$Tau)
plot(IBM_IC_long$N ~ IBM_IC_long$Tau)
plot(IBM_IC_long$DormPer ~ IBM_IC_long$Tau)
```

##IBM - N
```{r}
N_IBM_gam <- gam(log(N, 10) ~ s(Tau), family = gaussian(link = "identity"), data = subset(IBM_IC_long, IBM_IC_long$N > 0), method = "REML")

summary(N_IBM_gam)

N_IBM <- predict_gam(N_IBM_gam) %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_point(data = subset(IBM_IC_long, IBM_IC_long$N > 0), aes(x = Tau, y = log(N, 10)))+
  ylab("N (number of individuals)")+
  theme(axis.title.x = element_blank())+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(N_IBM_gam)$dev, 2))))

N_IBM

ggsave("./output/IBM_N.pdf")
ggsave("./output/IBM_N.png", width = 6.5, height = 5)
```

#IBM - S
```{r}
S_IBM_gam <- gam(log(S, 10) ~ s(Tau), family = gaussian(link = "identity"), data = subset(IBM_IC_long, IBM_IC_long$S > 0), method = "REML")

summary(S_IBM_gam)

S_IBM <- predict_gam(S_IBM_gam) %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_point(data = subset(IBM_IC_long, IBM_IC_long$S > 0), aes(x = Tau, y = log(S, 10)))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(0, 3), breaks = c(0, 1, 2, 3))+
  ylab("Observed S")+
  theme(axis.title.x = element_blank())+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(S_IBM_gam)$dev, 2))))

S_IBM

ggsave("./output/IBM_S.pdf")
ggsave("./output/IBM_S.png", width = 6.5, height = 5)
```
#IBM - Dorm
```{r}
Dorm_IBM_gam <- gam(DormPer ~ s(Tau), family = gaussian(link = "identity"), data = IBM_IC_long, method = "REML")

summary(Dorm_IBM_gam)

Dorm_IBM <- predict_gam(Dorm_IBM_gam) %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_point(data = IBM_IC_long, aes(x = Tau, y = DormPer))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab("Dormant cells (% of total)")+
  xlab(expression(paste(tau, " (time steps)")))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(Dorm_IBM_gam)$dev, 2))))+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.4), "cm"))

Dorm_IBM

ggsave("./output/IBM_Dorm.pdf")
ggsave("./output/IBM_Dorm.png", width = 6.5, height = 5)
```
#IBM - P
```{r}
P_IBM_gam <- gam(log(P + 1, 10) ~ s(Tau), family = gaussian(link = "identity"), data = IBM_IC_long, method = "REML")

summary(P_IBM_gam)

P_IBM <- predict_gam(P_IBM_gam) %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.4)+
  geom_point(data = IBM_IC_long, aes(x = Tau, y = log(P + 1, 10)))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab("Productivity")+
  xlab(expression(paste(tau, " (time steps)")))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(P_IBM_gam)$dev, 2))))+
  theme(plot.margin = unit(c(1.5, 0.2, 0, 0.2), "cm"))

P_IBM

ggsave("./output/IBM_P.pdf")
ggsave("./output/IBM_P.png", width = 6.5, height = 5)
```

```{r}
ggdraw()+
  draw_plot(N_IBM, x = 0, y = 0.65, width = 1, height = 0.3) + 
  draw_plot(S_IBM, x = 0, y = 0.35, width = 1, height = 0.3) + 
  draw_plot(Dorm_IBM, x = 0, y = 0.0, width = 1, height = 0.35)+
  draw_plot_label(label = c("A", "B", "C"), size = 30, x = c(0.05,0.05,0.05), y = c(0.98,0.67,0.4))+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("./output/IBM_results.pdf")
ggsave("./output/IBM_results.png", width = 5, height = 10)

```

#% dormancy
```{r}
N_RSG <- read.csv("./data/RTLC/2021_RTLC_RSG_N.csv", header = TRUE)
N_RSG$DorPer <- (N_RSG$N-N_RSG$RSG)/N_RSG$N
N_RSG$Tau <- log(((10^N_RSG$Tau)/60), 10)

Dorm_gam <- gam(data = N_RSG, (DorPer * 100) ~ s(Tau), family = gaussian(link = "identity"), method = "REML")
Dorm_gam_re <- gam(data = N_RSG, (DorPer * 100) ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), method = "REML")
summary(Dorm_gam)
AIC(Dorm_gam, Dorm_gam_re)
plot(Dorm_gam)
k.check(Dorm_gam)
gam.check(Dorm_gam)

Dorm_Tau <- predict_gam(Dorm_gam) %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.4)+
  geom_point(data = N_RSG, aes(x = Tau, y = (DorPer * 100)))+
  xlab(expression(paste(tau, " (h)")))+
  ylab("% Dormancy")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(Dorm_gam)$dev, 2))))

Dorm_Tau


ggsave("./output/RTLC_Dorm.pdf")
ggsave("./output/RTLC_Dorm.png")
```

#Unused figures

```{r}
for(x in unique(imp.spp$Phylum)){
  list <- data.frame(rep(0, 35))
  colnames(list) <- c(as.character(x))
  for(y in imp.spp$OTU){
    if(imp.spp[y, "Phylum"] == x){
      list[,x] <- list[,x] + as.data.frame(imp.otus[,y])
    }
  }
  imp.otus[,as.character(x)] <- list[,x]
}

imp.phylum <- cbind(imp.otus$Tau, imp.otus[ ,338:357])
colnames(imp.phylum) <- c("Tau", colnames(imp.otus[,338:357]))
imp.phylum.long <- gather(imp.phylum, phylum, REL, Proteobacteria:Hydrogenedentes)

imp.phylum.long$Tau <- as.numeric(imp.phylum.long$Tau)

imp_phylum_plot <- ggplot(imp.phylum.long, aes(x = Tau, y = REL, group = phylum))+
  geom_smooth(method = "loess", se = F, aes(color = phylum))+
  ylab("Relative abundance")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  labs(color = "Phylum")+
  xlab(expression(paste(tau, " (h)")))+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 15), axis.title = element_text(size = 20))
imp_phylum_plot

ggsave("./output/RTLC_imp_phylum.pdf")
ggsave("./output/RTLC_imp_phylum.png", width = 12, height = 8)
```

##Biofilm Formation
```{r}

OT_N_re <- lmer(log(OT/N) ~ Tau + (1|Set), data = Tau)
OT_N <- lm(log(OT/N) ~ Tau, data = Tau)

summary(OT_N_re)
summary(OT_N)
AIC(OT_N, OT_N_re)
anova(OT_N_re, OT_N)

OT_Tau <- ggplot(Tau, aes(x = Tau, y = log(OT/N)))+
  geom_point(size = 2, alpha = 0.6)+
  xlab(expression(paste(tau, " (h)")))+
  ylab("log(Biofilm Formation/Cell) \n OD 595 nm")+
  geom_smooth(method = "lm")+
  labs(title = paste(" P =", Sig$`Pr(>F)`[1]))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),
               label.x = "right", label.y = "top", formula = y ~ x, parse = TRUE, size = 4)

OT_Tau

ggsave("./output/RTLC_OT.pdf")
ggsave("./output/RTLC_OT.png")
```


##Resource use vs. Time for all Tau
```{r}
Avg_Hr <- ggplot(subset(long_EP, long_EP$C_Source =="Avg"), aes(x = Hours, y = OD, group = Tau, color = Tau))+
  geom_line(size = 1)+
  geom_point(size = 3)+
  xlab("Hours")+
  ylab("Average Well Response (OD590)")+
  labs(group = "Tau")+
  scale_color_continuous(type = "viridis", labels = label_math(expr = 10^.x, format = force))+
  theme(legend.title = element_text(size = 16))
  
Avg_Hr

ggsave("./output/RTLC_AvgRes_Hr.pdf")
ggsave("./output/RTLC_AvgRes_Hr.png")

```

```{r}
Res_heat <- as.data.frame(colnames(EcoPlate))
colnames(Res_heat) <- c("Res")
Res_heat$N <- colSums(EcoPlate)
rownames(EcoPlate) <- c(EcoPlate_env$Tau)
 

for (res in Res_heat$Res){
  x = 0
  for(site in rownames(EcoPlate)){
    x = x + (EcoPlate[site, res] * as.numeric(site))
  }
  Res_heat$Tau[Res_heat$Res == res] <- x/Res_heat$N[Res_heat$Res == res]
}

colnames(Res_heat) <- c("Res", "N", "Weight")

Res_heat <- Res_heat[order(Res_heat$Weight),]

EcoPlate <- cbind(EcoPlate_env, EcoPlate)

colnames(EcoPlate)

#########
eco_long <- gather(EcoPlate, Res, N, X2.Hydroxy.Benzoic.Acid:Tween.80, factor_key = TRUE)
eco_long$Res <- factor(eco_long$Res, levels = rev(as.list(Res_heat$Res)))
print(levels(eco_long$Res))

eco_long$Tau <- as.factor(eco_long$Tau)

ecoplate_heat <- ggplot(eco_long, aes(Tau, Res, fill = N))+
  geom_tile()+
  theme(axis.ticks.x = element_blank(), axis.ticks.y = element_blank(), axis.text.x = element_blank(), axis.text.y = element_text(size = 10))+
  scale_fill_continuous(type = "viridis")

ecoplate_heat

ggsave("./output/res_heatmap.pdf")
ggsave("./output/res_heatmap.png")
```
#Unused figures - Niche overlap

```{r}

L1.niche_plot <- ggplot()+
  geom_histogram(data = all.niche.overlap, aes(x = L1.niche.overlap), binwidth = 0.000005)+
  ylab("Count (OTUs in >70% samples)")+
  geom_vline(xintercept = L1.initial.overlap)+
  xlab("Pianka's niche overlap index")+
  theme(axis.title.x = element_blank())

L1.niche_plot

M1.niche_plot <- ggplot()+
  geom_histogram(data = all.niche.overlap, aes(x = M1.niche.overlap), binwidth = 0.000005)+
  ylab("Count (OTUs in >70% samples)")+
  geom_vline(xintercept = M1.initial.overlap)+
  xlab("Pianka's niche overlap index")

M1.niche_plot

A.niche_plot <- ggplot()+
  geom_histogram(data = all.niche.overlap, aes(x = A.niche.overlap), binwidth = 0.000005)+
  ylab("Count (OTUs in >70% samples)")+
  geom_vline(xintercept = initial.overlap)+
#  xlim(c(0.52, 0.72))+
  xlab("Pianka's niche overlap index")+
  theme(axis.title.x = element_blank())

A.niche_plot

ggdraw()+
  draw_plot(A.niche_plot, x = 0, y = 0.65, width = 1, height = 0.3) + 
  draw_plot(L1.niche_plot, x = 0, y = 0.35, width = 1, height = 0.3) + 
  draw_plot(M1.niche_plot, x = 0, y = 0.0, width = 1, height = 0.35)+
  draw_plot_label(label = c("A", "B", "C"), size = 30, x = c(0.06,0.06,0.06), y = c(0.96,0.65,0.35))+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("./output/Niche_hists.pdf")
ggsave("./output/Niche_hists.png", width = 8, height = 10)
```

##Fixed-zeros
```{r}
# OTUs.70 <- as.data.frame(cbind(Tau_20$Tau, OTUsREL_20[,colnames(OTUsREL_20) %in% colnames(as.data.frame(OTUsPA_20) %>% select_if(colSums(OTUsPA_20) >= 25))]))[, -1]
# 
# L1.niche.overlap <- c()
# M1.niche.overlap <- c()
# A.niche.overlap <- c()
# 
# x <- 1
# while(x < 100){
#   for(col in 1:ncol(OTUs.70)){
#     n.zeros <- colSums(OTUs.70 == 0)[[col]] + 1
#     OTUs.70 <- OTUs.70[order(OTUs.70[,col]),]
#     OTUs.70[,col] <- c(rep(0, n.zeros-1), sample(OTUs.70[n.zeros:length(OTUs.70[,col]),][,col]))
#   }
#   L1.p <- mean(niche.overlap(subset(OTUs.70, as.numeric(rownames(OTUs.70)) > turnover), method = "pianka"))
#   M1.p <- mean(niche.overlap(subset(OTUs.70, as.numeric(rownames(OTUs.70)) < turnover), method = "pianka"))
#   A.p <- mean(niche.overlap(OTUs.70, method = "pianka"))
#   print(paste(x, " : ", L1.p, M1.p, A.p))
#   L1.niche.overlap <- c(L1.niche.overlap, L1.p)
#   M1.niche.overlap <- c(M1.niche.overlap, M1.p)
#   A.niche.overlap <- c(A.niche.overlap, A.p)
#   x = x + 1
# }
# 
# all.niche.overlap <- as.data.frame(L1.niche.overlap)
# all.niche.overlap$M1.niche.overlap <- M1.niche.overlap
# all.niche.overlap$A.niche.overlap <- A.niche.overlap
# 
# write.csv(all.niche.overlap, file = "./data/fixed_zeros_nicheoverlap.csv")


fixedzero.niche.overlap <- read.csv("./data/fixed_zeros_nicheoverlap.csv", header = TRUE)[,-1]

hline <- data.frame(cat = c(1,2,3), v=c(initial.overlap, L1.initial.overlap, M1.initial.overlap), col=c("A.niche.overlap", "L1.niche.overlap", "M1.niche.overlap"))


fixedzero.niche.overlap <- gather(fixedzero.niche.overlap, key = "Subset", value = "Pianka", L1.niche.overlap:A.niche.overlap)

niche_violin_fixedzero <- ggplot()+
  geom_segment(data=hline, aes(cat-0.4, xend=cat+0.4, y=v, yend=v, color = col), linetype = "dashed", cex = 2)+
  geom_violin(data = fixedzero.niche.overlap, aes(x = Subset, y = Pianka, color = Subset, fill = Subset), cex = 2)+
  geom_jitter(data = fixedzero.niche.overlap, aes(x = Subset, y = Pianka), cex = 0.1, alpha = 0.2)+
  ylab("Pianka's niche overlap index")+
  scale_x_discrete(labels = c("All sites", "<1 turnover", ">1 turnover"))+
  theme(legend.position = "none", axis.title.x = element_blank())

niche_violin_fixedzero

ggsave("./output/Niche_violin_fixedzero.pdf")
ggsave("./output/Niche_violin_fixedzero.png", width = 6, height = 8)
```

##Uniform Fixed-zeros
```{r}
# OTUs.70 <- as.data.frame(cbind(Tau_20$Tau, OTUsREL_20[,colnames(OTUsREL_20) %in% colnames(as.data.frame(OTUsPA_20) %>% select_if(colSums(OTUsPA_20) >= 25))]))[, -1]
# 
# L1.niche.overlap <- c()
# M1.niche.overlap <- c()
# A.niche.overlap <- c()
# 
# x <- 1
# while(x < 100){
#   for(col in 1:ncol(OTUs.70)){
#     n.zeros <- colSums(OTUs.70 == 0)[[col]] + 1
#     OTUs.70 <- OTUs.70[order(OTUs.70[,col]),]
#     OTUs.70[,col] <- c(rep(0, n.zeros-1), runif(36-n.zeros, min = 0, max = 1))
#   }
#   OTUs.70 <-decostand(OTUs.70, method = "total")
#   L1.p <- mean(niche.overlap(subset(OTUs.70, as.numeric(rownames(OTUs.70)) > turnover), method = "pianka"))
#   M1.p <- mean(niche.overlap(subset(OTUs.70, as.numeric(rownames(OTUs.70)) < turnover), method = "pianka"))
#   A.p <- mean(niche.overlap(OTUs.70, method = "pianka"))
#   print(paste(x, " : ", L1.p, M1.p, A.p))
#   L1.niche.overlap <- c(L1.niche.overlap, L1.p)
#   M1.niche.overlap <- c(M1.niche.overlap, M1.p)
#   A.niche.overlap <- c(A.niche.overlap, A.p)
#   x = x + 1
# }
# 
# all.niche.overlap <- as.data.frame(L1.niche.overlap)
# all.niche.overlap$M1.niche.overlap <- M1.niche.overlap
# all.niche.overlap$A.niche.overlap <- A.niche.overlap
# 
# write.csv(all.niche.overlap, file = "./data/uniform_fixedzeros_nicheoverlap.csv")


uniform.fixedzero.niche.overlap <- read.csv("./data/uniform_fixedzeros_nicheoverlap.csv", header = TRUE)[,-1]

hline <- data.frame(cat = c(1,2,3), v=c(initial.overlap, L1.initial.overlap, M1.initial.overlap), col=c("A.niche.overlap", "L1.niche.overlap", "M1.niche.overlap"))


uniform.fixedzero.niche.overlap <- gather(uniform.fixedzero.niche.overlap, key = "Subset", value = "Pianka", L1.niche.overlap:A.niche.overlap)

cohensD(x = initial.overlap, y = subset(uniform.fixedzero.niche.overlap, uniform.fixedzero.niche.overlap$Subset == "A.niche.overlap")$Pianka)

niche_violin_uniform.fixedzero <- ggplot()+
  geom_segment(data=hline, aes(cat-0.4, xend=cat+0.4, y=v, yend=v, color = col), linetype = "dashed", cex = 2)+
  geom_violin(data = uniform.fixedzero.niche.overlap, aes(x = Subset, y = Pianka, color = Subset, fill = Subset), cex = 2)+
  geom_jitter(data = uniform.fixedzero.niche.overlap, aes(x = Subset, y = Pianka), cex = 0.1, alpha = 0.2)+
  ylab("Pianka's niche overlap index")+
  scale_x_discrete(labels = c("All sites", "<1 turnover", ">1 turnover"))+
  theme(legend.position = "none", axis.title.x = element_blank())

niche_violin_uniform.fixedzero

ggsave("./output/Niche_violin_uniform_fixedzero.pdf")
ggsave("./output/Niche_violin_uniform_fixedzero.png", width = 6, height = 8)
```


##Uniform Scrambled
```{r}
# OTUs.70 <- as.data.frame(cbind(Tau_20$Tau, OTUsREL_20[,colnames(OTUsREL_20) %in% colnames(as.data.frame(OTUsPA_20) %>% select_if(colSums(OTUsPA_20) >= 25))]))[, -1]
# 
# L1.niche.overlap <- c()
# M1.niche.overlap <- c()
# A.niche.overlap <- c()
# 
# x <- 1
# while(x < 100){
#   for(col in 1:ncol(OTUs.70)){
#     OTUs.70[,col] <- runif(35, min = 0, max = 1)
#   }
#   OTUs.70 <-decostand(OTUs.70, method = "total")
#   L1.p <- mean(niche.overlap(subset(OTUs.70, as.numeric(rownames(OTUs.70)) > turnover), method = "pianka"))
#   M1.p <- mean(niche.overlap(subset(OTUs.70, as.numeric(rownames(OTUs.70)) < turnover), method = "pianka"))
#   A.p <- mean(niche.overlap(OTUs.70, method = "pianka"))
#   print(paste(x, " : ", L1.p, M1.p, A.p))
#   L1.niche.overlap <- c(L1.niche.overlap, L1.p)
#   M1.niche.overlap <- c(M1.niche.overlap, M1.p)
#   A.niche.overlap <- c(A.niche.overlap, A.p)
#   x = x + 1
# }
# 
# all.niche.overlap <- as.data.frame(L1.niche.overlap)
# all.niche.overlap$M1.niche.overlap <- M1.niche.overlap
# all.niche.overlap$A.niche.overlap <- A.niche.overlap
# 
# write.csv(all.niche.overlap, file = "./data/uniform_nicheoverlap.csv")


uniform.niche.overlap <- read.csv("./data/uniform_nicheoverlap.csv", header = TRUE)[,-1]

hline <- data.frame(cat = c(1,2,3), v=c(initial.overlap, L1.initial.overlap, M1.initial.overlap), col=c("A.niche.overlap", "L1.niche.overlap", "M1.niche.overlap"))


uniform.niche.overlap <- gather(uniform.niche.overlap, key = "Subset", value = "Pianka", L1.niche.overlap:A.niche.overlap)

cohensD(x = initial.overlap, y = subset(uniform.niche.overlap, uniform.niche.overlap$Subset == "A.niche.overlap")$Pianka)


niche_violin_uniform <- ggplot()+
  geom_segment(data=hline, aes(cat-0.4, xend=cat+0.4, y=v, yend=v, color = col), linetype = "dashed", cex = 2)+
  geom_violin(data = uniform.niche.overlap, aes(x = Subset, y = Pianka, color = Subset, fill = Subset), cex = 2)+
  geom_jitter(data = uniform.niche.overlap, aes(x = Subset, y = Pianka), cex = 0.1, alpha = 0.2)+
  ylab("Pianka's niche overlap index")+
  scale_x_discrete(labels = c("All sites", "<1 turnover", ">1 turnover"))+
  theme(legend.position = "none", axis.title.x = element_blank())

niche_violin_uniform

ggsave("./output/Niche_violin_uniform.pdf")
ggsave("./output/Niche_violin_uniform.png", width = 6, height = 8)
```

```{r}
OTUs.70 <- as.data.frame(cbind(Tau_20$Tau, OTUsREL_20[,colnames(OTUsREL_20) %in% colnames(as.data.frame(OTUsPA_20) %>% select_if(colSums(OTUsPA_20) >= 25))]))[, -1]

# L1.niche.overlap <- c()
# M1.niche.overlap <- c()
# A.niche.overlap <- c()
# 
# x <- 1
# while(x < 1000){
#   for(col in 1:ncol(OTUs.70)){
#     OTUs.70[,col] <- sample(OTUs.70[,col])
#   }
#   L1.p <- mean(niche.overlap(subset(OTUs.70, as.numeric(rownames(OTUs.70)) > turnover), method = "pianka"))
#   M1.p <- mean(niche.overlap(subset(OTUs.70, as.numeric(rownames(OTUs.70)) < turnover), method = "pianka"))
#   A.p <- mean(niche.overlap(OTUs.70, method = "pianka"))
#   print(paste(x, " : ", L1.p, M1.p, A.p))
#   L1.niche.overlap <- c(L1.niche.overlap, L1.p)
#   M1.niche.overlap <- c(M1.niche.overlap, M1.p)
#   A.niche.overlap <- c(A.niche.overlap, A.p)
#   x = x + 1
# }
# 
# all.niche.overlap <- as.data.frame(L1.niche.overlap)
# all.niche.overlap$M1.niche.overlap <- M1.niche.overlap
# all.niche.overlap$A.niche.overlap <- A.niche.overlap
# 
# write.csv(all.niche.overlap, file = "./data/all_nicheoverlap_null.csv")

all.niche.overlap <- read.csv("./data/all_nicheoverlap_null.csv", header = TRUE)[1:99,-1]

hline <- data.frame(cat = c(1,2,3), v=c(initial.overlap, L1.initial.overlap, M1.initial.overlap), col=c("A.niche.overlap", "L1.niche.overlap", "M1.niche.overlap"))


all.niche.overlap <- gather(all.niche.overlap, key = "Subset", value = "Pianka", L1.niche.overlap:A.niche.overlap)

cohensD(x = initial.overlap, y = subset(all.niche.overlap, all.niche.overlap$Subset == "A.niche.overlap")$Pianka)


niche_violin <- ggplot()+
  geom_segment(data=hline, aes(cat-0.4, xend=cat+0.4, y=v, yend=v, color = col), linetype = "dashed", cex = 2)+
  geom_violin(data = all.niche.overlap, aes(x = Subset, y = Pianka, color = Subset, fill = Subset), cex = 2)+
  geom_jitter(data = all.niche.overlap, aes(x = Subset, y = Pianka), cex = 0.1, alpha = 0.2)+
  ylab("Pianka's niche overlap index")+
  scale_x_discrete(labels = c("All sites", "<1 turnover", ">1 turnover"))+
  theme(legend.position = "none", axis.title.x = element_blank())

niche_violin

ggsave("./output/Niche_violin.pdf")
ggsave("./output/Niche_violin.png", width = 6, height = 8)
```


#Unused figures - Sequence Based

##Generate heat map of Day 20 communities with BC distance
```{r}
OTUsr_20_or <- OTUsr_20[order(rownames(OTUsr_20)),]

tau.db <- vegdist(OTUsr_20_or, method = "bray", upper = TRUE, diag = TRUE)
order <- rev(attr(tau.db, "Labels"))

pdf(file = "./output/RTLC_BC_heat.pdf")
jpeg(file = "./output/RTLC_BC_heat.jpeg")
levelplot(as.matrix(tau.db) [, order], aspect = "iso", col.regions = inferno, xlab = "log(Tau, 10)", ylab = "log(Tau, 10)", scales = list(x=list(at=seq(1,49,1), labels = c("1.5", rep("",2) , "2.5", rep("",4),  "3.5", rep("", 3),"4", rep("", 2), "4.5", rep("",4), "5.25", rep("", 3), "6", rep("", 5), "7", rep("", 3), "8")), y=list(at=seq(1,49,1), labels = rev(c("1.5", rep("",2) , "2.5", rep("",4),  "3.5", rep("", 3),"4", rep("", 2), "4.5", rep("",4), "5.25", rep("", 3), "6", rep("", 5), "7", rep("", 3), "8")))), main = "Bray-Curtis Distance")
dev.off()
```

##Want OTUs by tau heat map to show horseshoe
```{r}
OTU_r_trim <- OTU_r[1:69,]
OTU_heat <- as.data.frame(colnames(OTU_r_trim))
colnames(OTU_heat) <- c("OTU")
OTU_heat$N <- colSums(OTU_r_trim)


for (otu in OTU_heat$OTU){
  x = 0
  for(site in rownames(OTU_r_trim)){
    x = x + (OTU_r_trim[site, otu] * Tau_Seq$Tau[Tau_Seq$Seq_Sample == site])
  }
  OTU_heat$Tau[OTU_heat$OTU == otu] <- x/OTU_heat$N[OTU_heat$OTU == otu]
}

OTU_heat <- OTU_heat[order(OTU_heat$Tau),]

OTU_heat <- OTU_heat[order(OTU_heat$Weight),]

colnames(OTU_heat) <- c("OTU", "N", "Weight")

OTU_heat <- cbind(OTU_heat, Tau_Seq[1:69,"Tau"])

plot(x = as.numeric(OTUsr_20$Tau), y = OTUsr_20$Otu00628)
#########

colnames(OTUsr_20)[1] <- c("Tau")
data_long <- gather(OTUsr_20, OTU, N, Otu00001:Otu33565, factor_key = TRUE)
data_long$OTU <- factor(data_long$OTU, levels = rev(as.list(OTU_heat$OTU)))
print(levels(data_long$OTU))

map <- ggplot(data_long, aes(Tau, OTU, fill = log(N, 10)))+
  geom_tile()+
  theme(axis.ticks.x = element_blank(), axis.ticks.y = element_blank(), axis.text.x = element_blank(), axis.text.y = element_blank())+
  scale_fill_continuous(type = "viridis", labels = label_math(expr = 10^.x, format = force))

map

ggsave("./output/map.pdf")
ggsave("./output/map.png")

```

##Species Turnover (Whittaker's Turnover between two sites)
```{r}
#Betaw = (S1 - gamma) + (S2 - gamma)/mean(S1, S2)
Betaw <- function(site1 = "", site2 = ""){
  site1 <- t(OTUs.PA[site1,])
  site2 <- t(OTUs.PA[site2,])
  site1 <- as.data.frame(subset(site1, select = site1 > 0))
  site2 <- as.data.frame(subset(site2, select = site2 > 0))
  gamma <- as.numeric(length(intersect(colnames(site1), colnames(site2))))
  bw <- ((ncol(site1) - gamma) + (ncol(site2) - gamma))/mean(ncol(site1), ncol(site2))
  return(bw)
}

Tau$Betaw <- NA
row <- 1
while(row < nrow(Tau)){
  if(!is.na(Tau[row, "Day_0_Seq"]) && !is.na(Tau[row, "Day_20_Seq"])){
    Tau[row, "Betaw"] <- Betaw(Tau[row, "Day_0_Seq"], Tau[row, "Day_20_Seq"])
  }
  row = row + 1
}

Bw_lm <- lm(Betaw ~ Tau, data = Tau)
Bw_poly <- lm(Betaw ~ poly(Tau, 2, raw = TRUE), data = Tau)
Bw_gam <- gam(Betaw ~ s(Tau), family = gaussian(link = "identity"), data = Tau, method = "REML")
Bw_gam_re <- gam(Betaw ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")

AIC(Bw_lm, Bw_poly, Bw_gam, Bw_gam_re)
anova(Bw_lm, Bw_poly, Bw_gam, Bw_gam_re)

summary(Bw_lm)
summary(Bw_poly)
summary(Bw_gam)
summary(Bw_gam_re)

k.check(Bw_gam)
k.check(Bw_gam_re)
gam.check(Bw_gam)
gam.check(Bw_gam_re)

betaw <- ggplot(data = Tau, aes(x = Tau, y = Betaw))+
  geom_point()+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  geom_smooth(method = "lm", formula = y ~ poly(x, 2, raw = TRUE))+
  xlab(expression(paste(tau, " (h)")))+
  stat_poly_eq(aes(label = paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),
            label.x = "left", label.y = "top", formula = y~poly(x, 2, raw = TRUE), parse = TRUE, size = 4)+
  ylab(expression(beta[W]))
  
betaw

ggsave("./output/RTLC_Bw.pdf")
ggsave("./output/RTLC_Bw.png")
```
#Original IBMs by V
```{r}
#IBM <- read.csv("data/IBM/SimData.csv", header = TRUE, sep = ",")

#IBM_2 <- subset(IBM, log(IBM$V, 10) <= 2 & log(IBM$V, 10) > 1)
#IBM_sum <- as.data.frame(unique(IBM_2$sim))
#colnames(IBM_sum) <- c("sim")

#for(x in unique(IBM_sum$sim)){
#  IBM_sum[IBM_sum$sim == x, "N"] <- mean(subset(IBM_2, IBM_2$sim == x)$total.abundance)
#  IBM_sum[IBM_sum$sim == x, "V"] <- mean(subset(IBM_2, IBM_2$sim == x)$V)
#  IBM_sum[IBM_sum$sim == x, "Q"] <- mean(subset(IBM_2, IBM_2$sim == x)$Q)
#  IBM_sum[IBM_sum$sim == x, "S"] <- mean(subset(IBM_2, IBM_2$sim == x)$species.richness)
#  IBM_sum[IBM_sum$sim == x, "E"] <- mean(subset(IBM_2, IBM_2$sim == x)$simpson.e, na.rm = TRUE)
#  IBM_sum[IBM_sum$sim == x, "P"] <- mean(subset(IBM_2, IBM_2$sim == x)$ind.production, na.rm = TRUE)
#  IBM_sum[IBM_sum$sim == x, "B"] <- mean(subset(IBM_2, IBM_2$sim == x)$whittakers.turnover, na.rm = TRUE)
#  IBM_sum[IBM_sum$sim == x, "Res"] <- mean(subset(IBM_2, IBM_2$sim == x)$avg.per.capita.efficiency1e, na.rm = TRUE)
#}

#write.csv(IBM_sum, "data/IBM/IBM_sum.csv", row.names = FALSE)

IBM_sum <- read.csv("data/IBM/IBM_sum.csv", header = TRUE, sep = ",")
```

```{r}
N_IBM <- ggplot(data = subset(IBM_sum, log(IBM_sum$V, 10) <= 2 & log(IBM_sum$V, 10) > 1), aes(y = log(N, 10), x = log(V/Q, 10)))+
  geom_point()+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(1, 5))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force))+
  xlab(expression(paste(tau)))+
  ylab(expression(paste("Total abundance, ", italic("N"))))+
  stat_smooth(method = "loess", color = "blue", cex = 1.25, level = 0.95, fill = alpha("black", 0.4))+
  theme(axis.title = element_text(size = 25))

N_IBM


ggsave("./output/IBM_N.pdf")
ggsave("./output/IBM_N.png", width = 6.5, height = 5)


S_IBM <- ggplot(data = subset(IBM_sum, log(IBM_sum$V, 10) <= 2 & log(IBM_sum$V, 10) > 1), aes(y = log(S, 10), x = log(V/Q, 10)))+
  geom_point()+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(1, 5))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(0, 1.5))+
  xlab(expression(paste(tau)))+
  ylab(expression(paste("Species richness, ", italic("S"))))+
  stat_smooth(method = "loess", color = "blue", cex = 1.25, level = 0.95, fill = alpha("black", 0.4))+
  theme(axis.title = element_text(size = 25))

S_IBM

ggsave("./output/IBM_S.pdf")
ggsave("./output/IBM_S.png", width = 6.5, height = 5)


E_IBM <- ggplot(data = subset(IBM_sum, log(IBM_sum$V, 10) <= 2 & log(IBM_sum$V, 10) > 1), aes(y = log(E, 10), x = log(V/Q, 10)))+
  geom_point()+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(1, 5))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(-0.5, 0))+
  xlab(expression(paste(tau)))+
  ylab(expression(paste("Species evenness, ", italic("E"))))+
  stat_smooth(method = "loess", color = "blue", cex = 1.25, level = 0.95, fill = alpha("black", 0.4))+
  theme(axis.title = element_text(size = 25))

E_IBM

ggsave("./output/IBM_E.pdf")
ggsave("./output/IBM_E.png", width = 6.5, height = 5)


P_IBM <- ggplot(data = subset(IBM_sum, log(IBM_sum$V, 10) <= 2 & log(IBM_sum$V, 10) > 1), aes(y = log(P, 10), x = log(V/Q, 10)))+
  geom_point()+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(1, 5))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(-2, 1.5))+
  xlab(expression(paste(tau)))+
  ylab(expression(paste("Productivity, ", italic("P"))))+
  stat_smooth(method = "loess", color = "blue", cex = 1.25, level = 0.95, fill = alpha("black", 0.4))+
  theme(axis.title = element_text(size = 25))

P_IBM


ggsave("./output/IBM_P.pdf")
ggsave("./output/IBM_P.png", width = 6.5, height = 5)


B_IBM <- ggplot(data = subset(IBM_sum, log(IBM_sum$V, 10) <= 2 & log(IBM_sum$V, 10) > 1), aes(y = B, x = log(V/Q, 10)))+
  geom_point()+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(1, 5))+
  ylim(c(0,2))+ 
  xlab(expression(paste(tau)))+
  ylab(expression(paste("Species turnover, ", beta[w])))+
  stat_smooth(method = "loess", color = "blue", cex = 1.25, level = 0.95, fill = alpha("black", 0.4))+
  theme(axis.title = element_text(size = 25))

B_IBM


ggsave("./output/IBM_B.pdf")
ggsave("./output/IBM_B.png", width = 6.5, height = 5)

Res_IBM <- ggplot(data = subset(IBM_sum, log(IBM_sum$V, 10) <= 2 & log(IBM_sum$V, 10) > 1), aes(y = log(Res, 10), x = log(V/Q, 10)))+
  geom_point()+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(1, 5))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(-2, 1))+
  xlab(expression(paste(tau)))+
  ylab("Resource specialization")+
  stat_smooth(method = "loess", color = "blue", cex = 1.25, level = 0.95, fill = alpha("black", 0.4))+
  theme(axis.title = element_text(size = 25))

Res_IBM


ggsave("./output/IBM_Res.pdf")
ggsave("./output/IBM_Res.png", width = 6.5, height = 5)
```