---
title: "Residence Time Experiment"
author: "Emmi Mueller"
date: "July 23, 2019"
output:
  pdf_document: default
  html_document: default
editor_options: 
  chunk_output_type: inline
---

##Setup
```{r setup, include = FALSE}
rm(list=ls())
getwd()

package.list <- c("Matrix", "jpeg", "cli", "png", "vegan", "ggplot2", "ggpubr", "cowplot", "ggpmisc", "scales", "fitdistrplus", "tidyr", "ade4", "viridis", "gplots", "BiodiversityR", "indicspecies", "knitr", "here", "dplyr", "BiocManager", "car", "here", "actuar", "tibble", "scales", "dplyr", "stringr", "iNEXT", "lmtest", "betapart", "reshape2", "mgcv", "modelr", "purrr", "lme4", "tidymv", "merTools", "MuMIn", "htmltools", "ggrepel", "broom", "tidyverse", "pander", "spaa")

for (package in package.list) {
  if (!require(package, character.only=T, quietly=T)) { 
    install.packages(package, dependencies = TRUE)
    library(package, character.only=T)
  } }

#BiocManager::install()
#BiocManager::install(c("flowCore", "ggcyto"))

source("./code/residence-time-test_functions.R")
source("./code/mothur_tools.R")
```

##Set up figure theme
```{r figure_setup}
my.cols <- RColorBrewer::brewer.pal(n = 4, name = "Greys")[3:4]

# Set theme for figures in the paper
theme_set(theme_classic() + 
  theme(axis.title = element_text(size = 16),
        axis.title.x = element_text(margin = margin(t = 15, b = 15)),
        axis.title.y = element_text(margin = margin(l = 15, r = 15)),
        axis.text = element_text(size = 14),
        axis.text.x = element_text(margin = margin(t = 5)),
        axis.text.y = element_text(margin = margin(r = 5)),
        #axis.line.x = element_line(linewidth = 1),
        #axis.line.y = element_line(linewidth = 1),
        axis.line.x = element_blank(),
        axis.line.y = element_blank(),
        axis.ticks.x = element_line(linewidth = 1),
        axis.ticks.y = element_line(linewidth = 1),
        axis.ticks.length = unit(.1, "in"),
        panel.border = element_rect(color = "black", fill = NA, linewidth = 1.5),
        legend.text = element_text(size = 10),
        strip.text = element_text(size = 14),
        strip.background = element_blank()
        ))
```


```{r}
#Define Inputs

#Tau = general design file for experiment, paired Day 0 and 20
Tau <- read.csv("data/RTLC/2021_RTLC_Tau_Combined.csv", header = TRUE)
Tau_ctrl <- Tau[50:51,]
Tau <- Tau[1:49,]
Tau$Tau <- as.numeric(Tau$Tau)
Tau$Set <- as.character(Tau$Set)
Tau$Day_0_Seq[Tau$Day_0_Seq == ""] <- NA
Tau$Day_20_Seq[Tau$Day_20_Seq == ""] <- NA

#Tau_Seq = Full list of sequenced samples, non-paired Day 0 and 20
Tau_Seq <- read.csv("data/RTLC/2021_RTLC_Tau_Seq.csv", header = TRUE)

#Import Shared Files
OTUs <- read.otu(shared = "mothur/RTLC_A+B/RTLC.final.shared", cutoff = "0.03")
rownames(OTUs)[rownames(OTUs) == "GSF3365_RTLC"] <- "GSF3365_RTLC_002"

ASVs <- read.csv("data/RTLC_ASV.csv", header = TRUE, sep = ",")
ASVs$X <- sub("-", "_", ASVs$X)
rownames(ASVs) <- ASVs$X
ASVs <- ASVs[,-1]
head(ASVs[,1:10])

#Import Taxonomy
OTU.tax <- read.tax(taxonomy = "mothur/RTLC_A+B/RTLC.final.taxonomy", format = "rdp")

#Abundance data
#Tau <- N_fxn(read.csv("data/RTLC/RTLC_S1/20210602_RTLC_S1_N.csv", header = TRUE), Tau)
Tau <- N_fxn(read.csv("data/RTLC/RTLC_S2/20210628_RTLC_S2_N_M1gate.csv", header = TRUE), Tau)
Tau <- N_fxn(read.csv("data/RTLC/RTLC_S3/20210723_RTLC_S3_N_M1gate.csv", header = TRUE), Tau)
Tau <- N_fxn(read.csv("data/RTLC/RTLC_S4/20210904_RTLC_S4_N_M1gate.csv", header = TRUE), Tau)


#Biofilm data
Tau <- OT_fxn(read.csv("data/RTLC/RTLC_S1/20210602_RTLC_S1_Otoole.csv", header = TRUE), Tau)
Tau <- OT_fxn(read.csv("data/RTLC/RTLC_S2/20210628_RTLC_S2_Otoole.csv", header = TRUE), Tau)
Tau <- OT_fxn(read.csv("data/RTLC/RTLC_S3/20210723_RTLC_S3_Otoole.csv", header = TRUE), Tau)
Tau <- OT_fxn(read.csv("data/RTLC/RTLC_S4/20210904_RTLC_S4_Otoole.csv", header = TRUE), Tau)


#BP data
Tau <- as.data.frame(BP_fxn(read.csv("data/RTLC/RTLC_S1/20210602_RTLC_S1_BP.csv", header = FALSE), Tau, 1))
Tau <- as.data.frame(BP_fxn(read.csv("data/RTLC/RTLC_S2/20210628_RTLC_S2_BP.csv", header = FALSE), Tau, 2))
Tau <- as.data.frame(BP_fxn(read.csv("data/RTLC/RTLC_S3/20210723_RTLC_S3_BP.csv", header = FALSE), Tau, 3))
Tau <- as.data.frame(BP_fxn(read.csv("data/RTLC/RTLC_S4/20210904_RTLC_S4_BP.csv", header = FALSE), Tau, 4))
Tau$ind_P <- Tau$uMChr/Tau$N


#Biolog data
Tau <- EP_fxn(read.csv("data/RTLC/eco.data_rt.txt", header = TRUE, sep = "\t"), Tau)

Tau$Set <- as.factor(Tau$Set)


Tau$Tau <- log(((10^Tau$Tau)/60), 10)

Tau$N_turn <- (20 * 24)/(10^Tau$Tau)

turnover <- log(20*24, 10)

Tau$Turn <- Tau$Tau <= log(24*20, 10)

Tau$Pump <- Tau$Tau < 1.6
```



##Read in all EcoPlate 48 hour data and create a resource by sample matrix (wbs - well response by site) and environmental matrix (env - Tau only)

```{r}
EcoPlate <- read.csv("data/RTLC/eco.data_rt_S1_48.txt", header = TRUE, sep = "\t")
EcoPlate <- rbind(EcoPlate, read.csv("data/RTLC/eco.data_rt_S2_48.txt", header = TRUE, sep = "\t"))
EcoPlate <- rbind(EcoPlate, read.csv("data/RTLC/eco.data_rt_S3_48.txt", header = TRUE, sep = "\t"))
EcoPlate <- rbind(EcoPlate, read.csv("data/RTLC/eco.data_rt_S4_48.txt", header = TRUE, sep = "\t"))
EcoPlate_env <- subset(EcoPlate, select = c(Tau, Set))
EcoPlate$Tau <- log(((10^EcoPlate$Tau)/60), 10)
rownames(EcoPlate) <- EcoPlate$Tau
EcoPlate <- subset(EcoPlate, select =-c(Tau, NumRes, Avg, Set, Hours))
```

##Load Resource Use data for first 72 hours
```{r}
EcoPlate_72 <- read.csv("data/RTLC/eco.data_rt_S1_0.txt", header = TRUE, sep = "\t")
file_list <- c("data/RTLC/eco.data_rt_S1_24.txt", "data/RTLC/eco.data_rt_S1_48.txt", "data/RTLC/eco.data_rt_S1_72.txt", "data/RTLC/eco.data_rt_S2_0.txt", "data/RTLC/eco.data_rt_S2_24.txt", "data/RTLC/eco.data_rt_S2_48.txt", "data/RTLC/eco.data_rt_S2_72.txt", "data/RTLC/eco.data_rt_S3_0.txt", "data/RTLC/eco.data_rt_S3_24.txt", "data/RTLC/eco.data_rt_S3_48.txt", "data/RTLC/eco.data_rt_S3_72.txt", "data/RTLC/eco.data_rt_S4_0.txt", "data/RTLC/eco.data_rt_S4_24.txt", "data/RTLC/eco.data_rt_S4_48.txt", "data/RTLC/eco.data_rt_S4_72.txt")

for(file in file_list){
  EcoPlate_72 <- rbind(EcoPlate_72, read.csv(file, header = TRUE, sep = "\t"))
}

EcoPlate_72$Tau <- log(((10^EcoPlate_72$Tau)/60), 10)

long_EP <- EcoPlate_72 %>% gather(C_Source, OD, X2.Hydroxy.Benzoic.Acid:Avg)
```

##Load slopes of resource use data (48-24 hours)
```{r}
EcoPlate_S1 <- read.csv("data/RTLC/eco.data_rt_S1_48.txt", header = TRUE, sep = "\t")
EcoPlate_S1 <- cbind(EcoPlate_S1[,1:2], (EcoPlate_S1[,4:34] - read.csv("data/RTLC/eco.data_rt_S1_24.txt", header = TRUE, sep = "\t")[,4:34]))
EcoPlate_S1[,4:33] <- EcoPlate_S1[,4:33]/2

EcoPlate_S2 <- read.csv("data/RTLC/eco.data_rt_S2_48.txt", header = TRUE, sep = "\t")
EcoPlate_S2 <- cbind(EcoPlate_S2[,1:2], (EcoPlate_S2[,4:34] - read.csv("data/RTLC/eco.data_rt_S2_24.txt", header = TRUE, sep = "\t")[,4:34]))
EcoPlate_S2[,4:33] <- EcoPlate_S2[,4:33]/2

EcoPlate_S3 <- read.csv("data/RTLC/eco.data_rt_S3_48.txt", header = TRUE, sep = "\t")
EcoPlate_S3 <- cbind(EcoPlate_S3[,1:2], (EcoPlate_S3[,4:34] - read.csv("data/RTLC/eco.data_rt_S3_24.txt", header = TRUE, sep = "\t")[,4:34]))
EcoPlate_S3[,4:33] <- EcoPlate_S3[,4:33]/2

EcoPlate_S4 <- read.csv("data/RTLC/eco.data_rt_S4_48.txt", header = TRUE, sep = "\t")
EcoPlate_S4 <- cbind(EcoPlate_S4[,1:2], (EcoPlate_S4[,4:34] - read.csv("data/RTLC/eco.data_rt_S4_24.txt", header = TRUE, sep = "\t")[,4:34]))
EcoPlate_S4[,4:33] <- EcoPlate_S4[,4:33]/2

EcoPlate_Slope <- rbind(EcoPlate_S1, EcoPlate_S2, EcoPlate_S3, EcoPlate_S4)

ResType <- read.csv("code/resource_type.txt", header = TRUE, sep = "\t")

colnames(EcoPlate_Slope) <- c("Tau", "Set", "2-Hydroxy.Benzoic.Acid", "4-Hydroxy.Benzoic.Acid", "alpha-Cyclodextrin", "alpha-D-Lactose", "alpha-Ketobutyric.Acid", "beta-Methyl-D-Glucoside", "D-Cellobiose", "D-Galactonic.Acid.gamma-Lactone", "D-Galacturonic.Acid","D-Glucosaminic.Acid", "D-Malic.Acid", "D-Mannitol", "D-Xylose", "D,L-alpha-Glycerol.Phosphate","gamma-Hydroxybutyric.Acid", "Glucose-1-Phosphate", "Glycogen", "Glycyl-L-Glutamic.Acid", "i-Erythritol","Itaconic.Acid","L-Arginine", "L-Asparagine","L-Phenylalanine", "L-Serine", "L-Threonine","N-Acetyl-D-Glucosamine", "Phenylethylamine", "Putrescine", "Pyruvic.Acid.Methyl.Ester", "Tween.40", "Tween.80")

EcoPlate_Slope$Tau <- log(((10^EcoPlate_Slope$Tau)/60), 10)


long_EP_slope <- EcoPlate_Slope %>% gather(C_Source, Slope, "2-Hydroxy.Benzoic.Acid":"Tween.80")

ResType <- distinct(ResType)

long_EP_slope$Type <- NA

for(row in rownames(long_EP_slope)){
  long_EP_slope[row,"Type"] <- ResType$Type[ResType$Resource == long_EP_slope[row,"C_Source"]]
}


```

##Clean and rarefy the OTU table
```{r}
#Remove samples with fewer than 10000 reads
coverage <- rowSums(OTUs)
cutoff <- 10000
lows <- which(coverage < cutoff)
if (length(lows) > 0){
  OTUs <- OTUs[-which(coverage < cutoff),]
}

#Remove OTUs with less thatn 2 reads across all samples
OTUs <- OTUs[,which(colSums(OTUs) > 2)]

#Generate rarefacation plot
otu.min <- min(rowSums(OTUs))
asv.min <- min(rowSums(ASVs))

# png("./output/OTU_rarefaction.png", width = 800, height = 480)
# par(mar = c(1, 1, 1, 1) + 0.1)
#   rarecurve(x = OTUs, step = 100, sample = otu.min, col = "blue", cex = 0.6, las = 1)
# #  abline(0,1, col = "red")
# #  text(1500, 1500, "1:1", pos = 2, col = "red")
# dev.off()

#rarefy OTUs to the sample with the lowest number of reads then remove OTUs with 0 reads after rarefaction
set.seed(47405)
OTU_r <- rrarefy(OTUs, otu.min)
OTU_r <- OTU_r[,-which(colSums(OTU_r) == 0)]
rm(OTUs)

ASV_r <- rrarefy(ASVs, asv.min)
ASV_r <- ASV_r[,-which(colSums(ASV_r) == 0)]
rm(ASVs)

ASVsr_20 <- ASV_r[rownames(ASV_r) %in% unique(Tau$Day_20_Seq),]
ASVsr_20 <- ASVsr_20[,which(colSums(ASVsr_20) >0)]
rownames(ASVsr_20) <- log(((10^as.numeric(subset(Tau_Seq, Day == 20)$Tau))/60), 10)

Tau_20 <- subset(Tau[is.na(Tau$Day_20_Seq) == 0,])
Tau_Seq$Day <- as.factor(Tau_Seq$Day)
Tau_Seq$Tau <- as.numeric(Tau_Seq$Tau)

#Generate OTU tables with p/a, relative abundance
OTUsREL <-decostand(OTU_r, method = "total")
OTUsREL_20 <- OTUsREL[rownames(OTUsREL) %in% unique(Tau$Day_20_Seq),]
OTUsREL_20 <- OTUsREL_20[,which(colSums(OTUsREL_20) > 0)]
rownames(OTUsREL_20) <- log(((10^subset(Tau_Seq, Day == 20)$Tau)/60), 10)
# 
OTUs.PA <- decostand(OTU_r, method = "pa")
OTUsPA_20<- OTUs.PA[rownames(OTUs.PA) %in% unique(Tau$Day_20_Seq),]
OTUsPA_20 <- OTUsPA_20[,which(colSums(OTUsPA_20) > 0)]
rownames(OTUsPA_20) <- log(((10^subset(Tau_Seq, Day == 20)$Tau)/60), 10)


OTUsr_20 <- OTU_r[rownames(OTU_r) %in% unique(Tau$Day_20_Seq),]
OTUsr_20 <- OTUsr_20[,which(colSums(OTUsr_20) >0)]
rownames(OTUsr_20) <- log(((10^subset(Tau_Seq, Day == 20)$Tau)/60), 10)

OTU.tax.20 <- OTU.tax[OTU.tax$OTU %in% unique(colnames(OTUsr_20)),]

Tau_Seq$ACE <- S.ace(OTU_r)

#
S <- as.data.frame(cbind(row.names(OTU_r), unname(S.cal(OTU_r))))
S_ASV <- as.data.frame(cbind(row.names(ASV_r), unname(S.cal(ASV_r))))
Tau <- S_fxn(S_ASV, Tau)
#Tau <- S_fxn(S, Tau)
Tau$Day_20.S <- as.numeric(Tau$Day_20.S)
Tau$Day_0.S <- as.numeric(Tau$Day_0.S)

SimpE <- as.data.frame(cbind(row.names(OTU_r), unname(SimpE.cal(OTU_r))))
SimpE_ASV <- as.data.frame(cbind(row.names(ASV_r), unname(SimpE.cal(ASV_r))))
Tau <- SimpE_fxn(SimpE_ASV, Tau)
#Tau <- SimpE_fxn(SimpE, Tau)
Tau$Day_0.SimpE <- as.numeric(Tau$Day_0.SimpE)
Tau$Day_20.SimpE <- as.numeric(Tau$Day_20.SimpE)
```
#Figure 1. Community Structure

##Cstat - Microbial abundance
```{r}
# test <- as.data.frame(seq(0, 0.1, 0.0001))
# colnames(test) <- "n"
# for(i in seq(0, 0.1, 0.0001)){
#   N_gam_re <- gam(log_N ~ s(Tau, sp = i) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")
#   test[test$n == i, "AIC"] <- N_gam_re$aic
#   test[test$n == i, "REML"] <- N_gam_re$gcv.ubre[[1]]
# }
# 
# plot(test$AIC ~ test$n)
# plot(test$REML ~ test$n, ylim = c(0,30))

N_gam <- gam(log_N ~ s(Tau), family = gaussian(link = "identity"), data = Tau, method = "REML")
N_gam_re <- gam(log_N ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")

AIC(N_gam, N_gam_re)
anova(N_gam, N_gam_re, test = "Chisq")

summary(N_gam_re)
k.check(N_gam_re)
gam.check(N_gam_re)
mean(summary(N_gam_re)$s.table[,4])

gam.vcomp(N_gam_re)

N_Tau <- predict_gam(N_gam_re, exclude_terms = "s(Set)") %>%
  filter(Set == "1") %>% 
  ggplot(aes(Tau, fit))+
#  geom_vline(xintercept = turnover, linetype = "dashed", color = "blue", size = 1)+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_point(data = Tau, aes(x = Tau, y = log_N))+
  theme(axis.title.x = element_blank())+
  ylab("N (cells/mL)")+
  scale_x_continuous(labels = label_math(10^.x))+
  scale_y_continuous(breaks = c(6, 7, 8), labels = label_math(10^.x))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(N_gam_re)$dev, 2))))+
theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.8), "cm"))

N_Tau

summary(N_gam_re)$dev

ggsave("./output/RTLC_N.pdf")
ggsave("./output/RTLC_N.png")
```

##Cstat - Observed Richness
```{r}
# test <- as.data.frame(seq(0, 0.1, 0.0001))
# colnames(test) <- "n"
# for(i in seq(0, 0.1, 0.0001)){
#   S_gam_re <- gam(log(Day_20.S, 10) ~ s(Tau, sp = i) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")
#   test[test$n == i, "AIC"] <- S_gam_re$aic
#   test[test$n == i, "REML"] <- S_gam_re$gcv.ubre[[1]]
# }
# 
# plot(test$AIC ~ test$n)
# plot(test$REML ~ test$n, ylim = c(-50,-20))

S_gam <- gam(Day_20.S ~ s(Tau, k = 14), family = gaussian(link = "identity"), data = Tau, method = "REML")
S_gam_re <- gam(Day_20.S ~ s(Tau)  + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")

k.check(S_gam)
gam.check(S_gam)

AIC(S_gam, S_gam_re)
anova(S_gam, S_gam_re, test = "Chisq")

summary(S_gam_re)
k.check(S_gam_re)
gam.check(S_gam_re)
mean(summary(S_gam_re)$s.table[,4])

S.ob_Tau <- predict_gam(S_gam) %>%
  ggplot(aes(Tau, fit))+
#  geom_vline(xintercept = turnover, linetype = "dashed", color = "blue", size = 1)+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_point(data = Tau, aes(x = Tau, y = Day_20.S))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
#  scale_y_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab("Observed S")+
  theme(axis.title.x = element_blank())+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(S_gam)$dev, 2))))+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.5), "cm"))
  

S.ob_Tau

ggsave("./output/RTLC_S.pdf")
ggsave("./output/RTLC_S.png")
```

#Cstat - Simpson's Evenness (E[1/D])
```{r}
# test <- as.data.frame(seq(0, 0.1, 0.0001))
# colnames(test) <- "n"
# for(i in seq(0, 0.1, 0.0001)){
#   SimpE_gam <- gam(Day_20.SimpE ~ s(Tau, sp = i), family = gaussian(link = "identity"), data = Tau, method = "REML")
#   test[test$n == i, "AIC"] <- SimpE_gam$aic
#   test[test$n == i, "REML"] <- SimpE_gam$gcv.ubre[[1]]
# }
# 
# plot(test$AIC ~ test$n)
# plot(test$REML ~ test$n)

SimpE_gam <- gam(Day_20.SimpE ~ s(Tau, k = 15), family = gaussian(link = "identity"), data = Tau, method = "REML")
SimpE_gam_re <- gam(Day_20.SimpE ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")

AIC(SimpE_gam, SimpE_gam_re)
anova(SimpE_gam, SimpE_gam_re, test = "Chisq")

summary(SimpE_gam)
k.check(SimpE_gam)
gam.check(SimpE_gam)
mean(summary(SimpE_gam)$s.table[,4])

SimpE_Tau <- predict_gam(SimpE_gam) %>%
  ggplot(aes(Tau, fit))+
#  geom_vline(xintercept = turnover, linetype = "dashed", color = "blue", size = 1)+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_point(data = Tau, aes(x = Tau, y = Day_20.SimpE))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab(expression("E"[D^"-1"/S]))+
  xlab(expression(paste(tau, " (h)")))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(SimpE_gam)$dev, 2))))

SimpE_Tau

ggsave("./output/RTLC_SimpE.pdf")
ggsave("./output/RTLC_SimpE.png")

```
##Draw Figure
```{r}
ggdraw()+
  draw_plot(N_Tau, x = 0, y = 0.65, width = 1, height = 0.3) + 
  draw_plot(S.ob_Tau, x = 0, y = 0.35, width = 1, height = 0.3) + 
  draw_plot(SimpE_Tau, x = 0, y = 0.0, width = 1, height = 0.35)+
  draw_plot_label(label = c("A", "B", "C"), size = 30, x = c(0.06,0.06,0.06), y = c(0.96,0.65,0.35))+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("./output/CommStr_Cstat_ASV.pdf")
ggsave("./output/CommStr_Cstat_ASV.png", width = 5, height = 10)
```
```{r}
niche.overlap <- c()

print(mean(niche.overlap(as.data.frame(OTUs.90), method = "pianka")))
print(paste(x, " : ", mean(niche.overlap(as.data.frame(OTUs.90), method = "pianka"))))

x <- 1
while(x < 100){
  for(col in 1:ncol(OTUs.90)){
    OTUs.90[,col] <- sample(OTUs.90[,col])
  }
  print(paste(x, " : ", mean(niche.overlap(as.data.frame(OTUs.90), method = "pianka"))))
  niche.overlap <- c(niche.overlap, mean(niche.overlap(as.data.frame(OTUs.90), method = "pianka")))
  x = x + 1
}
```

#Figure 2. Community composition
```{r}
tau.db <- vegdist(OTUsr_20, method = "bray", upper = TRUE, diag = TRUE)

tau.pcoa <- cmdscale(tau.db, eig = TRUE, k = 3)
explainvar1 <- round(tau.pcoa$eig[1] / sum(tau.pcoa$eig), 3) *100
explainvar2 <- round(tau.pcoa$eig[2] / sum(tau.pcoa$eig), 3) *100
explainvar3 <- round(tau.pcoa$eig[3] / sum(tau.pcoa$eig), 3) *100
sum.eig <- sum(explainvar3, explainvar2, explainvar1)

tau.plot <- as.data.frame(tau.pcoa$points)
tau.plot <- cbind(tau.plot, Tau_20)
tau.plot$Pump <- factor(tau.plot$Pump, levels=c("TRUE", "FALSE"))
tau.plot$Turn <- factor(tau.plot$Turn, levels=c("TRUE", "FALSE"))

spe.corr <- add.spec.scores(tau.pcoa, OTUsREL_20, method = "cor.scores")$cproj
corrcut <- 0.7
imp.spp <- as.data.frame(spe.corr[abs(spe.corr[,1]) >= corrcut | abs(spe.corr[,2]) >= corrcut,])
imp.spp <- na.omit(imp.spp)

imp.tax <- subset(OTU.tax, OTU == rownames(imp.spp)[1])

x <- 2                        
while (x <= nrow(imp.spp)){
  imp.tax <- rbind(imp.tax, subset(OTU.tax, OTU == rownames(imp.spp)[x]))
  x <- x + 1
}

imp.spp <- cbind(imp.spp, imp.tax)

imp.otus <- as.data.frame(row.names(OTUsREL_20))
imp.spp.otu <- c()
for(otu in unique(imp.spp$OTU)){
    imp.otus <- cbind(imp.otus, OTUsREL_20[,otu])
    imp.spp.otu <- c(imp.spp.otu, otu)
}
colnames(imp.otus) <- c("Tau", imp.spp.otu)
imp.otus$Tau <- log(((10^as.numeric(imp.otus$Tau))/60), 10)
```

```{r}
OTUs.90 <- cbind(Tau_20$Tau, OTUsREL_20[,colnames(OTUsREL_20) %in% colnames(as.data.frame(OTUsPA_20) %>% select_if(colSums(OTUsPA_20) >= 28))])
write.csv(OTUs.90, "./data/OTUs_80.csv", row.names = TRUE)

colnames(OTUs.90) <- c("Tau", colnames(as.data.frame(OTUsPA_20) %>% select_if(colSums(OTUsPA_20) >= 28)))
OTUs.90 <- as.data.frame(OTUs.90) %>% gather(OTU, REL, Otu00001:Otu04427)

OTUs.90.REL <- data.frame()
for(otu in unique(OTUs.90$OTU)){
  max <- max(subset(OTUs.90, OTUs.90$OTU == otu)$REL)
  OTUs.90.REL <- rbind(OTUs.90.REL, subset(OTUs.90, OTUs.90$REL == max & OTUs.90$OTU == otu))
}

niche.tax <- subset(OTU.tax, OTU == OTUs.90.REL[1, 2])
x <- 2                        
while (x <= nrow(OTUs.90.REL)){
  niche.tax <- rbind(niche.tax, subset(OTU.tax, OTU == OTUs.90.REL[x, 2]))
  x <- x + 1
}

OTUs.90.REL <- cbind(OTUs.90.REL, niche.tax[,-1])
unique(OTUs.90.REL$Phylum)

OTUs_plot <- ggplot(OTUs.90.REL, aes(x = Tau, y = REL))+
  geom_point()+
  ylab("Relative abundance")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  xlab(expression(paste(tau, " (h)")))+
  theme(legend.position = "none", axis.title = element_text(size = 20))+
  facet_wrap(~Phylum)
OTUs_plot

ggsave("./output/Niche_phylum.pdf")
ggsave("./output/Niche_phylum.png", width = 20, height = 20)

OTUs_plot <- ggplot(subset(OTUs.90.REL, OTUs.90.REL$Phylum == "Acidobacteriota"), aes(x = Tau, y = REL))+
  geom_point(aes(color = Order))+
  ylab("Relative abundance")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  xlab(expression(paste(tau, " (h)")))+
  theme(legend.position = "none", axis.title = element_text(size = 20))+
  facet_wrap(~Class)
OTUs_plot

ggsave("./output/Niche_Acido.pdf")
ggsave("./output/Niche_Acido.png", width = 20, height = 20)

OTUs_plot <- ggplot(subset(OTUs.90.REL, OTUs.90.REL$Phylum == "Proteobacteria"), aes(x = Tau, y = REL))+
  geom_point(aes(color = Order))+
  ylab("Relative abundance")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  xlab(expression(paste(tau, " (h)")))+
  theme(legend.position = "none", axis.title = element_text(size = 20))+
  facet_wrap(~Class)
OTUs_plot

ggsave("./output/Niche_Proteo.pdf")
ggsave("./output/Niche_Proteo.png", width = 20, height = 20)

OTUs_plot <- ggplot(subset(OTUs.90.REL, OTUs.90.REL$Phylum == "Verrucomicrobiota"), aes(x = Tau, y = REL))+
  geom_point(aes(color = Order))+
  ylab("Relative abundance")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  xlab(expression(paste(tau, " (h)")))+
  theme(legend.position = "none", axis.title = element_text(size = 20))+
  facet_wrap(~Class)
OTUs_plot

ggsave("./output/Niche_Verruco.pdf")
ggsave("./output/Niche_Verruco.png", width = 20, height = 20)

OTUs_plot <- ggplot(subset(OTUs.90.REL, OTUs.90.REL$Phylum == "Bacteroidota"), aes(x = Tau, y = REL))+
  geom_point(aes(color = Order))+
  ylab("Relative abundance")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  xlab(expression(paste(tau, " (h)")))+
  theme(legend.position = "none", axis.title = element_text(size = 20))+
  facet_wrap(~Class)
OTUs_plot

ggsave("./output/Niche_Bacter.pdf")
ggsave("./output/Niche_Bacter.png", width = 20, height = 20)

OTUs_plot <- ggplot(subset(OTUs.90.REL, OTUs.90.REL$Phylum == "Planctomycetota"), aes(x = Tau, y = REL))+
  geom_point(aes(color = Order))+
  ylab("Relative abundance")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  xlab(expression(paste(tau, " (h)")))+
  theme(legend.position = "none", axis.title = element_text(size = 20))+
  facet_wrap(~Class)
OTUs_plot

ggsave("./output/Niche_Plancto.pdf")
ggsave("./output/Niche_Plancto.png", width = 20, height = 20)

OTUs_plot <- ggplot(OTUs.90.REL, aes(x = Tau))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  xlab(expression(paste(tau, " (h)")))+
  geom_bar(aes(color = Phylum))

OTUs_plot
```


```{r}
Dim1_sites <- tau.plot$V1
Dim2_sites <- tau.plot$V2

DNA_corr <- as.data.frame(matrix(NA, dim(OTUsREL_20)[2], 4))
row.names(DNA_corr) <- colnames(OTUsREL_20)
colnames(DNA_corr)  <- c("V1_rho", "V1_p.value", "V2_rho", "V2_p.value")

for (i in 1:dim(OTUsREL_20)[2]){
  out.i <- cor.test(OTUsREL_20[,i], tau.plot$V1, method="spearman",
                    exact=FALSE)
  DNA_corr[i,1] <- out.i$estimate
  DNA_corr[i,2] <- out.i$p.value
  
  out.i <- cor.test(OTUsREL_20[,i], tau.plot$V2, method="spearman",
                    exact=FALSE)
  DNA_corr[i,3] <- out.i$estimate
  DNA_corr[i,4] <- out.i$p.value
}

DNA_corr     <- na.omit(DNA_corr)
DNA_corr$V1_BH <- p.adjust(DNA_corr$V1_p.value, method = "BH")
DNA_corr$V2_BH <- p.adjust(DNA_corr$V2_p.value, method = "BH")
DNA_corr_V2 <- DNA_corr[abs(DNA_corr$V2_rho) > 0.7, ]
DNA_corr_V1 <- DNA_corr[abs(DNA_corr$V1_rho) > 0.7, ]

V1.tax <- subset(OTU.tax, OTU == rownames(DNA_corr_V1)[1])
V2.tax <- subset(OTU.tax, OTU == rownames(DNA_corr_V2)[1])

x <- 2                        
while (x <= nrow(DNA_corr_V1)){
  V1.tax <- rbind(V1.tax, subset(OTU.tax, OTU == rownames(DNA_corr_V1)[x]))
  x <- x + 1
}
DNA_corr_V1 <- cbind(DNA_corr_V1, V1.tax)

x <- 2                        
while (x <= nrow(DNA_corr_V2)){
  V2.tax <- rbind(V2.tax, subset(OTU.tax, OTU == rownames(DNA_corr_V2)[x]))
  x <- x + 1
}

DNA_corr_V2 <- cbind(DNA_corr_V2, V2.tax)

DNA_corr_V2_neg <- subset(DNA_corr_V2, DNA_corr_V2$V2_rho < -0.7)
DNA_corr_V2_neg <- DNA_corr_V2_neg[,c("OTU","V2_rho", "Domain", "Phylum", "Class", "Order", "Family", "Genus")]
write.csv(DNA_corr_V2_neg, "./data/DNA_corr_V2_neg.csv")
DNA_corr_V2_pos <- subset(DNA_corr_V2, DNA_corr_V2$V2_rho > 0.7)
DNA_corr_V2_pos <- DNA_corr_V2_pos[,c("OTU","V2_rho", "Domain", "Phylum", "Class", "Order", "Family", "Genus")]
write.csv(DNA_corr_V2_pos, "./data/DNA_corr_V2_pos.csv")

DNA_corr_V1_neg <- subset(DNA_corr_V1, DNA_corr_V1$V1_rho < -0.7)
DNA_corr_V1_neg <- DNA_corr_V1_neg[,c("OTU","V1_rho", "Domain", "Phylum", "Class", "Order", "Family", "Genus")]
write.csv(DNA_corr_V1_neg, "./data/DNA_corr_V1_neg.csv")
DNA_corr_V1_pos <- subset(DNA_corr_V1, DNA_corr_V1$V1_rho > 0.7)
DNA_corr_V1_pos <- DNA_corr_V1_pos[,c("OTU","V1_rho", "Domain", "Phylum", "Class", "Order", "Family", "Genus")]
write.csv(DNA_corr_V1_pos, "./data/DNA_corr_V1_pos.csv")

DNA_V2_family_neg <- data.frame()
for(x in unique(DNA_corr_V2_neg$Family)){
  DNA_V2_family_neg <- rbind(DNA_V2_family_neg, c(x, mean(subset(DNA_corr_V2_neg, DNA_corr_V2_neg$Family == x)$V2_rho), nrow(subset(DNA_corr_V2_neg, DNA_corr_V2_neg$Family == x))))
}
colnames(DNA_V2_family_neg) <- c("Family", "rho", "n")
DNA_V2_family_neg$n <- as.numeric(DNA_V2_family_neg$n)
DNA_V2_family_neg$rho <- as.numeric(DNA_V2_family_neg$rho)

DNA_V2_family_pos <- data.frame()
for(x in unique(DNA_corr_V2_pos$Family)){
  DNA_V2_family_pos <- rbind(DNA_V2_family_pos, c(x, mean(subset(DNA_corr_V2_pos, DNA_corr_V2_pos$Family == x)$V2_rho), as.numeric(nrow(subset(DNA_corr_V2_pos, DNA_corr_V2_pos$Family == x)))))
}
colnames(DNA_V2_family_pos) <- c("Family", "rho", "n")
DNA_V2_family_pos$n <- as.numeric(DNA_V2_family_pos$n)
DNA_V2_family_pos$rho <- as.numeric(DNA_V2_family_pos$rho)

DNA_V1_family_neg <- data.frame()
for(x in unique(DNA_corr_V1_neg$Family)){
  DNA_V1_family_neg <- rbind(DNA_V1_family_neg, c(x, mean(subset(DNA_corr_V1_neg, DNA_corr_V1_neg$Family == x)$V1_rho), nrow(subset(DNA_corr_V1_neg, DNA_corr_V1_neg$Family == x))))
}
colnames(DNA_V1_family_neg) <- c("Family", "rho", "n")
DNA_V1_family_neg$n <- as.numeric(DNA_V1_family_neg$n)
DNA_V1_family_neg$rho <- as.numeric(DNA_V1_family_neg$rho)

DNA_V1_family_pos <- data.frame()
for(x in unique(DNA_corr_V1_pos$Family)){
  DNA_V1_family_pos <- rbind(DNA_V1_family_pos, c(x, mean(subset(DNA_corr_V1_pos, DNA_corr_V1_pos$Family == x)$V1_rho), nrow(subset(DNA_corr_V1_pos, DNA_corr_V1_pos$Family == x))))
}
colnames(DNA_V1_family_pos) <- c("Family", "rho", "n")
DNA_V1_family_pos$n <- as.numeric(DNA_V1_family_pos$n)
DNA_V1_family_pos$rho <- as.numeric(DNA_V1_family_pos$rho)
```

```{r}
# taxa <- as.data.frame(imp.otus$Tau)
# for(x in unique(OTU.tax.20$Phylum)){
#   list <- data.frame(rep(0, 35))
#   colnames(list) <- c(as.character(x))
#   for(y in OTU.tax.20$OTU){
#     if(subset(OTU.tax.20, OTU.tax.20$OTU == y)$Phylum == x){
#       list[,x] <- list[,x] + as.data.frame(OTUsr_20[,y])
#     }
#   }
#   taxa[,as.character(x)] <- list[,x]
# }
# rownames(taxa) <- taxa$`imp.otus$Tau`
# taxa.20 <- taxa[,-1]
# write.csv(taxa.20, "./data/taxa.20.csv")

taxa.20 <- read.csv("./data/taxa.20.csv", header = TRUE)
rownames(taxa.20) <- taxa.20$X
taxa.20 <- taxa.20[,-1]
taxa.20.rel <- decostand(taxa.20, method = "total")
taxa.20.rel <- cbind(imp.otus$Tau, taxa.20.rel)
colnames(taxa.20.rel) <- c("Tau", unique(OTU.tax.20$Phylum))
taxa.20.long <- gather(taxa.20.rel, phylum, REL, Proteobacteria:Modulibacteria)

phylum_plot <- ggplot(taxa.20.long, aes(x = Tau, y = REL, group = phylum))+
  geom_smooth(method = "loess", se = F, aes(color = phylum, group = phylum))+
  ylab("Relative abundance")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  labs(color = "Phylum")+
  xlab(expression(paste(tau, " (h)")))+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 15), axis.title = element_text(size = 20))+
  guides(color = guide_legend(ncol = 3))
phylum_plot

ggsave("./output/RTLC_rel_phylum.pdf")
ggsave("./output/RTLC_rel_phylum.png", width = 12, height = 8)
```

```{r}
anova(betadisper(tau.db, group= Tau_20$Turn))
permutest(betadisper(tau.db, group= Tau_20$Turn))

Dim1 <- c()
Dim2 <- c()
Dim3 <- c()
for(x in unique(imp.spp$Phylum)){
  Dim1 <- c(Dim1, mean(as.numeric(subset(imp.spp, imp.spp$Phylum == x)$Dim1)))
  Dim2 <- c(Dim2, mean(as.numeric(subset(imp.spp, imp.spp$Phylum == x)$Dim2)))
  Dim3 <- c(Dim3, mean(as.numeric(subset(imp.spp, imp.spp$Phylum == x)$Dim3)))
}

spe.corr.means <- as.data.frame(unique(imp.spp$Phylum))
spe.corr.means <- cbind(spe.corr.means, as.vector(Dim1), as.vector(Dim2), as.vector(Dim3))
colnames(spe.corr.means) <- c("Phylum", "Dim1", "Dim2", "Dim3")

tau.plot.order <- tau.plot[order(as.numeric(tau.plot$Tau)),]
OTUsr_20.order <- OTUsr_20[order(as.numeric(row.names(OTUsr_20))),]

PCoA <- ggplot(tau.plot.order, aes(x = V1, y = V2))+
  geom_path(linewidth = 1.25, color = alpha("black", 0.2))+
  geom_point(cex = 3, aes(shape = Turn, color = as.numeric(Tau)))+
  xlab(paste("PCoA 1 (", explainvar1, "%)", sep = ""))+
  ylab(paste("PCoA 2 (", explainvar2, "%)", sep = ""))+
  labs(shape = "Full Turnover", color = expression(paste(tau,  " (h)")))+
  scale_shape_manual(values = c(15, 19), labels = c("Yes", "No"))+
  scale_color_viridis(direction = -1, labels = label_math(expr = 10^.x, format = force))+
  theme(legend.title = element_text(size = 14))

PCoA

ggsave("./output/RTLC_BC_PCoA.pdf")
ggsave("./output/RTLC_BC_PCoA.png", width = 8, height = 5)

sat <- as.data.frame(Tau_20[-1, "Tau"])
sat$BC <- rep(NA, 34)
row = 1
while(row <= nrow(sat)){
  print(tau.db[row])
  sat[(row), "BC"] <- tau.db[row]
  row = row + 1
}
sat$Turn <- Tau_20[-1, "Turn"]
colnames(sat) <- c("Tau", "BC")
plot(data = sat, BC ~ Tau)
```

```{r}
OTU_pumpturn <- ggplot(tau.plot.order, aes(x = V1, y = V2))+
  geom_path(linewidth = 1.25, color = alpha("black", 0.2))+
  xlab(paste("PCoA 1 (", explainvar1, "%)", sep = ""))+
  ylab(paste("PCoA 2 (", explainvar2, "%)", sep = ""))+
  geom_point(cex = 3, aes(shape = Turn, color = Pump))+
  labs(shape = "Number of\nturnovers", color = "Treatment\nmethod")+
  scale_color_manual(values = c("blue", "black"), labels = c("Pump", "Manual"))+
  scale_shape_manual(values = c(19, 1), labels = c("> 1", "< 1"))+
  theme(legend.title = element_text(size = 14))

OTU_pumpturn

ggsave("./output/RTLC_BC_OTUpumpturn.pdf")
ggsave("./output/RTLC_BC_OTUpumpturn.png", width = 8, height = 5)
```

```{r}
PCoA1_gam <- gam(V1 ~ s(Tau), family = gaussian(link = "identity"), data = tau.plot, method = "REML")
PCoA1_gam_re <- gam(V1 ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = tau.plot, method = "REML")

anova(PCoA1_gam, PCoA1_gam_re, test = "Chisq")

AIC(PCoA1_gam, PCoA1_gam_re)
summary(PCoA1_gam_re)
mean(summary(PCoA1_gam_re)$s.table[,4])

PCoA1 <- predict_gam(PCoA1_gam_re, exclude_terms = "s(Set)") %>%
  filter(Set == "1") %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_point(data = tau.plot, aes(x = Tau, y = V1))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab(paste("PCoA 1 (", explainvar1, "%)", sep = ""))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(PCoA1_gam_re)$dev, 2))))+
  theme(axis.title.x = element_blank())

PCoA1

ggsave("./output/RTLC_BC_PCoA1.pdf")
ggsave("./output/RTLC_BC_PCoA1.png", width = 6.5, height = 5)

PCoA2_gam <- gam(V2 ~ s(Tau), family = gaussian(link = "identity"), data = tau.plot, method = "REML")
PCoA2_gam_re <- gam(V2 ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = tau.plot, method = "REML")
summary(PCoA2_gam)
mean(summary(PCoA2_gam)$s.table[,4])
AIC(PCoA2_gam_re, PCoA2_gam)
anova(PCoA2_gam, PCoA2_gam_re, test = "Chisq")

PCoA2 <- predict_gam(PCoA2_gam) %>%
  ggplot(aes(Tau, fit))+
#  geom_vline(xintercept = turnover, linetype = "dashed", color = "blue", size = 1)+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_point(data = tau.plot, aes(x = Tau, y = V2))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  xlab(expression(paste(tau, " (h)")))+
  ylab(paste("PCoA 2 (", explainvar2, "%)", sep = ""))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(PCoA2_gam)$dev, 2))))+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.5), "cm"))

PCoA2

ggsave("./output/RTLC_BC_PCoA2.pdf")
ggsave("./output/RTLC_BC_PCoA2.png", width = 6.5, height = 5)
```

##Draw Figure
```{r}
ggdraw()+
  draw_plot(PCoA, x = 0, y = 0, width = 1, height = 01) + 
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("./output/CommPCoA.pdf")
ggsave("./output/CommPCoA.png", width = 8, height = 10)

ggdraw()+
  draw_plot(PCoA1, x = 0, y = 0.55, width = 1, height = 0.45) +
  draw_plot(PCoA2, x = 0, y = 0, width = 1, height = 0.55) + 
  theme(plot.background = element_rect(fill="white", color = NA))+
  draw_plot_label(label = c("A", "B"), size = 30, x = c(0,0), y = c(0.98,0.48))

ggsave("./output/CommPCoA_axes.pdf")
ggsave("./output/CommPCoA_axes.png", width = 8, height = 10)
```


#Figure 4. Community Function - P

##Cstat - Biomass Production (uM C/hr)

```{r}
# BP_lm <- lm(uMChr ~ Tau, data = Tau)
# BP_poly <- lm(uMChr ~ poly(Tau, 2, raw = TRUE), data = Tau)
# BP_gam <- gam(uMChr ~ s(Tau), family = gaussian(link = "identity"), data = Tau, method = "REML")
# BP_gam_gm <- gam(uMChr ~ s(Tau), family = Gamma(link = "log"), data = Tau, method = "REML")
# 
# AIC(BP_lm, BP_poly, BP_gam, BP_gam_gm)

BP_gam_re <- gam(log(uMChr, 10) ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")
BP_gam <- gam(log(uMChr, 10) ~ s(Tau), family = gaussian(link = "identity"), data = Tau, method = "REML")
anova(BP_gam, BP_gam_re, test = "Chisq")

summary(BP_gam_re)
mean(summary(BP_gam_re)$s.table[,4])
k.check(BP_gam_re)
gam.check(BP_gam_re)

BP_Tau <- predict_gam(BP_gam_re, exclude_terms = "s(Set)") %>%
  filter(Set == "1") %>% 
  ggplot(aes(Tau, fit))+
#  geom_vline(xintercept = turnover, linetype = "dashed", color = "blue", size = 1)+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_point(data = Tau, aes(x = Tau, y = log(uMChr, 10)))+
#  xlab(expression(paste(tau, " (h)")))+
  ylab(expression(paste("Bacterial productivity (", mu, "mol C/hr)")))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(limits = c(-3, -1), breaks = c(-1, -2, -3), labels = label_math(expr = 10^.x, format = force))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(BP_gam_re)$dev, 2))))+
  theme(axis.title.x = element_blank())

BP_Tau

ggsave("./output/RTLC_BP.pdf")
ggsave("./output/RTLC_BP.png", width = 6.5, height = 5)
```

##Average well response at 48 hours
```{r}

cor.test(Tau_20$Tau, tau.plot$NumRes, method="spearman", exact=FALSE)

Avg_lm <- lm(AvgRes ~ Tau, data = Tau)
Avg_gam <- gam(AvgRes ~ s(Tau), data = Tau, family = gaussian(link = "identity"), method = "REML")
Avg_gam_re <- gam(AvgRes ~ s(Tau) + s(Set, bs = "re"), data = Tau, family = gaussian(link = "identity"), method = "REML")

anova(Avg_gam, Avg_gam_re, test = "Chisq")
AIC(Avg_lm, Avg_gam, Avg_gam_re)

summary(Avg_gam_re)
mean(summary(Avg_gam_re)$s.table[,4])

AvgRes <- predict_gam(Avg_gam_re, exclude_terms = "s(Set)") %>%
  filter(Set == "1") %>% 
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_point(data = Tau, aes(x = Tau, y = AvgRes))+
  xlab(expression(paste(tau, " (h)")))+
  ylab("Average resource use (OD590)")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(Avg_gam_re)$dev, 2))))+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.5), "cm"))

AvgRes

ggsave("./output/RTLC_AvgRes.pdf")
ggsave("./output/RTLC_AvgRes.png", width = 6.5, height = 5)

```

```{r}
N_RSG <- read.csv("./data/RTLC/2021_RTLC_RSG_N.csv", header = TRUE)
N_RSG$DorPer <- (N_RSG$N-N_RSG$RSG)/N_RSG$N
N_RSG$Tau <- log(((10^N_RSG$Tau)/60), 10)

Dorm_gam <- gam(data = N_RSG, (DorPer * 100) ~ s(Tau), family = gaussian(link = "identity"), method = "REML")
Dorm_gam_re <- gam(data = N_RSG, (DorPer * 100) ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), method = "REML")
summary(Dorm_gam)
AIC(Dorm_gam, Dorm_gam_re)
plot(Dorm_gam)
k.check(Dorm_gam)
gam.check(Dorm_gam)

Dorm_Tau <- predict_gam(Dorm_gam) %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.4)+
  geom_point(data = N_RSG, aes(x = Tau, y = (DorPer * 100)))+
  xlab(expression(paste(tau, " (h)")))+
  ylab("% Dormancy")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(Dorm_gam)$dev, 2))))

Dorm_Tau


ggsave("./output/RTLC_Dorm.pdf")
ggsave("./output/RTLC_Dorm.png")
```

#Draw figure
```{r}
ggdraw()+
  draw_plot(BP_Tau, x = 0, y = 0.55, width = 1, height = 0.45) + 
  draw_plot(AvgRes, x = 0, y = 0, width = 1, height = 0.55) + 
  draw_plot_label(label = c("A", "B"), size = 30, x = c(0.02,0.02), y = c(1,0.55))+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("./output/Comm_Func.pdf")
ggsave("./output/Comm_Func.png", width = 8, height = 10)
```

```{r}
print(colnames(EcoPlate_20))

EcoPlate_20$Amine <- EcoPlate_20$Phenylethylamine + EcoPlate_20$Putrescine

EcoPlate_20$AA <- EcoPlate_20$L.Asparagine + EcoPlate_20$L.Arginine + EcoPlate_20$Glycyl.L.Glutamic.Acid + EcoPlate_20$L.Phenylalanine + EcoPlate_20$L.Serine + EcoPlate_20$L.Threonine

EcoPlate_20$Carb <- EcoPlate_20$alpha.D.Lactose + EcoPlate_20$beta.Methyl.D.Glucoside + EcoPlate_20$D.Cellulobiose + EcoPlate_20$D.Mannitol + EcoPlate_20$D.Xylose + EcoPlate_20$i.Erythritol + EcoPlate_20$N.Acetyl.D.Glucosamine

EcoPlate_20$Carboxy <- EcoPlate_20$gamma.Hydroxybutyric.Acid + EcoPlate_20$alpha.Ketobutyric.Acid + EcoPlate_20$X2.Hydroxy.Benzoic.Acid + EcoPlate_20$X4.Hydroxy.Benzoic.Acid + EcoPlate_20$D.Galactonic.Acid.gamma.Lactone + EcoPlate_20$D.Galacturonic.Acid + EcoPlate_20$D.Glucosaminic.Acid + EcoPlate_20$D.Malic.Acid + EcoPlate_20$Itaconic.Acid

EcoPlate_20$Phos <- EcoPlate_20$D.L.alpha.Glycerol.Phosphate + EcoPlate_20$Glucose.1.Phosphate

EcoPlate_20$Ester <- EcoPlate_20$Pyruvic.Acid.Methyl.Ester

EcoPlate_20$Poly <- EcoPlate_20$alpha.Cyclodextrin + EcoPlate_20$Glycogen + EcoPlate_20$Tween.40 + EcoPlate_20$Tween.80

EcoPlate_20_group <- cbind(rownames(EcoPlate_20), EcoPlate_20[,c("Amine", "AA", "Carb", "Carboxy", "Ester", "Phos", "Poly")])

EcoPlate_20_group <- gather(EcoPlate_20_group, Group, OD, Amine:Poly)

colnames(EcoPlate_20_group) <- c("Tau", "Group", "OD")


ggplot(data = EcoPlate_20_group, aes(x = as.numeric(Tau), y = OD, group = Group, color = Group))+
  geom_point()+
  geom_smooth(method = "lm")+
  stat_poly_eq(aes(label = paste(c("`Amino Acids`", "`Amines`", "`Carbohydrates`", "`Carboxylic Acid`", "`Ester`", "`Phosphorylated`", "`Polymers`")[stat(group)], stat(rr.label), stat(p.value.label), sep = "~\", \"~")),
              formula = y~x, parse = TRUE, size = 6, p.digits = 4, label.x = 0.95)+
  xlab(expression(paste(tau, " (h)")))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab("Carbon use at 48 hrs (OD 590)")+
  theme(legend.position = "none")
```

#Figure 5. Significant resource use
```{r}
csource_sig <- data.frame()
for(c in unique(long_EP_slope$C_Source)){
  csource.lm <- lm(data = subset(long_EP_slope, long_EP_slope$C_Source == c), Slope ~ Tau)
  csource_sig <- rbind(csource_sig, c(c, summary(csource.lm)$coefficients[2,4]))
  assign(paste("lm.", gsub("-", "_", c), sep = ""), csource.lm)
}
colnames(csource_sig) <- c("Resource", "P")
csource_sig$P_BH <- p.adjust(csource_sig$P, "BH")

#Glucose-1-Phosphate was non-linear
sig_csources <- subset(csource_sig, csource_sig$P_BH < 0.05 & csource_sig$Resource != "Glucose-1-Phosphate")$Resource
summary(lm.2_Hydroxy.Benzoic.Acid)


ep.mods <- as_tibble(rbind.data.frame(
  tidy(lm.4_Hydroxy.Benzoic.Acid) %>% add_column(Resource = "4-Hydroxy \nBenzoic Acid"),
  tidy(lm.D_Galactonic.Acid.gamma_Lactone) %>% add_column(Resource = "D-Galactonic Acid γ-Lactone"),
  tidy(lm.D_Galacturonic.Acid) %>% add_column(Resource = "D-Galacturonic Acid"),
  tidy(lm.D_Glucosaminic.Acid) %>% add_column(Resource = "D-Glucosaminic Acid"),
  tidy(lm.D_Mannitol) %>% add_column(Resource = "D-Mannitol"),
  tidy(lm.D_Xylose) %>% add_column(Resource = "D-Xylose"),
  tidy(lm.gamma_Hydroxybutyric.Acid) %>% add_column(Resource = "γ-Hydroxy Butyric Acid"),
  tidy(lm.Glycyl_L_Glutamic.Acid) %>% add_column(Resource = "Glycyl-L-Glutamic Acid"),
  tidy(lm.L_Arginine) %>% add_column(Resource = "L-Arginine"),
  tidy(lm.L_Asparagine) %>% add_column(Resource = "L-Asparagine"),
  tidy(lm.L_Serine) %>% add_column(Resource = "L-Serine"),
  tidy(lm.L_Threonine) %>% add_column(Resource = "L-Threonine"),
  tidy(lm.Putrescine) %>% add_column(Resource = "Putrescine"),
  tidy(lm.Pyruvic.Acid.Methyl.Ester) %>% add_column(Resource = "Pyruvic Acid \nMethyl Ester"),
  tidy(lm.Tween.40) %>% add_column(Resource = "Tween 40"),
  tidy(lm.Tween.80) %>% add_column(Resource = "Tween 80")
))

ep.mods %>%
  group_by(Resource)%>%
  rename( "Term" = term,
          "Estimate" = estimate,
          "Std. Error" = std.error,
          "Statistic" = statistic,
          "p-value" = p.value) %>%
  select(Resource, everything()) %>%
  pander(round = 4)

Csource_sig <- ggplot(data = long_EP_slope[long_EP_slope$C_Source %in% sig_csources,], aes(x = Tau, y = Slope, group = C_Source, color = C_Source))+
  geom_smooth(method = "lm", formula = y ~ x, cex = 1.25, fill = "#333333", alpha = 0.35)+
  geom_point()+
  stat_poly_eq(aes(label = paste(c("`4-Hydroxy Benzoic Acid`", "`D-Galactonic Acid γ-Lactone`", "`D-Galacturonic Acid`", "`D-Glucosaminic Acid`", "`D-Mannitol`", "`D-Xylose`", "`γ-Hydroxy Butyric Acid`", "`Glycyl-L-Glutamic Acid`", "`L-Arginine`", "`L-Asparagine`", "`L-Serine`", "`L-Threonine`", "`Putrescine`", "`Pyruvic Acid Methyl Ester`", "`Tween 40`", "`Tween 80`")[stat(group)], stat(rr.label), sep = "~\", \"~")),
              formula = y~x, parse = TRUE, size = 6, p.digits = 4, label.x = 0.95, label.y = c(0.95, 0.89, 0.71, 0.77, 0.89, 0.95, 0.83, 0.95, 0.77, 0.89, 0.71, 0.83, 0.95, 0.95, 0.95, 0.89))+
  xlab(expression(paste(tau, " (h)")))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab("Carbon source use per day (OD590)")+
  labs(color = "Carbon Source")+
  theme(axis.title = element_text(size = 25), axis.text =element_text(size = 20))+
  theme(legend.position="none", strip.text = element_text(size=25), axis.line = element_line())+
  facet_wrap(~Type, ncol = 2, nrow = 3, scales = 'free')
  
Csource_sig

ggsave("./output/RTLC_Res_Csource_sig.pdf", width = 15, height =20)
ggsave("./output/RTLC_Res_Csource_sig.png", width = 15, height = 20)
  
```

#Figure 5. Significant resource use
```{r}
long_EP_subset <- subset(long_EP, long_EP$Hours == 48 & long_EP$C_Source != "Avg" & long_EP$C_Source != "NumRes")
long_EP_subset$Type <- long_EP_slope$Type

csource_sig <- data.frame()
for(c in unique(long_EP_subset$C_Source)){
  csource.lm <- lm(data = subset(long_EP_subset, long_EP_subset$C_Source == c), OD ~ Tau)
  csource_sig <- rbind(csource_sig, c(c, summary(csource.lm)$coefficients[2,4]))
  assign(paste("lm.", gsub("-", "_", c), sep = ""), csource.lm)
}
colnames(csource_sig) <- c("Resource", "P")
csource_sig$P_BH <- p.adjust(csource_sig$P, "BH")

#Glucose-1-Phosphate was non-linear
sig_csources <- subset(csource_sig, csource_sig$P_BH < 0.05)$Resource
summary(lm.2_Hydroxy.Benzoic.Acid)


ep.mods <- as_tibble(rbind.data.frame(
  tidy(lm.4_Hydroxy.Benzoic.Acid) %>% add_column(Resource = "4-Hydroxy \nBenzoic Acid"),
  tidy(lm.D_Galactonic.Acid.gamma_Lactone) %>% add_column(Resource = "D-Galactonic Acid γ-Lactone"),
  tidy(lm.D_Galacturonic.Acid) %>% add_column(Resource = "D-Galacturonic Acid"),
  tidy(lm.D_Glucosaminic.Acid) %>% add_column(Resource = "D-Glucosaminic Acid"),
  tidy(lm.D_Mannitol) %>% add_column(Resource = "D-Mannitol"),
  tidy(lm.D_Xylose) %>% add_column(Resource = "D-Xylose"),
  tidy(lm.gamma_Hydroxybutyric.Acid) %>% add_column(Resource = "γ-Hydroxy Butyric Acid"),
  tidy(lm.Glycyl_L_Glutamic.Acid) %>% add_column(Resource = "Glycyl-L-Glutamic Acid"),
  tidy(lm.L_Arginine) %>% add_column(Resource = "L-Arginine"),
  tidy(lm.L_Asparagine) %>% add_column(Resource = "L-Asparagine"),
  tidy(lm.L_Serine) %>% add_column(Resource = "L-Serine"),
  tidy(lm.L_Threonine) %>% add_column(Resource = "L-Threonine"),
  tidy(lm.Putrescine) %>% add_column(Resource = "Putrescine"),
  tidy(lm.Pyruvic.Acid.Methyl.Ester) %>% add_column(Resource = "Pyruvic Acid \nMethyl Ester"),
  tidy(lm.Tween.40) %>% add_column(Resource = "Tween 40"),
  tidy(lm.Tween.80) %>% add_column(Resource = "Tween 80")
))

ep.mods %>%
  group_by(Resource)%>%
  rename( "Term" = term,
          "Estimate" = estimate,
          "Std. Error" = std.error,
          "Statistic" = statistic,
          "p-value" = p.value) %>%
  select(Resource, everything()) %>%
  pander(round = 4)

Csource_sig <- ggplot(data = long_EP_subset[long_EP_subset$C_Source %in% sig_csources,], aes(x = Tau, y = OD, group = C_Source, color = C_Source))+
  geom_smooth(method = "lm", formula = y ~ x, cex = 1.25, fill = "#333333", alpha = 0.35)+
  geom_point()+
  stat_poly_eq(aes(label = paste(c("`4-Hydroxy Benzoic Acid`", "`D-Galactonic Acid γ-Lactone`", "`D-Galacturonic Acid`", "`D-Glucosaminic Acid`", "`D-Mannitol`", "`D-Xylose`", "`γ-Hydroxy Butyric Acid`", "`Glucose-1-Phosphate`", "`Glycyl-L-Glutamic Acid`", "`L-Arginine`", "`L-Asparagine`", "`L-Serine`", "`L-Threonine`", "`Putrescine`", "`Pyruvic Acid Methyl Ester`", "`Tween 40`", "`Tween 80`")[stat(group)], stat(rr.label), sep = "~\", \"~")),
              formula = y~x, parse = TRUE, size = 6, p.digits = 4, label.x = 0.95, label.y = c(0.95, 0.89, 0.71, 0.77, 0.89, 0.95, 0.83, 0.5,  0.95, 0.77, 0.89, 0.71, 0.83, 0.95, 0.95, 0.95, 0.89))+
  xlab(expression(paste(tau, " (h)")))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab("Carbon source use per day (OD590)")+
  labs(color = "Carbon Source")+
  theme(axis.title = element_text(size = 25), axis.text =element_text(size = 20))+
  theme(legend.position="none", strip.text = element_text(size=25), axis.line = element_line())+
  facet_wrap(~Type, scales = 'free')
  
Csource_sig

ggsave("./output/RTLC_Res_Csource_sig.pdf", width = 15, height =20)
ggsave("./output/RTLC_Res_Csource_sig.png", width = 15, height = 20)
  
```

#Figure 6. Community Function - Resource Use
#Bray-Curtis distance Resource use
```{r}
EcoPlate_20 <- EcoPlate[rownames(OTUsr_20),]
EP.db.20 <- vegdist(EcoPlate_20, method = "bray")
EP.db <-vegdist(EcoPlate, method = "bray")
EP.pcoa <- cmdscale(EP.db, eig = TRUE)

EPREL <- EcoPlate
for(i in 1:nrow(EcoPlate)){
  EPREL[i,] = EcoPlate[i,]/sum(EcoPlate[i,])
}

EP.pcoa <- add.spec.scores(EP.pcoa, EPREL, method = "pcoa.scores")

explainvar1.ep <- round(EP.pcoa$eig[1] / sum(EP.pcoa$eig), 3) * 100
explainvar2.ep <- round(EP.pcoa$eig[2] / sum(EP.pcoa$eig), 3) * 100
explainvar3.ep <- round(EP.pcoa$eig[3] / sum(EP.pcoa$eig), 3) * 100

types <- read.csv("data/RTLC/EcoPlate/Csourcetypes.csv", header = TRUE, sep = ",")

EP.plot <- as.data.frame(EP.pcoa$points)
EP.plot <- cbind(EP.plot, Tau$Tau, Tau$Turn, Tau$Pump, Tau$Set)
colnames(EP.plot) <- c("V1", "V2", "Tau", "Turn", "Pump", "Set")
EP.plot$Turn <- factor(EP.plot$Turn, levels = c("TRUE", "FALSE"))
EP.plot$Pump <- factor(EP.plot$Pump, levels = c("TRUE", "FALSE"))

EP.cproj <- as.data.frame(EP.pcoa$cproj)
EP.cproj <- cbind(EP.cproj, as.data.frame(row.names(EP.pcoa$cproj)))
spe.corr.ep <- add.spec.scores(EP.pcoa, EPREL, method = "cor.scores")$cproj
spe.corr.ep <- as.data.frame(cbind(spe.corr.ep,types$Type))
corrcut.ep <- 0.7
imp.spp.ep <- as.data.frame(spe.corr.ep[abs(as.numeric(spe.corr.ep[,1])) >= corrcut.ep | abs(as.numeric(spe.corr.ep[,2])) >= corrcut.ep,])

Dim1.ep <- c()
Dim2.ep <- c()
for(x in unique(spe.corr.ep$V3)){
  Dim1.ep <- c(Dim1.ep, mean(as.numeric(subset(spe.corr.ep, spe.corr.ep$V3 == x)$Dim1)))
  Dim2.ep <- c(Dim2.ep, mean(as.numeric(subset(spe.corr.ep, spe.corr.ep$V3 == x)$Dim2)))
}

spe.corr.ep.means <- as.data.frame(unique(spe.corr.ep$V3))
spe.corr.ep.means <- cbind(spe.corr.ep.means, as.vector(Dim1.ep), as.vector(Dim2.ep))
colnames(spe.corr.ep.means) <- c("C_types", "Dim1", "Dim2")

fit <- envfit(EP.pcoa, EPREL, perm = 999)
fit

EP.plot$Tau <- as.numeric(EP.plot$Tau)
typeof(EP.plot$Tau)

EP_BC <- ggplot(EP.plot, aes(x = V1, y = V2, color = Turn))+
  geom_point(cex = 2)+
  xlab(paste("PCoA 1 (", explainvar1.ep, "%)", sep = ""))+
  ylab(paste("PCoA 2 (", explainvar2.ep, "%)", sep = ""))+
  xlim(c(-0.5, 0.3))+
  labs(color = "Number of turnovers")+
  theme(legend.position = "top", legend.title = element_text(size = 14))+
  scale_color_manual(values = c(alpha("blue", 0.3), alpha("black", 0.3)), labels = c("> 1", "< 1"))+
  geom_text(size = 5, data = spe.corr.ep.means, aes(x = Dim1, y = Dim2, label = C_types), color = "black")

EP_BC

ggsave("./output/RTLC_Res_BC_PCoA.pdf")
ggsave("./output/RTLC_Res_BC_PCoA.png", width = 8, height = 5)

EP_BC_turn <- ggplot(EP.plot, aes(x = V1, y = V2))+
  geom_point(cex = 3, aes(color = Pump, shape = Turn))+
  xlab(paste("PCoA 1 (", explainvar1.ep, "%)", sep = ""))+
  ylab(paste("PCoA 2 (", explainvar2.ep, "%)", sep = ""))+
  labs(shape = "Number of\nturnovers", color = "Treatment\nmethod")+
  scale_color_manual(values = c("blue", "black"), labels = c("Pump", "Manual"))+
  scale_shape_manual(values = c(19, 1), labels = c("> 1", "< 1"))+
  theme(legend.title = element_text(size = 14))

EP_BC_turn

ggsave("./output/RTLC_Res_BC_PCoA_turn.pdf")
ggsave("./output/RTLC_Res_BC_PCoA_turn.png", width = 8, height = 5)

EcoPlate_env$Set <- c(rep(1, 13), rep(2,12), rep(3,10), rep(4,14))

EP.dist <- vegdist(EcoPlate, method = "bray")
Env.dist <- vegdist(scale(EcoPlate_env[,-2]), method = "euclid")

mantel(EP.dist, Env.dist)

Res_PCoA1_lm <- lm(V1 ~ Tau, data = EP.plot)
Res_PCoA1_gam <- gam(data = EP.plot, V1 ~ s(Tau), family = gaussian(link = "identity"), method = "REML")
Res_PCoA1_gam_re <- gam(data = EP.plot, V1 ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), method = "REML")

anova(Res_PCoA1_gam, Res_PCoA1_gam_re, test = "Chisq")
summary(Res_PCoA1_gam)
mean(summary(Res_PCoA1_gam)$s.table[,4])

AIC(Res_PCoA1_lm, Res_PCoA1_gam, Res_PCoA1_gam_re)


Res_PCoA1 <- predict_gam(Res_PCoA1_gam) %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
#  geom_vline(xintercept = turnover, linetype = "dashed", color = "blue", size = 1)+
  geom_point(data = EP.plot, aes(x = Tau, y = V1))+
  ylab(paste("PCoA 1 (", explainvar1.ep, "%)\n", sep = ""))+
#  labs(color = "Number of turnovers")+
#  scale_color_manual(values = c(alpha("blue", 0.3), alpha("black", 0.3)), labels = c("> 1", "< 1"))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(Res_PCoA1_gam)$dev, 2))))+
  theme(axis.title.x = element_blank())

Res_PCoA1

ggsave("./output/RTLC_Res_BC_PCoA1.pdf")
ggsave("./output/RTLC_Res_BC_PCoA1.png", width = 6.5, height = 5)

Res_PCoA2_gam <- gam(data = EP.plot, V2 ~ s(Tau), family = gaussian(link = "identity"), method = "REML")
Res_PCoA2_gam_re <- gam(data = EP.plot, V2 ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), method = "REML")
Res_PCoA2_lm <- lm(data = EP.plot, V2 ~ Tau)
anova(Res_PCoA2_gam, Res_PCoA2_gam_re, test = "Chisq")
summary(Res_PCoA2_gam_re)

mean(summary(Res_PCoA2_gam_re)$s.table[,4])

AIC(Res_PCoA2_gam, Res_PCoA2_gam_re, Res_PCoA2_lm)

Res_PCoA2 <- predict_gam(Res_PCoA2_gam_re, exclude_terms = "s(Set)") %>%
  filter(Set == "1") %>% 
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
#  geom_vline(xintercept = turnover, linetype = "dashed", color = "blue", size = 1)+
  geom_point(data = EP.plot, aes(x = Tau, y = V2))+
  ylab(paste("PCoA 2 (", explainvar2.ep, "%)\n", sep = ""))+
  xlab(expression(paste(tau, " (h)")))+
#  labs(color = "Number of turnovers")+
#  scale_color_manual(values = c(alpha("blue", 0.3), alpha("black", 0.3)), labels = c("> 1", "< 1"))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(Res_PCoA2_gam_re)$dev, 2))))

Res_PCoA2

ggsave("./output/RTLC_Res_BC_PCoA2.pdf")
ggsave("./output/RTLC_Res_BC_PCoA2.png", width = 6.5, height = 5)
```



```{r}
Dim1_sites <- EP.plot$V1
Dim2_sites <- EP.plot$V2

EP_corr <- as.data.frame(matrix(NA, dim(EPREL)[2], 4))
row.names(EP_corr) <- colnames(EPREL)
colnames(EP_corr)  <- c("V1_rho", "V1_p.value", "V2_rho", "V2_p.value")

for (i in 1:dim(EPREL)[2]){
  out.i <- cor.test(EPREL[,i], EP.plot$V1, method="spearman",
                    exact=FALSE)
  EP_corr[i,1] <- out.i$estimate
  EP_corr[i,2] <- out.i$p.value
  
  out.i <- cor.test(EPREL[,i], EP.plot$V2, method="spearman",
                    exact=FALSE)
  EP_corr[i,3] <- out.i$estimate
  EP_corr[i,4] <- out.i$p.value
}

EP_corr     <- na.omit(EP_corr)
EP_corr$V1_BH <- p.adjust(EP_corr$V1_p.value, method = "BH")
EP_corr$V2_BH <- p.adjust(EP_corr$V2_p.value, method = "BH")
EP_corr_V2 <- EP_corr[abs(EP_corr$V2_rho) > 0.5, ]
EP_corr_V1 <- EP_corr[abs(EP_corr$V1_rho) > 0.5, ]
```


##Draw Figure
```{r}
ggdraw()+
  draw_plot(EP_BC, x = 0, y = 0, width = 1, height = 1) + 
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("./output/CommRes_PCoA.pdf")
ggsave("./output/CommRes_PCoA.png", width = 5, height = 5)

ggdraw()+
  draw_plot(Res_PCoA1, x = 0, y = 0.55, width = 1, height = 0.45) + 
  draw_plot(Res_PCoA2, x = 0, y = 0, width = 1, height = 0.55) + 
  draw_plot_label(label = c("A", "B"), size = 30, x = c(0.02,0.02), y = c(1,0.55))+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("./output/CommRes_PCoA_Axes.pdf")
ggsave("./output/CommRes_PCoA_Axes.png", width = 8, height = 10)
```


#Figure 7. IBMs

```{r}
IBM_IC <- read.csv("../residence-time/Model/results/data/SimData.csv", header = TRUE, sep = ",")
IBM_IC_long <- as.data.frame(unique(IBM_IC$sim))
colnames(IBM_IC_long) <- c("sim")
IBM_IC_long$ct <- rep(1000, 50)

for(x in unique(IBM_IC_long$sim)){
  IBM_IC_long[IBM_IC_long$sim == x, "V"] <- mean(subset(IBM_IC, IBM_IC$sim == x)$V)
  IBM_IC_long[IBM_IC_long$sim == x, "Q"] <- mean(subset(IBM_IC, IBM_IC$sim == x)$Q)
  IBM_IC_long[IBM_IC_long$sim == x, "N"] <- mean(subset(IBM_IC, IBM_IC$sim == x)$total.abundance)
  IBM_IC_long[IBM_IC_long$sim == x, "R"] <- mean(subset(IBM_IC, IBM_IC$sim == x)$resource.particles)
  IBM_IC_long[IBM_IC_long$sim == x, "P"] <- mean(subset(IBM_IC, IBM_IC$sim == x)$ind.production)
  IBM_IC_long[IBM_IC_long$sim == x, "Dorm"] <- mean(subset(IBM_IC, IBM_IC$sim == x)$dormant.total.abundance)
  IBM_IC_long[IBM_IC_long$sim == x, "S"] <- mean(subset(IBM_IC, IBM_IC$sim == x)$species.richness)
}

IBM_IC_long$N_turn <- IBM_IC_long$ct/(IBM_IC_long$V/IBM_IC_long$Q)
IBM_IC_long$turn <- IBM_IC_long$N_turn > 1
IBM_IC_long$ct_f <- as.factor(IBM_IC_long$ct)
IBM_IC_long$Tau <- log(IBM_IC_long$V/IBM_IC_long$Q, 10)
IBM_IC_long$DormPer <- (IBM_IC_long$Dorm/IBM_IC_long$N)*100

plot(IBM_IC_long$P ~ IBM_IC_long$Tau)
plot(IBM_IC_long$N ~ IBM_IC_long$Tau)
plot(IBM_IC_long$DormPer ~ IBM_IC_long$Tau)
```

##IBM - N
```{r}
N_IBM_gam <- gam(log(N, 10) ~ s(Tau), family = gaussian(link = "identity"), data = subset(IBM_IC_long, IBM_IC_long$N > 0), method = "REML")

summary(N_IBM_gam)

N_IBM <- predict_gam(N_IBM_gam) %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_point(data = subset(IBM_IC_long, IBM_IC_long$N > 0), aes(x = Tau, y = log(N, 10)))+
  ylab("N (number of individuals)")+
  theme(axis.title.x = element_blank())+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(N_IBM_gam)$dev, 2))))

N_IBM

ggsave("./output/IBM_N.pdf")
ggsave("./output/IBM_N.png", width = 6.5, height = 5)
```

#IBM - S
```{r}
S_IBM_gam <- gam(log(S, 10) ~ s(Tau), family = gaussian(link = "identity"), data = subset(IBM_IC_long, IBM_IC_long$S > 0), method = "REML")

summary(S_IBM_gam)

S_IBM <- predict_gam(S_IBM_gam) %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_point(data = subset(IBM_IC_long, IBM_IC_long$S > 0), aes(x = Tau, y = log(S, 10)))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(0, 3), breaks = c(0, 1, 2, 3))+
  ylab("Observed S")+
  theme(axis.title.x = element_blank())+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(S_IBM_gam)$dev, 2))))

S_IBM

ggsave("./output/IBM_S.pdf")
ggsave("./output/IBM_S.png", width = 6.5, height = 5)
```
#IBM - Dorm
```{r}
Dorm_IBM_gam <- gam(DormPer ~ s(Tau), family = gaussian(link = "identity"), data = IBM_IC_long, method = "REML")

summary(Dorm_IBM_gam)

Dorm_IBM <- predict_gam(Dorm_IBM_gam) %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.35)+
  geom_point(data = IBM_IC_long, aes(x = Tau, y = DormPer))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab("Dormant cells (% of total)")+
  xlab(expression(paste(tau, " (time steps)")))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(Dorm_IBM_gam)$dev, 2))))+
  theme(plot.margin = unit(c(0.2, 0.2, 0.2, 0.4), "cm"))

Dorm_IBM

ggsave("./output/IBM_Dorm.pdf")
ggsave("./output/IBM_Dorm.png", width = 6.5, height = 5)
```
#IBM - P
```{r}
P_IBM_gam <- gam(log(P + 1, 10) ~ s(Tau), family = gaussian(link = "identity"), data = IBM_IC_long, method = "REML")

summary(P_IBM_gam)

P_IBM <- predict_gam(P_IBM_gam) %>%
  ggplot(aes(Tau, fit))+
  geom_smooth_ci(color = "blue", cex = 1.25, ci_alpha = 0.4)+
  geom_point(data = IBM_IC_long, aes(x = Tau, y = log(P + 1, 10)))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force))+
  ylab("Productivity")+
  xlab(expression(paste(tau, " (time steps)")))+
  labs(title = bquote("Deviance explained =" ~ .(signif(summary(P_IBM_gam)$dev, 2))))+
  theme(plot.margin = unit(c(1.5, 0.2, 0, 0.2), "cm"))

P_IBM

ggsave("./output/IBM_P.pdf")
ggsave("./output/IBM_P.png", width = 6.5, height = 5)
```

```{r}
ggdraw()+
  draw_plot(N_IBM, x = 0, y = 0.65, width = 1, height = 0.3) + 
  draw_plot(S_IBM, x = 0, y = 0.35, width = 1, height = 0.3) + 
  draw_plot(Dorm_IBM, x = 0, y = 0.0, width = 1, height = 0.35)+
  draw_plot_label(label = c("A", "B", "C"), size = 30, x = c(0.05,0.05,0.05), y = c(0.98,0.67,0.4))+
  theme(plot.background = element_rect(fill="white", color = NA))

ggsave("./output/IBM_results.pdf")
ggsave("./output/IBM_results.png", width = 5, height = 10)

```

#Figure S1.
##Total number of resources used at 48 hours
```{r}
NR_lm <- lm(NumRes ~ Tau, data = Tau)
NR_lm_re <- lmer(NumRes ~ Tau + (1|Set), data = Tau)
NR_lm_2 <- lm(NumRes ~ poly(Tau, 2, raw = TRUE), data = Tau)
summary(NR_lm)
AIC(NR_lm)
AIC(NR_lm_2)
cor(Tau$Tau, Tau$NumRes)

anova(NR_lm_re, NR_lm_2, NR_lm)

NumRes <- ggplot(Tau, aes(x = Tau, y = NumRes))+
  geom_point(size = 2, alpha = 0.6)+
  ylim(c(20, 33))+
  xlab(expression(paste(tau, " (h)")))+
  ylab("Number of Resources Used \n at 48 hours")+
  geom_smooth(method = "lm", formula = y~x, color = "blue", fill = alpha("black", 0.4))+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),
               formula = y~x, parse = TRUE, size = 4)+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))


NumRes

ggsave("./output/RTLC_NumRes.pdf")
ggsave("./output/RTLC_NumRes.png")
```

#Pairwise OTU dist vs. Res dist
```{r}
test.pw <- data.frame(site1=numeric(),
                        site2=numeric(), 
                        otupw=numeric(), 
                         ecopw=numeric())
rownames(EcoPlate) <- EcoPlate_env$Tau
x <- 2.5
y <- 3
for(x in row.names(OTUsr_20)){
  for(y in row.names(OTUsr_20)){
    test.pw <- rbind(test.pw, c(x, y, vegdist(OTUsr_20[c(x,y),], method = "bray", upper = TRUE, diag = TRUE)[1], vegdist(EcoPlate[c(x,y),], method = "euclidean", upper = TRUE, diag = TRUE)[1]))
  }
}

colnames(test.pw) <- c("site1", "site2", "otupw", "ecopw")
test.pw$site1 <- as.numeric(test.pw$site1)
test.pw$site1 <- log(((10^test.pw$site1)/60), 10)
test.pw$site2 <- as.numeric(test.pw$site2)
test.pw$site2 <- log(((10^test.pw$site2)/60), 10)
test.pw$otupw <- as.numeric(test.pw$otupw)
test.pw$ecopw <- as.numeric(test.pw$ecopw)
test.pw$taudiff <- test.pw$site2 - test.pw$site1
test.pw[1, "site1"]
for(row in row(test.pw)){
  if((test.pw[row, "site1"] > 2.6 && test.pw[row, "site2"] > 2.6)){
    test.pw[row, "Turn"] <- "Less"
  }
  else if((test.pw[row, "site1"] < 2.6 && test.pw[row, "site2"] < 2.6)){
    test.pw[row, "Turn"] <- "More"
  }
  else{
    test.pw[row, "Turn"] <- NA
  }
}

pw_lm <- lm(otupw ~ ecopw * Turn, data = subset(test.pw, is.na(test.pw$Turn) == FALSE))
AIC(pw_lm)
summary(pw_lm)
anova
mantel(tau.db, EP.db.20)


pw.bc <- ggplot(data = subset(test.pw, is.na(test.pw$Turn) == FALSE), aes(x = ecopw, y = otupw, group = Turn))+
  geom_point(aes(color = Turn))+
  xlab("Resource use distance")+
  ylab("Community distance")+
  scale_color_manual(values = c(alpha("green",0.3), alpha("#FC0FC0", 0.3)), labels = c("No", "Yes"))+
  labs(color = "Full Turnover")+
  geom_smooth(method = "lm", formula = y ~ x, color = "blue", cex = 1.25, fill = alpha("black", 0.4))+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),
               formula = y~x, parse = TRUE, size = 4, label.x = "right", label.y= "bottom")

pw.bc

ggsave("./output/RTLC_BC_PW.pdf")
ggsave("./output/RTLC_BC_PW.png")
```



#Unused figures

```{r}
for(x in unique(imp.spp$Phylum)){
  list <- data.frame(rep(0, 35))
  colnames(list) <- c(as.character(x))
  for(y in imp.spp$OTU){
    if(imp.spp[y, "Phylum"] == x){
      list[,x] <- list[,x] + as.data.frame(imp.otus[,y])
    }
  }
  imp.otus[,as.character(x)] <- list[,x]
}

imp.phylum <- cbind(imp.otus$Tau, imp.otus[ ,338:357])
colnames(imp.phylum) <- c("Tau", colnames(imp.otus[,338:357]))
imp.phylum.long <- gather(imp.phylum, phylum, REL, Proteobacteria:Hydrogenedentes)

imp.phylum.long$Tau <- as.numeric(imp.phylum.long$Tau)

imp_phylum_plot <- ggplot(imp.phylum.long, aes(x = Tau, y = REL, group = phylum))+
  geom_smooth(method = "loess", se = F, aes(color = phylum))+
  ylab("Relative abundance")+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  labs(color = "Phylum")+
  xlab(expression(paste(tau, " (h)")))+
  theme(legend.title = element_text(size = 20), legend.text = element_text(size = 15), axis.title = element_text(size = 20))
imp_phylum_plot

ggsave("./output/RTLC_imp_phylum.pdf")
ggsave("./output/RTLC_imp_phylum.png", width = 12, height = 8)
```


##Individual Productivity (uM C/hr/Cell)
```{r}
IP_lm <- lm(ind_P ~ Tau, data = Tau)
IP_lm_2 <- lm(log(ind_P, 10) ~ poly(Tau, 2, raw = TRUE), data = Tau)
IP_exp <- lm(log(ind_P, 10) ~ Tau, data = Tau)
summary(IP_lm)
summary(IP_lm_2)
summary(IP_exp)
AIC(IP_lm)
AIC(IP_lm_2)
AIC(IP_exp)
plot(IP_lm_2)
plot(IP_lm)
plot(IP_exp)

IP <- ggplot(Tau, aes(x = Tau, y = log(ind_P, 10)))+
  geom_point(size = 2, alpha = 0.6)+
  xlab(expression(paste(tau, " (h)")))+
  ylab(expression(atop("Biomass Production", paste("(", mu, "M C/hr/Cell)"))))+
  geom_smooth(method = "lm", formula = y ~ poly(x, 2, raw = TRUE))+
  stat_poly_eq(aes(label =  paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),
               label.x = "right", label.y = "top", formula = y ~ poly(x, 2, raw = TRUE), parse = TRUE, size = 4)+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force))+
  labs(title = bquote("R"^2~ "=" ~ .(signif(summary(N_gam_re)$r.sq, 3))))

IP

ggsave("./output/RTLC_IP.pdf")
ggsave("./output/RTLC_IP.png")

```

##Biofilm Formation
```{r}

OT_N_re <- lmer(log(OT/N) ~ Tau + (1|Set), data = Tau)
OT_N <- lm(log(OT/N) ~ Tau, data = Tau)

summary(OT_N_re)
summary(OT_N)
AIC(OT_N, OT_N_re)
anova(OT_N_re, OT_N)


Sig <- car::Anova(OT_N_re, test.statistic = "F")

OT_Tau <- ggplot(Tau, aes(x = Tau, y = log(OT/N)))+
  geom_point(size = 2, alpha = 0.6)+
  xlab(expression(paste(tau, " (h)")))+
  ylab("log(Biofilm Formation/Cell) \n OD 595 nm")+
  geom_smooth(method = "lm")+
  labs(title = paste(" P =", Sig$`Pr(>F)`[1]))+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  labs(title = bquote("R"^2~ "=" ~ .(signif(summary(N_gam_re)$r.sq, 3))))

OT_Tau

ggsave("./output/RTLC_OT.pdf")
ggsave("./output/RTLC_OT.png")
```


##Plot resource use vs. Time for all Tau

```{r}
Avg_Hr <- ggplot(subset(long_EP, long_EP$C_Source =="Avg"), aes(x = Hours, y = OD, group = Tau, color = Tau))+
  geom_line(size = 1)+
  geom_point(size = 3)+
  xlab("Hours")+
  ylab("Average Well Response (OD590)")+
  labs(group = "Tau")+
  scale_color_continuous(type = "viridis", labels = label_math(expr = 10^.x, format = force))+
  theme(legend.title = element_text(size = 16))
  
Avg_Hr

ggsave("./output/RTLC_AvgRes_Hr.pdf")
ggsave("./output/RTLC_AvgRes_Hr.png")

```

#Euclidean distance Resource use
```{r}
EP.eu <-vegdist(EcoPlate, method = "euclidean")
EP.pca <- cmdscale(EP.eu, eig = TRUE)
EP.pca <- add.spec.scores(EP.pca, EcoPlate, method = "pcoa.scores")

explainvar1 <- round(EP.pca$eig[1] / sum(EP.pca$eig), 3) * 100
explainvar2 <- round(EP.pca$eig[2] / sum(EP.pca$eig), 3) * 100
explainvar3 <- round(EP.pca$eig[3] / sum(EP.pca$eig), 3) * 100

EP.plot <- as.data.frame(EP.pca$points)
EP.plot <- cbind(EP.plot, Tau$Tau, Tau$Turn)
colnames(EP.plot) <- c("V1", "V2", "Tau", "Turn")
EP.cproj <- as.data.frame(EP.pca$cproj)
EP.cproj <- cbind(EP.cproj, as.data.frame(row.names(EP.pca$cproj)))

EP_EU <- ggplot(EP.plot, aes(x = V1, y = V2, colour = Turn))+
  geom_point(cex = 3)+
  xlab(paste("PCA 1 (", explainvar1, "%)", sep = ""))+
  ylab(paste("PCA 2 (", explainvar2, "%)", sep = ""))+
  labs(color = "Full Turnover")+
  scale_color_manual(values = c(alpha("green",0.3), alpha("#FC0FC0", 0.3)), labels = c("No", "Yes"))+
  theme(legend.title = element_text(size = 16))


EP_EU

ggsave("./output/RTLC_Res_Eu_PCA.pdf")
ggsave("./output/RTLC_Res_Eu_PCA.png")

```

```{r}
Res_heat <- as.data.frame(colnames(EcoPlate))
colnames(Res_heat) <- c("Res")
Res_heat$N <- colSums(EcoPlate)
rownames(EcoPlate) <- c(EcoPlate_env$Tau)
 

for (res in Res_heat$Res){
  x = 0
  for(site in rownames(EcoPlate)){
    x = x + (EcoPlate[site, res] * as.numeric(site))
  }
  Res_heat$Tau[Res_heat$Res == res] <- x/Res_heat$N[Res_heat$Res == res]
}

colnames(Res_heat) <- c("Res", "N", "Weight")

Res_heat <- Res_heat[order(Res_heat$Weight),]

EcoPlate <- cbind(EcoPlate_env, EcoPlate)

colnames(EcoPlate)

#########
eco_long <- gather(EcoPlate, Res, N, X2.Hydroxy.Benzoic.Acid:Tween.80, factor_key = TRUE)
eco_long$Res <- factor(eco_long$Res, levels = rev(as.list(Res_heat$Res)))
print(levels(eco_long$Res))

eco_long$Tau <- as.factor(eco_long$Tau)

ecoplate_heat <- ggplot(eco_long, aes(Tau, Res, fill = N))+
  geom_tile()+
  theme(axis.ticks.x = element_blank(), axis.ticks.y = element_blank(), axis.text.x = element_blank(), axis.text.y = element_text(size = 10))+
  scale_fill_continuous(type = "viridis")

ecoplate_heat

ggsave("./output/res_heatmap.pdf")
ggsave("./output/res_heatmap.png")
```

#Unused figures - Sequence Based

##Generate heat map of Day 20 communities with BC distance
```{r}
OTUsr_20_or <- OTUsr_20[order(rownames(OTUsr_20)),]

tau.db <- vegdist(OTUsr_20_or, method = "bray", upper = TRUE, diag = TRUE)
order <- rev(attr(tau.db, "Labels"))

pdf(file = "./output/RTLC_BC_heat.pdf")
jpeg(file = "./output/RTLC_BC_heat.jpeg")
levelplot(as.matrix(tau.db) [, order], aspect = "iso", col.regions = inferno, xlab = "log(Tau, 10)", ylab = "log(Tau, 10)", scales = list(x=list(at=seq(1,49,1), labels = c("1.5", rep("",2) , "2.5", rep("",4),  "3.5", rep("", 3),"4", rep("", 2), "4.5", rep("",4), "5.25", rep("", 3), "6", rep("", 5), "7", rep("", 3), "8")), y=list(at=seq(1,49,1), labels = rev(c("1.5", rep("",2) , "2.5", rep("",4),  "3.5", rep("", 3),"4", rep("", 2), "4.5", rep("",4), "5.25", rep("", 3), "6", rep("", 5), "7", rep("", 3), "8")))), main = "Bray-Curtis Distance")
dev.off()
```

##Want OTUs by pH of site heat map to show horseshoe
```{r}
# OTU_r_trim <- OTU_r[1:69,]
# OTU_heat <- as.data.frame(colnames(OTU_r_trim))
# colnames(OTU_heat) <- c("OTU")
# OTU_heat$N <- colSums(OTU_r_trim)
#  
# 
# for (otu in OTU_heat$OTU){
#   x = 0
#   for(site in rownames(OTU_r_trim)){
#     x = x + (OTU_r_trim[site, otu] * Tau_Seq$Tau[Tau_Seq$Seq_Sample == site])
#   }
#   OTU_heat$Tau[OTU_heat$OTU == otu] <- x/OTU_heat$N[OTU_heat$OTU == otu]
# }
# 
# OTU_heat <- OTU_heat[order(OTU_heat$Tau),]
# 
# OTU_heat <- OTU_heat[order(OTU_heat$Weight),]
# 
# colnames(OTU_heat) <- c("OTU", "N", "Weight")
# 
# OTU_heat <- cbind(OTU_heat, Tau_Seq[1:69,"Tau"])
# 
# plot(x = as.numeric(OTUsr_20$Tau), y = OTUsr_20$Otu00628)
# #########
# 
# colnames(OTUsr_20)[1] <- c("Tau")
# data_long <- gather(OTUsr_20, OTU, N, Otu00001:Otu33565, factor_key = TRUE)
# data_long$OTU <- factor(data_long$OTU, levels = rev(as.list(OTU_heat$OTU)))
# print(levels(data_long$OTU))
# 
# map <- ggplot(data_long, aes(Tau, OTU, fill = log(N, 10)))+
#   geom_tile()+
#   theme(axis.ticks.x = element_blank(), axis.ticks.y = element_blank(), axis.text.x = element_blank(), axis.text.y = element_blank())+
#   scale_fill_continuous(type = "viridis", labels = label_math(expr = 10^.x, format = force))
# 
# map
# 
# ggsave("./output/map.pdf")
# ggsave("./output/map.png")

```

##Species Turnover (Whittaker's Turnover between two sites)
```{r}
#Betaw = (S1 - gamma) + (S2 - gamma)/mean(S1, S2)
Betaw <- function(site1 = "", site2 = ""){
  site1 <- t(OTUs.PA[site1,])
  site2 <- t(OTUs.PA[site2,])
  site1 <- as.data.frame(subset(site1, select = site1 > 0))
  site2 <- as.data.frame(subset(site2, select = site2 > 0))
  gamma <- as.numeric(length(intersect(colnames(site1), colnames(site2))))
  bw <- ((ncol(site1) - gamma) + (ncol(site2) - gamma))/mean(ncol(site1), ncol(site2))
  return(bw)
}

Tau$Betaw <- NA
row <- 1
while(row < nrow(Tau)){
  if(!is.na(Tau[row, "Day_0_Seq"]) && !is.na(Tau[row, "Day_20_Seq"])){
    Tau[row, "Betaw"] <- Betaw(Tau[row, "Day_0_Seq"], Tau[row, "Day_20_Seq"])
  }
  row = row + 1
}

Bw_lm <- lm(Betaw ~ Tau, data = Tau)
Bw_poly <- lm(Betaw ~ poly(Tau, 2, raw = TRUE), data = Tau)
Bw_gam <- gam(Betaw ~ s(Tau), family = gaussian(link = "identity"), data = Tau, method = "REML")
Bw_gam_re <- gam(Betaw ~ s(Tau) + s(Set, bs = "re"), family = gaussian(link = "identity"), data = Tau, method = "REML")

AIC(Bw_lm, Bw_poly, Bw_gam, Bw_gam_re)
anova(Bw_lm, Bw_poly, Bw_gam, Bw_gam_re)

summary(Bw_lm)
summary(Bw_poly)
summary(Bw_gam)
summary(Bw_gam_re)

k.check(Bw_gam)
k.check(Bw_gam_re)
gam.check(Bw_gam)
gam.check(Bw_gam_re)

betaw <- ggplot(data = Tau, aes(x = Tau, y = Betaw))+
  geom_point()+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force))+
  geom_smooth(method = "lm", formula = y ~ poly(x, 2, raw = TRUE))+
  xlab(expression(paste(tau, " (h)")))+
  stat_poly_eq(aes(label = paste(stat(rr.label), "*\" and \"*", stat(p.value.label), sep = "")),
            label.x = "left", label.y = "top", formula = y~poly(x, 2, raw = TRUE), parse = TRUE, size = 4)+
  ylab(expression(beta[W]))
  
betaw

ggsave("./output/RTLC_Bw.pdf")
ggsave("./output/RTLC_Bw.png")
```
#Unused Plots - IBMs

```{r}
#IBM <- read.csv("data/IBM/SimData.csv", header = TRUE, sep = ",")

#IBM_2 <- subset(IBM, log(IBM$V, 10) <= 2 & log(IBM$V, 10) > 1)
#IBM_sum <- as.data.frame(unique(IBM_2$sim))
#colnames(IBM_sum) <- c("sim")

#for(x in unique(IBM_sum$sim)){
#  IBM_sum[IBM_sum$sim == x, "N"] <- mean(subset(IBM_2, IBM_2$sim == x)$total.abundance)
#  IBM_sum[IBM_sum$sim == x, "V"] <- mean(subset(IBM_2, IBM_2$sim == x)$V)
#  IBM_sum[IBM_sum$sim == x, "Q"] <- mean(subset(IBM_2, IBM_2$sim == x)$Q)
#  IBM_sum[IBM_sum$sim == x, "S"] <- mean(subset(IBM_2, IBM_2$sim == x)$species.richness)
#  IBM_sum[IBM_sum$sim == x, "E"] <- mean(subset(IBM_2, IBM_2$sim == x)$simpson.e, na.rm = TRUE)
#  IBM_sum[IBM_sum$sim == x, "P"] <- mean(subset(IBM_2, IBM_2$sim == x)$ind.production, na.rm = TRUE)
#  IBM_sum[IBM_sum$sim == x, "B"] <- mean(subset(IBM_2, IBM_2$sim == x)$whittakers.turnover, na.rm = TRUE)
#  IBM_sum[IBM_sum$sim == x, "Res"] <- mean(subset(IBM_2, IBM_2$sim == x)$avg.per.capita.efficiency1e, na.rm = TRUE)
#}

#write.csv(IBM_sum, "data/IBM/IBM_sum.csv", row.names = FALSE)

IBM_sum <- read.csv("data/IBM/IBM_sum.csv", header = TRUE, sep = ",")
```

```{r}
N_IBM <- ggplot(data = subset(IBM_sum, log(IBM_sum$V, 10) <= 2 & log(IBM_sum$V, 10) > 1), aes(y = log(N, 10), x = log(V/Q, 10)))+
  geom_point()+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(1, 5))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force))+
  xlab(expression(paste(tau)))+
  ylab(expression(paste("Total abundance, ", italic("N"))))+
  stat_smooth(method = "loess", color = "blue", cex = 1.25, level = 0.95, fill = alpha("black", 0.4))+
  theme(axis.title = element_text(size = 25))

N_IBM


ggsave("./output/IBM_N.pdf")
ggsave("./output/IBM_N.png", width = 6.5, height = 5)


S_IBM <- ggplot(data = subset(IBM_sum, log(IBM_sum$V, 10) <= 2 & log(IBM_sum$V, 10) > 1), aes(y = log(S, 10), x = log(V/Q, 10)))+
  geom_point()+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(1, 5))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(0, 1.5))+
  xlab(expression(paste(tau)))+
  ylab(expression(paste("Species richness, ", italic("S"))))+
  stat_smooth(method = "loess", color = "blue", cex = 1.25, level = 0.95, fill = alpha("black", 0.4))+
  theme(axis.title = element_text(size = 25))

S_IBM

ggsave("./output/IBM_S.pdf")
ggsave("./output/IBM_S.png", width = 6.5, height = 5)


E_IBM <- ggplot(data = subset(IBM_sum, log(IBM_sum$V, 10) <= 2 & log(IBM_sum$V, 10) > 1), aes(y = log(E, 10), x = log(V/Q, 10)))+
  geom_point()+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(1, 5))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(-0.5, 0))+
  xlab(expression(paste(tau)))+
  ylab(expression(paste("Species evenness, ", italic("E"))))+
  stat_smooth(method = "loess", color = "blue", cex = 1.25, level = 0.95, fill = alpha("black", 0.4))+
  theme(axis.title = element_text(size = 25))

E_IBM

ggsave("./output/IBM_E.pdf")
ggsave("./output/IBM_E.png", width = 6.5, height = 5)


P_IBM <- ggplot(data = subset(IBM_sum, log(IBM_sum$V, 10) <= 2 & log(IBM_sum$V, 10) > 1), aes(y = log(P, 10), x = log(V/Q, 10)))+
  geom_point()+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(1, 5))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(-2, 1.5))+
  xlab(expression(paste(tau)))+
  ylab(expression(paste("Productivity, ", italic("P"))))+
  stat_smooth(method = "loess", color = "blue", cex = 1.25, level = 0.95, fill = alpha("black", 0.4))+
  theme(axis.title = element_text(size = 25))

P_IBM


ggsave("./output/IBM_P.pdf")
ggsave("./output/IBM_P.png", width = 6.5, height = 5)


B_IBM <- ggplot(data = subset(IBM_sum, log(IBM_sum$V, 10) <= 2 & log(IBM_sum$V, 10) > 1), aes(y = B, x = log(V/Q, 10)))+
  geom_point()+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(1, 5))+
  ylim(c(0,2))+ 
  xlab(expression(paste(tau)))+
  ylab(expression(paste("Species turnover, ", beta[w])))+
  stat_smooth(method = "loess", color = "blue", cex = 1.25, level = 0.95, fill = alpha("black", 0.4))+
  theme(axis.title = element_text(size = 25))

B_IBM


ggsave("./output/IBM_B.pdf")
ggsave("./output/IBM_B.png", width = 6.5, height = 5)

Res_IBM <- ggplot(data = subset(IBM_sum, log(IBM_sum$V, 10) <= 2 & log(IBM_sum$V, 10) > 1), aes(y = log(Res, 10), x = log(V/Q, 10)))+
  geom_point()+
  scale_x_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(1, 5))+
  scale_y_continuous(labels = label_math(expr = 10^.x, format = force), limits = c(-2, 1))+
  xlab(expression(paste(tau)))+
  ylab("Resource specialization")+
  stat_smooth(method = "loess", color = "blue", cex = 1.25, level = 0.95, fill = alpha("black", 0.4))+
  theme(axis.title = element_text(size = 25))

Res_IBM


ggsave("./output/IBM_Res.pdf")
ggsave("./output/IBM_Res.png", width = 6.5, height = 5)
```